// @backstage-community/plugin-tech-insights-react/dist/api/TechInsightsClient.esm.js
import { TechInsightsClient as TechInsightsClient$1 } from '@backstage-community/plugin-tech-insights-common/client';
import { jsonRulesEngineCheckResultRenderer } from '../components/CheckResultRenderer.esm.js';

class TechInsightsClient extends TechInsightsClient$1 {
  renderers;
  customGetEntityLinks;
  constructor(options) {
    super(options);
    this.renderers = options.renderers;
    this.customGetEntityLinks = options.getEntityLinks ?? (() => []);
  }
  /**
   * Get the check result renderers for the given check types.
   *
   * @param types - The types of checks to get renderers for
   * @returns An array of check result renderers
   *
   * @public
   */
  getCheckResultRenderers(types) {
    const renderers = this.renderers ?? [jsonRulesEngineCheckResultRenderer];
    return renderers.filter((d) => types.includes(d.type));
  }
  /**
   * Determine if a check result represents a failure.
   *
   * @param check - The check result to evaluate
   * @returns true if the check result represents a failure, false otherwise
   *
   * @public
   */
  isCheckResultFailed(check) {
    const checkResultRenderers = this.getCheckResultRenderers([
      check.check.type
    ]);
    if (checkResultRenderers[0] && checkResultRenderers[0].isFailed) {
      return checkResultRenderers[0].isFailed(check);
    }
    return true;
  }
  /**
   * Get the links for a given check result and entity.
   *
   * @param result - The check result to get links for
   * @param entity - The entity to get links for
   * @param options - Optional options for the links
   * @returns An array of check links
   *
   * @public
   */
  getLinksForEntity(result, entity, options = {}) {
    const links = this.customGetEntityLinks(result, entity);
    if (options.includeStaticLinks) {
      links.push(...result.check.links ?? []);
    }
    return links;
  }
}

export { TechInsightsClient };
