// @backstage-community/plugin-tech-insights/dist/components/ScorecardsList/ScorecardsList.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { useApi } from '@backstage/core-plugin-api';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
import { MarkdownContent } from '@backstage/core-components';
import { techInsightsApiRef, ResultCheckIcon } from '@backstage-community/plugin-tech-insights-react';
import Tooltip from '@material-ui/core/Tooltip';

const useStyles = makeStyles((theme) => ({
  listItemText: {
    paddingRight: theme.spacing(0.5)
  }
}));
const itemTooltip = (children, title, enabled) => enabled ? /* @__PURE__ */ jsx(Tooltip, { title, children }) : children;
const ScorecardsList = (props) => {
  const { checkResults, entity, dense, hideDescription } = props;
  const classes = useStyles();
  const api = useApi(techInsightsApiRef);
  const types = [...new Set(checkResults.map(({ check }) => check.type))];
  const checkResultRenderers = api.getCheckResultRenderers(types);
  return /* @__PURE__ */ jsx(List, { dense, disablePadding: true, children: checkResults.map((result, index) => {
    const checkResultRenderer = checkResultRenderers.find(
      (renderer2) => renderer2.type === result.check.type
    );
    const renderer = checkResultRenderer?.description;
    const description = renderer ? renderer(result) : /* @__PURE__ */ jsx(MarkdownContent, { content: result.check.description });
    return itemTooltip(
      /* @__PURE__ */ jsxs(ListItem, { disableGutters: true, children: [
        /* @__PURE__ */ jsx(
          ListItemText,
          {
            primary: result.check.name,
            ...!props.hideDescription ? {
              secondary: description
            } : {},
            className: classes.listItemText
          },
          index
        ),
        /* @__PURE__ */ jsx(
          ResultCheckIcon,
          {
            result,
            entity,
            checkResultRenderer
          }
        )
      ] }, result.check.id),
      description,
      hideDescription
    );
  }) });
};

export { ScorecardsList };
