// @backstage-community/plugin-sonarqube/dist/api/SonarQubeClient.esm.js
import fetch from 'cross-fetch';

class SonarQubeClient {
  discoveryApi;
  identityApi;
  constructor(options) {
    this.discoveryApi = options.discoveryApi;
    this.identityApi = options.identityApi;
  }
  async callApi(path, query) {
    const { token: idToken } = await this.identityApi.getCredentials();
    const apiUrl = `${await this.discoveryApi.getBaseUrl("sonarqube")}`;
    const response = await fetch(
      `${apiUrl}/${path}?${new URLSearchParams(query).toString()}`,
      {
        headers: {
          "Content-Type": "application/json",
          ...idToken && { Authorization: `Bearer ${idToken}` }
        }
      }
    );
    if (response.status === 200) {
      return await response.json();
    }
    return void 0;
  }
  async getFindingSummary({
    componentKey,
    projectInstance
  } = {}) {
    if (!componentKey) {
      return void 0;
    }
    const instanceKey = projectInstance || "";
    const metrics = {
      alert_status: void 0,
      bugs: void 0,
      reliability_rating: void 0,
      vulnerabilities: void 0,
      security_rating: void 0,
      security_hotspots_reviewed: void 0,
      security_review_rating: void 0,
      code_smells: void 0,
      sqale_rating: void 0,
      coverage: void 0,
      duplicated_lines_density: void 0
    };
    const baseUrlWrapper = await this.callApi(
      "instanceUrl",
      {
        instanceKey
      }
    );
    let baseUrl = baseUrlWrapper?.instanceUrl;
    if (!baseUrl) {
      return void 0;
    }
    if (!baseUrl.endsWith("/")) {
      baseUrl += "/";
    }
    const findings = await this.callApi("findings", {
      componentKey,
      instanceKey
    });
    if (!findings) {
      return void 0;
    }
    findings.measures.forEach((m) => {
      metrics[m.metric] = m.value;
    });
    return {
      lastAnalysis: findings.analysisDate,
      metrics,
      projectUrl: `${baseUrl}dashboard?id=${encodeURIComponent(componentKey)}`,
      getIssuesUrl: (identifier) => `${baseUrl}project/issues?id=${encodeURIComponent(
        componentKey
      )}&types=${identifier.toLocaleUpperCase("en-US")}&resolved=false`,
      getComponentMeasuresUrl: (identifier) => `${baseUrl}component_measures?id=${encodeURIComponent(
        componentKey
      )}&metric=${identifier.toLocaleLowerCase(
        "en-US"
      )}&resolved=false&view=list`,
      getSecurityHotspotsUrl: () => `${baseUrl}${baseUrl === "https://sonarcloud.io/" ? "project/" : ""}security_hotspots?id=${encodeURIComponent(componentKey)}`
    };
  }
}

export { SonarQubeClient };
