// @backstage-community/plugin-jfrog-artifactory/dist/components/JfrogArtifactoryRepository/JfrogArtifactoryRepository.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { useState } from 'react';
import { useAsync } from 'react-use';
import { Progress, Table, Link } from '@backstage/core-components';
import { useApi } from '@backstage/core-plugin-api';
import { makeStyles, Box, Chip } from '@material-ui/core';
import { jfrogArtifactoryApiRef } from '../../api/index.esm.js';
import { useStyles, columns } from './tableHeading.esm.js';
import { formatByteSize, formatDate } from '../../utils.esm.js';

const useLocalStyles = makeStyles({
  chip: {
    margin: 0,
    marginRight: ".2em",
    height: "1.5em",
    "& > span": {
      padding: ".3em"
    }
  }
});
function JfrogArtifactoryRepository({ image, target }) {
  const jfrogArtifactoryClient = useApi(jfrogArtifactoryApiRef);
  const classes = useStyles();
  const localClasses = useLocalStyles();
  const [edges, setEdges] = useState([]);
  const titleprop = `Jfrog Artifactory repository: ${image}`;
  const { loading } = useAsync(async () => {
    const tagsResponse = await jfrogArtifactoryClient.getTags(image, target);
    setEdges(tagsResponse.data.versions.edges);
    return tagsResponse;
  });
  if (loading) {
    return /* @__PURE__ */ jsx(Progress, {});
  }
  const data = edges?.map((edge) => {
    const shortHash = edge.node.files.find((manifest) => manifest.name === "manifest.json")?.sha256.substring(0, 12);
    return {
      name: edge.node.name,
      last_modified: formatDate(edge.node.modified),
      size: formatByteSize(Number(edge.node.size)),
      manifest_digest: /* @__PURE__ */ jsxs(Box, { sx: { display: "flex", alignItems: "center" }, children: [
        /* @__PURE__ */ jsx(Chip, { label: "sha256", className: localClasses.chip }),
        shortHash
      ] }),
      repositories: `${edge.node.repos.length} | ${edge.node.repos.map((repo) => repo.name).join("| ")}`
    };
  });
  return /* @__PURE__ */ jsx("div", { style: { border: "1px solid #ddd" }, children: /* @__PURE__ */ jsx(
    Table,
    {
      title: titleprop,
      options: { paging: true, padding: "dense" },
      data,
      columns,
      emptyContent: /* @__PURE__ */ jsxs("div", { className: classes.empty, children: [
        "No data was added yet,\xA0",
        /* @__PURE__ */ jsx(Link, { to: "https://backstage.io/", children: "learn how to add data" }),
        "."
      ] })
    }
  ) });
}

export { JfrogArtifactoryRepository };
