// @backstage/ui/dist/components/Dialog/Dialog.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { Modal, Dialog as Dialog$1, Heading, DialogTrigger as DialogTrigger$1 } from 'react-aria-components';
import clsx from 'clsx';
import { RiCloseLine } from '@remixicon/react';
import { Button } from '../Button/Button.esm.js';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { DialogDefinition } from './definition.esm.js';
import { Flex } from '../Flex/Flex.esm.js';
import styles from './Dialog.module.css.esm.js';

const DialogTrigger = (props) => {
  return /* @__PURE__ */ jsx(DialogTrigger$1, { ...props });
};
const Dialog = forwardRef(
  (props, ref) => {
    const { classNames, cleanedProps } = useStyles(DialogDefinition, props);
    const { className, children, width, height, style, ...rest } = cleanedProps;
    return /* @__PURE__ */ jsx(
      Modal,
      {
        ref,
        className: clsx(classNames.overlay, styles[classNames.overlay]),
        isDismissable: true,
        isKeyboardDismissDisabled: false,
        ...rest,
        children: /* @__PURE__ */ jsx(
          Dialog$1,
          {
            className: clsx(
              classNames.dialog,
              styles[classNames.dialog],
              className
            ),
            style: {
              ["--bui-dialog-min-width"]: typeof width === "number" ? `${width}px` : width || "400px",
              ["--bui-dialog-min-height"]: height ? typeof height === "number" ? `${height}px` : height : "auto",
              ...style
            },
            children
          }
        )
      }
    );
  }
);
Dialog.displayName = "Dialog";
const DialogHeader = forwardRef((props, ref) => {
  const { classNames, cleanedProps } = useStyles(DialogDefinition, props);
  const { className, children, ...rest } = cleanedProps;
  return /* @__PURE__ */ jsxs(
    Flex,
    {
      ref,
      className: clsx(classNames.header, styles[classNames.header], className),
      ...rest,
      children: [
        /* @__PURE__ */ jsx(
          Heading,
          {
            slot: "title",
            className: clsx(classNames.headerTitle, styles[classNames.headerTitle]),
            children
          }
        ),
        /* @__PURE__ */ jsx(Button, { name: "close", "aria-label": "Close", variant: "tertiary", slot: "close", children: /* @__PURE__ */ jsx(RiCloseLine, {}) })
      ]
    }
  );
});
DialogHeader.displayName = "DialogHeader";
const DialogBody = forwardRef(
  (props, ref) => {
    const { classNames, cleanedProps } = useStyles(DialogDefinition, props);
    const { className, children, ...rest } = cleanedProps;
    return /* @__PURE__ */ jsx(
      "div",
      {
        className: clsx(classNames.body, styles[classNames.body], className),
        ref,
        ...rest,
        children
      }
    );
  }
);
DialogBody.displayName = "DialogBody";
const DialogFooter = forwardRef((props, ref) => {
  const { classNames, cleanedProps } = useStyles(DialogDefinition, props);
  const { className, children, ...rest } = cleanedProps;
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: clsx(classNames.footer, styles[classNames.footer], className),
      ...rest,
      children
    }
  );
});
DialogFooter.displayName = "DialogFooter";

export { Dialog, DialogBody, DialogFooter, DialogHeader, DialogTrigger };
