// @backstage/plugin-home/dist/components/VisitList/VisitListItem.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
import { ItemName } from './ItemName.esm.js';
import { ItemDetail } from './ItemDetail.esm.js';
import { ItemCategory } from './ItemCategory.esm.js';

const useStyles = makeStyles((_theme) => ({
  avatar: {
    minWidth: 0
  }
}));
const VisitListItem = ({
  visit,
  detailType
}) => {
  const classes = useStyles();
  return /* @__PURE__ */ jsxs(ListItem, { disableGutters: true, children: [
    /* @__PURE__ */ jsx(ListItemAvatar, { className: classes.avatar, children: /* @__PURE__ */ jsx(ItemCategory, { visit }) }),
    /* @__PURE__ */ jsx(
      ListItemText,
      {
        primary: /* @__PURE__ */ jsx(ItemName, { visit }),
        secondary: /* @__PURE__ */ jsx(ItemDetail, { visit, type: detailType }),
        disableTypography: true
      }
    )
  ] });
};

export { VisitListItem };
