// @backstage/plugin-home/dist/components/VisitList/VisitListSkeleton.esm.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import Collapse from '@material-ui/core/Collapse';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
import Skeleton from '@material-ui/lab/Skeleton';

const useStyles = makeStyles((_theme) => ({
  skeleton: {
    borderRadius: 30
  }
}));
const VisitListItemSkeleton = () => {
  const classes = useStyles();
  return /* @__PURE__ */ jsxs(ListItem, { disableGutters: true, children: [
    /* @__PURE__ */ jsx(ListItemAvatar, { children: /* @__PURE__ */ jsx(
      Skeleton,
      {
        className: classes.skeleton,
        variant: "rect",
        width: 50,
        height: 24
      }
    ) }),
    /* @__PURE__ */ jsx(
      ListItemText,
      {
        primary: /* @__PURE__ */ jsx(Skeleton, { variant: "text", width: "100%", height: 28 }),
        disableTypography: true
      }
    )
  ] });
};
const VisitListSkeleton = ({
  numVisitsOpen,
  numVisitsTotal,
  collapsed
}) => /* @__PURE__ */ jsxs(Fragment, { children: [
  Array(numVisitsOpen).fill(null).map((_e, index) => /* @__PURE__ */ jsx(VisitListItemSkeleton, {}, index)),
  numVisitsTotal > numVisitsOpen && /* @__PURE__ */ jsx(Collapse, { in: !collapsed, children: Array(numVisitsTotal - numVisitsOpen).fill(null).map((_e, index) => /* @__PURE__ */ jsx(VisitListItemSkeleton, {}, index)) })
] });

export { VisitListSkeleton };
