// @backstage/ui/dist/components/Header/Header.esm.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { HeaderToolbar } from './HeaderToolbar.esm.js';
import { Tabs, TabList, Tab } from '../Tabs/Tabs.esm.js';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { HeaderDefinition } from './definition.esm.js';
import styles from './Header.module.css.esm.js';
import clsx from 'clsx';

const Header = (props) => {
  const { classNames, cleanedProps } = useStyles(HeaderDefinition, props);
  const {
    className,
    tabs,
    icon,
    title,
    titleLink,
    customActions,
    onTabSelectionChange
  } = cleanedProps;
  const hasTabs = tabs && tabs.length > 0;
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(
      HeaderToolbar,
      {
        icon,
        title,
        titleLink,
        customActions,
        hasTabs
      }
    ),
    tabs && /* @__PURE__ */ jsx(
      "div",
      {
        className: clsx(
          classNames.tabsWrapper,
          styles[classNames.tabsWrapper],
          className
        ),
        children: /* @__PURE__ */ jsx(Tabs, { onSelectionChange: onTabSelectionChange, children: /* @__PURE__ */ jsx(TabList, { children: tabs?.map((tab) => /* @__PURE__ */ jsx(
          Tab,
          {
            id: tab.id,
            href: tab.href,
            matchStrategy: tab.matchStrategy,
            children: tab.label
          },
          tab.id
        )) }) })
      }
    )
  ] });
};

export { Header };
