// @aws-sdk/client-codecommit/dist-es/schemas/schemas_0.js
const _A = "Approval";
const _AARTWR = "AssociateApprovalRuleTemplateWithRepository";
const _AARTWRI = "AssociateApprovalRuleTemplateWithRepositoryInput";
const _ADNEE = "ActorDoesNotExistException";
const _ADNEEu = "AuthorDoesNotExistException";
const _AL = "ApprovalList";
const _AR = "ApprovalRule";
const _ARCRE = "ApprovalRuleContentRequiredException";
const _ARDNEE = "ApprovalRuleDoesNotExistException";
const _AREM = "ApprovalRuleEventMetadata";
const _ARL = "ApprovalRulesList";
const _ARNAEE = "ApprovalRuleNameAlreadyExistsException";
const _ARNRE = "ApprovalRuleNameRequiredException";
const _AROEM = "ApprovalRuleOverriddenEventMetadata";
const _ART = "ApprovalRuleTemplate";
const _ARTCRE = "ApprovalRuleTemplateContentRequiredException";
const _ARTDNEE = "ApprovalRuleTemplateDoesNotExistException";
const _ARTIUE = "ApprovalRuleTemplateInUseException";
const _ARTNAEE = "ApprovalRuleTemplateNameAlreadyExistsException";
const _ARTNRE = "ApprovalRuleTemplateNameRequiredException";
const _ASCEM = "ApprovalStateChangedEventMetadata";
const _ASRE = "ApprovalStateRequiredException";
const _Ar = "Arn";
const _BAARTWR = "BatchAssociateApprovalRuleTemplateWithRepositories";
const _BAARTWRE = "BatchAssociateApprovalRuleTemplateWithRepositoriesError";
const _BAARTWREL = "BatchAssociateApprovalRuleTemplateWithRepositoriesErrorsList";
const _BAARTWRI = "BatchAssociateApprovalRuleTemplateWithRepositoriesInput";
const _BAARTWRO = "BatchAssociateApprovalRuleTemplateWithRepositoriesOutput";
const _BCIAACIASE = "BeforeCommitIdAndAfterCommitIdAreSameException";
const _BDARTFR = "BatchDisassociateApprovalRuleTemplateFromRepositories";
const _BDARTFRE = "BatchDisassociateApprovalRuleTemplateFromRepositoriesError";
const _BDARTFREL = "BatchDisassociateApprovalRuleTemplateFromRepositoriesErrorsList";
const _BDARTFRI = "BatchDisassociateApprovalRuleTemplateFromRepositoriesInput";
const _BDARTFRO = "BatchDisassociateApprovalRuleTemplateFromRepositoriesOutput";
const _BDMC = "BatchDescribeMergeConflicts";
const _BDMCE = "BatchDescribeMergeConflictsError";
const _BDMCEa = "BatchDescribeMergeConflictsErrors";
const _BDMCI = "BatchDescribeMergeConflictsInput";
const _BDMCO = "BatchDescribeMergeConflictsOutput";
const _BDNEE = "BranchDoesNotExistException";
const _BGC = "BatchGetCommits";
const _BGCE = "BatchGetCommitsError";
const _BGCEL = "BatchGetCommitsErrorsList";
const _BGCI = "BatchGetCommitsInput";
const _BGCO = "BatchGetCommitsOutput";
const _BGR = "BatchGetRepositories";
const _BGRE = "BatchGetRepositoriesError";
const _BGREL = "BatchGetRepositoriesErrorsList";
const _BGRI = "BatchGetRepositoriesInput";
const _BGRO = "BatchGetRepositoriesOutput";
const _BI = "BranchInfo";
const _BIDNEE = "BlobIdDoesNotExistException";
const _BIRE = "BlobIdRequiredException";
const _BM = "BlobMetadata";
const _BNEE = "BranchNameExistsException";
const _BNITNE = "BranchNameIsTagNameException";
const _BNRE = "BranchNameRequiredException";
const _C = "Comment";
const _CART = "CreateApprovalRuleTemplate";
const _CARTI = "CreateApprovalRuleTemplateInput";
const _CARTO = "CreateApprovalRuleTemplateOutput";
const _CB = "CreateBranch";
const _CBI = "CreateBranchInput";
const _CC = "CreateCommit";
const _CCI = "CreateCommitInput";
const _CCO = "CreateCommitOutput";
const _CCRE = "CommentContentRequiredException";
const _CCSLEE = "CommentContentSizeLimitExceededException";
const _CDARFTE = "CannotDeleteApprovalRuleFromTemplateException";
const _CDE = "CommentDeletedException";
const _CDNEE = "CommentDoesNotExistException";
const _CDNEEo = "CommitDoesNotExistException";
const _CFCC = "CommentsForComparedCommit";
const _CFCCD = "CommentsForComparedCommitData";
const _CFPR = "CommentsForPullRequest";
const _CFPRD = "CommentsForPullRequestData";
const _CIDNEE = "CommitIdDoesNotExistException";
const _CILEE = "CommitIdsLimitExceededException";
const _CILRE = "CommitIdsListRequiredException";
const _CIRE = "CommentIdRequiredException";
const _CIREo = "CommitIdRequiredException";
const _CM = "ConflictMetadata";
const _CMARFTE = "CannotModifyApprovalRuleFromTemplateException";
const _CML = "ConflictMetadataList";
const _CMLEE = "CommitMessageLengthExceededException";
const _CNCBCE = "CommentNotCreatedByCallerException";
const _COL = "CommitObjectsList";
const _CPR = "CreatePullRequest";
const _CPRAR = "CreatePullRequestApprovalRule";
const _CPRARI = "CreatePullRequestApprovalRuleInput";
const _CPRARO = "CreatePullRequestApprovalRuleOutput";
const _CPRI = "CreatePullRequestInput";
const _CPRO = "CreatePullRequestOutput";
const _CR = "ConflictResolution";
const _CRE = "CommitRequiredException";
const _CRI = "CreateRepositoryInput";
const _CRO = "CreateRepositoryOutput";
const _CRTRE = "ClientRequestTokenRequiredException";
const _CRUE = "ConcurrentReferenceUpdateException";
const _CRr = "CreateRepository";
const _CUMC = "CreateUnreferencedMergeCommit";
const _CUMCI = "CreateUnreferencedMergeCommitInput";
const _CUMCO = "CreateUnreferencedMergeCommitOutput";
const _Co = "Commit";
const _Com = "Comments";
const _Con = "Conflict";
const _Conf = "Conflicts";
const _D = "Difference";
const _DART = "DeleteApprovalRuleTemplate";
const _DARTFR = "DisassociateApprovalRuleTemplateFromRepository";
const _DARTFRI = "DisassociateApprovalRuleTemplateFromRepositoryInput";
const _DARTI = "DeleteApprovalRuleTemplateInput";
const _DARTO = "DeleteApprovalRuleTemplateOutput";
const _DB = "DeleteBranch";
const _DBCBDE = "DefaultBranchCannotBeDeletedException";
const _DBI = "DeleteBranchInput";
const _DBO = "DeleteBranchOutput";
const _DCC = "DeleteCommentContent";
const _DCCI = "DeleteCommentContentInput";
const _DCCO = "DeleteCommentContentOutput";
const _DF = "DeleteFile";
const _DFE = "DeleteFileEntry";
const _DFEe = "DeleteFileEntries";
const _DFI = "DeleteFileInput";
const _DFO = "DeleteFileOutput";
const _DL = "DifferenceList";
const _DMC = "DescribeMergeConflicts";
const _DMCI = "DescribeMergeConflictsInput";
const _DMCO = "DescribeMergeConflictsOutput";
const _DNCWFNE = "DirectoryNameConflictsWithFileNameException";
const _DPRAR = "DeletePullRequestApprovalRule";
const _DPRARI = "DeletePullRequestApprovalRuleInput";
const _DPRARO = "DeletePullRequestApprovalRuleOutput";
const _DPRE = "DescribePullRequestEvents";
const _DPREI = "DescribePullRequestEventsInput";
const _DPREO = "DescribePullRequestEventsOutput";
const _DR = "DeleteRepository";
const _DRI = "DeleteRepositoryInput";
const _DRO = "DeleteRepositoryOutput";
const _E = "Evaluation";
const _EICFE = "EncryptionIntegrityChecksFailedException";
const _EKADE = "EncryptionKeyAccessDeniedException";
const _EKDE = "EncryptionKeyDisabledException";
const _EKIIE = "EncryptionKeyInvalidIdException";
const _EKIUE = "EncryptionKeyInvalidUsageException";
const _EKNFE = "EncryptionKeyNotFoundException";
const _EKRE = "EncryptionKeyRequiredException";
const _EKUE = "EncryptionKeyUnavailableException";
const _EPRAR = "EvaluatePullRequestApprovalRules";
const _EPRARI = "EvaluatePullRequestApprovalRulesInput";
const _EPRARO = "EvaluatePullRequestApprovalRulesOutput";
const _F = "File";
const _FCASFSE = "FileContentAndSourceFileSpecifiedException";
const _FCRE = "FileContentRequiredException";
const _FCSLEE = "FileContentSizeLimitExceededException";
const _FCSLEEo = "FolderContentSizeLimitExceededException";
const _FDNEE = "FileDoesNotExistException";
const _FDNEEo = "FolderDoesNotExistException";
const _FERE = "FileEntryRequiredException";
const _FL = "FileList";
const _FLo = "FolderList";
const _FM = "FileMetadata";
const _FMRE = "FileModeRequiredException";
const _FMi = "FileModes";
const _FMil = "FilesMetadata";
const _FNCWDNE = "FileNameConflictsWithDirectoryNameException";
const _FPCWSPE = "FilePathConflictsWithSubmodulePathException";
const _FS = "FileSizes";
const _FTLE = "FileTooLargeException";
const _FV = "FileVersion";
const _Fo = "Folder";
const _GART = "GetApprovalRuleTemplate";
const _GARTI = "GetApprovalRuleTemplateInput";
const _GARTO = "GetApprovalRuleTemplateOutput";
const _GB = "GetBlob";
const _GBI = "GetBlobInput";
const _GBIe = "GetBranchInput";
const _GBO = "GetBlobOutput";
const _GBOe = "GetBranchOutput";
const _GBe = "GetBranch";
const _GC = "GetComment";
const _GCFCC = "GetCommentsForComparedCommit";
const _GCFCCI = "GetCommentsForComparedCommitInput";
const _GCFCCO = "GetCommentsForComparedCommitOutput";
const _GCFPR = "GetCommentsForPullRequest";
const _GCFPRI = "GetCommentsForPullRequestInput";
const _GCFPRO = "GetCommentsForPullRequestOutput";
const _GCI = "GetCommentInput";
const _GCIe = "GetCommitInput";
const _GCO = "GetCommentOutput";
const _GCOe = "GetCommitOutput";
const _GCR = "GetCommentReactions";
const _GCRI = "GetCommentReactionsInput";
const _GCRO = "GetCommentReactionsOutput";
const _GCe = "GetCommit";
const _GD = "GetDifferences";
const _GDI = "GetDifferencesInput";
const _GDO = "GetDifferencesOutput";
const _GF = "GetFile";
const _GFI = "GetFileInput";
const _GFIe = "GetFolderInput";
const _GFO = "GetFileOutput";
const _GFOe = "GetFolderOutput";
const _GFe = "GetFolder";
const _GMC = "GetMergeCommit";
const _GMCI = "GetMergeCommitInput";
const _GMCIe = "GetMergeConflictsInput";
const _GMCO = "GetMergeCommitOutput";
const _GMCOe = "GetMergeConflictsOutput";
const _GMCe = "GetMergeConflicts";
const _GMO = "GetMergeOptions";
const _GMOI = "GetMergeOptionsInput";
const _GMOO = "GetMergeOptionsOutput";
const _GPR = "GetPullRequest";
const _GPRAS = "GetPullRequestApprovalStates";
const _GPRASI = "GetPullRequestApprovalStatesInput";
const _GPRASO = "GetPullRequestApprovalStatesOutput";
const _GPRI = "GetPullRequestInput";
const _GPRO = "GetPullRequestOutput";
const _GPROS = "GetPullRequestOverrideState";
const _GPROSI = "GetPullRequestOverrideStateInput";
const _GPROSO = "GetPullRequestOverrideStateOutput";
const _GR = "GetRepository";
const _GRI = "GetRepositoryInput";
const _GRO = "GetRepositoryOutput";
const _GRT = "GetRepositoryTriggers";
const _GRTI = "GetRepositoryTriggersInput";
const _GRTO = "GetRepositoryTriggersOutput";
const _IAAE = "InvalidActorArnException";
const _IAAEn = "InvalidAuthorArnException";
const _IARCE = "InvalidApprovalRuleContentException";
const _IARNE = "InvalidApprovalRuleNameException";
const _IARTCE = "InvalidApprovalRuleTemplateContentException";
const _IARTDE = "InvalidApprovalRuleTemplateDescriptionException";
const _IARTNE = "InvalidApprovalRuleTemplateNameException";
const _IASE = "InvalidApprovalStateException";
const _IBF = "IsBinaryFile";
const _IBIE = "InvalidBlobIdException";
const _IBNE = "InvalidBranchNameException";
const _ICDLE = "InvalidConflictDetailLevelException";
const _ICE = "InvalidCommitException";
const _ICIE = "InvalidCommentIdException";
const _ICIEn = "InvalidCommitIdException";
const _ICRE = "InvalidConflictResolutionException";
const _ICRSE = "InvalidConflictResolutionStrategyException";
const _ICRTE = "InvalidClientRequestTokenException";
const _ICTE = "InvalidContinuationTokenException";
const _IDCSE = "InvalidDestinationCommitSpecifierException";
const _IDE = "InvalidDescriptionException";
const _IDPE = "InvalidDeletionParameterException";
const _IEE = "InvalidEmailException";
const _IFLE = "InvalidFileLocationException";
const _IFME = "InvalidFileModeException";
const _IFPE = "InvalidFilePositionException";
const _IMCFE = "InvalidMaxConflictFilesException";
const _IMMHE = "InvalidMaxMergeHunksException";
const _IMOE = "InvalidMergeOptionException";
const _IMRE = "InvalidMaxResultsException";
const _IOE = "InvalidOrderException";
const _IOSE = "InvalidOverrideStatusException";
const _IPCIE = "InvalidParentCommitIdException";
const _IPE = "InvalidPathException";
const _IPME = "IdempotencyParameterMismatchException";
const _IPRETE = "InvalidPullRequestEventTypeException";
const _IPRIE = "InvalidPullRequestIdException";
const _IPRSE = "InvalidPullRequestStatusException";
const _IPRSUE = "InvalidPullRequestStatusUpdateException";
const _IRAE = "InvalidResourceArnException";
const _IRCE = "InvalidReplacementContentException";
const _IRCSE = "InvalidRuleContentSha256Exception";
const _IRDE = "InvalidRepositoryDescriptionException";
const _IRFVEE = "InvalidRelativeFileVersionEnumException";
const _IRIE = "InvalidRevisionIdException";
const _IRNE = "InvalidReferenceNameException";
const _IRNEn = "InvalidRepositoryNameException";
const _IRTBNE = "InvalidRepositoryTriggerBranchNameException";
const _IRTCDE = "InvalidRepositoryTriggerCustomDataException";
const _IRTDAE = "InvalidRepositoryTriggerDestinationArnException";
const _IRTE = "InvalidReplacementTypeException";
const _IRTEE = "InvalidRepositoryTriggerEventsException";
const _IRTNE = "InvalidRepositoryTriggerNameException";
const _IRTRE = "InvalidRepositoryTriggerRegionException";
const _IRUAE = "InvalidReactionUserArnException";
const _IRVE = "InvalidReactionValueException";
const _ISBE = "InvalidSortByException";
const _ISCSE = "InvalidSourceCommitSpecifierException";
const _ISTUE = "InvalidSystemTagUsageException";
const _ITBE = "InvalidTargetBranchException";
const _ITE = "InvalidTargetException";
const _ITEn = "InvalidTargetsException";
const _ITEnv = "InvalidTitleException";
const _ITKLE = "InvalidTagKeysListException";
const _ITME = "InvalidTagsMapException";
const _L = "Location";
const _LAARTFR = "ListAssociatedApprovalRuleTemplatesForRepository";
const _LAARTFRI = "ListAssociatedApprovalRuleTemplatesForRepositoryInput";
const _LAARTFRO = "ListAssociatedApprovalRuleTemplatesForRepositoryOutput";
const _LART = "ListApprovalRuleTemplates";
const _LARTI = "ListApprovalRuleTemplatesInput";
const _LARTO = "ListApprovalRuleTemplatesOutput";
const _LB = "ListBranches";
const _LBI = "ListBranchesInput";
const _LBO = "ListBranchesOutput";
const _LFCH = "ListFileCommitHistory";
const _LFCHR = "ListFileCommitHistoryRequest";
const _LFCHRi = "ListFileCommitHistoryResponse";
const _LPR = "ListPullRequests";
const _LPRI = "ListPullRequestsInput";
const _LPRO = "ListPullRequestsOutput";
const _LR = "ListRepositories";
const _LRFART = "ListRepositoriesForApprovalRuleTemplate";
const _LRFARTI = "ListRepositoriesForApprovalRuleTemplateInput";
const _LRFARTO = "ListRepositoriesForApprovalRuleTemplateOutput";
const _LRI = "ListRepositoriesInput";
const _LRO = "ListRepositoriesOutput";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _MBBFF = "MergeBranchesByFastForward";
const _MBBFFI = "MergeBranchesByFastForwardInput";
const _MBBFFO = "MergeBranchesByFastForwardOutput";
const _MBBS = "MergeBranchesBySquash";
const _MBBSI = "MergeBranchesBySquashInput";
const _MBBSO = "MergeBranchesBySquashOutput";
const _MBBTW = "MergeBranchesByThreeWay";
const _MBBTWI = "MergeBranchesByThreeWayInput";
const _MBBTWO = "MergeBranchesByThreeWayOutput";
const _MBEE = "MaximumBranchesExceededException";
const _MCREE = "MultipleConflictResolutionEntriesException";
const _MCREEE = "MaximumConflictResolutionEntriesExceededException";
const _MFCTLEE = "MaximumFileContentToLoadExceededException";
const _MFEEE = "MaximumFileEntriesExceededException";
const _MH = "MergeHunk";
const _MHD = "MergeHunkDetail";
const _MHe = "MergeHunks";
const _MITCEE = "MaximumItemsToCompareExceededException";
const _MM = "MergeMetadata";
const _MMRE = "ManualMergeRequiredException";
const _MNOAEE = "MaximumNumberOfApprovalsExceededException";
const _MO = "MergeOperations";
const _MOPREE = "MaximumOpenPullRequestsExceededException";
const _MORE = "MergeOptionRequiredException";
const _MPRBFF = "MergePullRequestByFastForward";
const _MPRBFFI = "MergePullRequestByFastForwardInput";
const _MPRBFFO = "MergePullRequestByFastForwardOutput";
const _MPRBS = "MergePullRequestBySquash";
const _MPRBSI = "MergePullRequestBySquashInput";
const _MPRBSO = "MergePullRequestBySquashOutput";
const _MPRBTW = "MergePullRequestByThreeWay";
const _MPRBTWI = "MergePullRequestByThreeWayInput";
const _MPRBTWO = "MergePullRequestByThreeWayOutput";
const _MR = "MaxResults";
const _MRIPRE = "MultipleRepositoriesInPullRequestException";
const _MRNEE = "MaximumRepositoryNamesExceededException";
const _MRTAWRE = "MaximumRuleTemplatesAssociatedWithRepositoryException";
const _MRTEE = "MaximumRepositoryTriggersExceededException";
const _NCE = "NoChangeException";
const _NLEE = "NameLengthExceededException";
const _NOREE = "NumberOfRulesExceededException";
const _NORTEE = "NumberOfRuleTemplatesExceededException";
const _NT = "NextToken";
const _OART = "OriginApprovalRuleTemplate";
const _OASE = "OverrideAlreadySetException";
const _ONAE = "OperationNotAllowedException";
const _OPRAR = "OverridePullRequestApprovalRules";
const _OPRARI = "OverridePullRequestApprovalRulesInput";
const _OSRE = "OverrideStatusRequiredException";
const _OT = "ObjectTypes";
const _PCDNEE = "ParentCommitDoesNotExistException";
const _PCFCC = "PostCommentForComparedCommit";
const _PCFCCI = "PostCommentForComparedCommitInput";
const _PCFCCO = "PostCommentForComparedCommitOutput";
const _PCFPR = "PostCommentForPullRequest";
const _PCFPRI = "PostCommentForPullRequestInput";
const _PCFPRO = "PostCommentForPullRequestOutput";
const _PCIOE = "ParentCommitIdOutdatedException";
const _PCIRE = "ParentCommitIdRequiredException";
const _PCR = "PostCommentReply";
const _PCRI = "PostCommentReplyInput";
const _PCRIu = "PutCommentReactionInput";
const _PCRO = "PostCommentReplyOutput";
const _PCRu = "PutCommentReaction";
const _PDNEE = "PathDoesNotExistException";
const _PF = "PutFile";
const _PFE = "PutFileEntry";
const _PFECE = "PutFileEntryConflictException";
const _PFEu = "PutFileEntries";
const _PFI = "PutFileInput";
const _PFO = "PutFileOutput";
const _PR = "PullRequest";
const _PRACE = "PullRequestAlreadyClosedException";
const _PRARNSE = "PullRequestApprovalRulesNotSatisfiedException";
const _PRCBABAE = "PullRequestCannotBeApprovedByAuthorException";
const _PRCEM = "PullRequestCreatedEventMetadata";
const _PRDNEE = "PullRequestDoesNotExistException";
const _PRE = "PathRequiredException";
const _PREL = "PullRequestEventList";
const _PREu = "PullRequestEvent";
const _PRIRE = "PullRequestIdRequiredException";
const _PRMSCEM = "PullRequestMergedStateChangedEventMetadata";
const _PRSCEM = "PullRequestStatusChangedEventMetadata";
const _PRSRE = "PullRequestStatusRequiredException";
const _PRSRUEM = "PullRequestSourceReferenceUpdatedEventMetadata";
const _PRT = "PullRequestTarget";
const _PRTI = "PutRepositoryTriggersInput";
const _PRTL = "PullRequestTargetList";
const _PRTO = "PutRepositoryTriggersOutput";
const _PRTu = "PutRepositoryTriggers";
const _RARE = "ResourceArnRequiredException";
const _RCE = "ReplaceContentEntry";
const _RCEe = "ReplaceContentEntries";
const _RCRE = "ReplacementContentRequiredException";
const _RD = "RevisionDag";
const _RDNEE = "ReferenceDoesNotExistException";
const _RDNEEe = "RepositoryDoesNotExistException";
const _RFC = "ReactionForComment";
const _RFCL = "ReactionsForCommentList";
const _RIRE = "RevisionIdRequiredException";
const _RLEE = "ReactionLimitExceededException";
const _RLEEe = "RepositoryLimitExceededException";
const _RM = "RepositoryMetadata";
const _RML = "RepositoryMetadataList";
const _RNAWPRE = "RepositoryNotAssociatedWithPullRequestException";
const _RNCE = "RevisionNotCurrentException";
const _RNEE = "RepositoryNameExistsException";
const _RNIP = "RepositoryNameIdPair";
const _RNIPL = "RepositoryNameIdPairList";
const _RNRE = "ReferenceNameRequiredException";
const _RNREe = "RepositoryNameRequiredException";
const _RNREep = "RepositoryNamesRequiredException";
const _RSFE = "RestrictedSourceFileException";
const _RT = "RepositoryTrigger";
const _RTBNLRE = "RepositoryTriggerBranchNameListRequiredException";
const _RTDARE = "RepositoryTriggerDestinationArnRequiredException";
const _RTEF = "RepositoryTriggerExecutionFailure";
const _RTEFL = "RepositoryTriggerExecutionFailureList";
const _RTELRE = "RepositoryTriggerEventsListRequiredException";
const _RTL = "RepositoryTriggersList";
const _RTLRE = "RepositoryTriggersListRequiredException";
const _RTNRE = "RepositoryTriggerNameRequiredException";
const _RTNSE = "ReferenceTypeNotSupportedException";
const _RTRE = "ReplacementTypeRequiredException";
const _RVF = "ReactionValueFormats";
const _RVRE = "ReactionValueRequiredException";
const _SADASE = "SourceAndDestinationAreSameException";
const _SFCE = "SameFileContentException";
const _SFME = "SetFileModeEntry";
const _SFMEe = "SetFileModeEntries";
const _SFOCRE = "SourceFileOrContentRequiredException";
const _SFS = "SourceFileSpecifier";
const _SL = "SymbolicLink";
const _SLL = "SymbolicLinkList";
const _SM = "SubModule";
const _SML = "SubModuleList";
const _SPRE = "SamePathRequestException";
const _T = "Target";
const _TDEE = "TipsDivergenceExceededException";
const _TKLRE = "TagKeysListRequiredException";
const _TL = "TargetList";
const _TMRE = "TagsMapRequiredException";
const _TMTE = "TooManyTagsException";
const _TOSRIDE = "TipOfSourceReferenceIsDifferentException";
const _TPE = "TagPolicyException";
const _TR = "TagResource";
const _TRE = "TargetRequiredException";
const _TREa = "TargetsRequiredException";
const _TREi = "TitleRequiredException";
const _TRI = "TagResourceInput";
const _TRT = "TestRepositoryTriggers";
const _TRTI = "TestRepositoryTriggersInput";
const _TRTO = "TestRepositoryTriggersOutput";
const _UARTC = "UpdateApprovalRuleTemplateContent";
const _UARTCI = "UpdateApprovalRuleTemplateContentInput";
const _UARTCO = "UpdateApprovalRuleTemplateContentOutput";
const _UARTD = "UpdateApprovalRuleTemplateDescription";
const _UARTDI = "UpdateApprovalRuleTemplateDescriptionInput";
const _UARTDO = "UpdateApprovalRuleTemplateDescriptionOutput";
const _UARTN = "UpdateApprovalRuleTemplateName";
const _UARTNI = "UpdateApprovalRuleTemplateNameInput";
const _UARTNO = "UpdateApprovalRuleTemplateNameOutput";
const _UC = "UpdateComment";
const _UCI = "UpdateCommentInput";
const _UCO = "UpdateCommentOutput";
const _UDB = "UpdateDefaultBranch";
const _UDBI = "UpdateDefaultBranchInput";
const _UI = "UserInfo";
const _UPRARC = "UpdatePullRequestApprovalRuleContent";
const _UPRARCI = "UpdatePullRequestApprovalRuleContentInput";
const _UPRARCO = "UpdatePullRequestApprovalRuleContentOutput";
const _UPRAS = "UpdatePullRequestApprovalState";
const _UPRASI = "UpdatePullRequestApprovalStateInput";
const _UPRD = "UpdatePullRequestDescription";
const _UPRDI = "UpdatePullRequestDescriptionInput";
const _UPRDO = "UpdatePullRequestDescriptionOutput";
const _UPRS = "UpdatePullRequestStatus";
const _UPRSI = "UpdatePullRequestStatusInput";
const _UPRSO = "UpdatePullRequestStatusOutput";
const _UPRT = "UpdatePullRequestTitle";
const _UPRTI = "UpdatePullRequestTitleInput";
const _UPRTO = "UpdatePullRequestTitleOutput";
const _UR = "UntagResource";
const _URD = "UpdateRepositoryDescription";
const _URDI = "UpdateRepositoryDescriptionInput";
const _UREK = "UpdateRepositoryEncryptionKey";
const _UREKI = "UpdateRepositoryEncryptionKeyInput";
const _UREKO = "UpdateRepositoryEncryptionKeyOutput";
const _URI = "UntagResourceInput";
const _URN = "UpdateRepositoryName";
const _URNI = "UpdateRepositoryNameInput";
const _a = "author";
const _aA = "authorArn";
const _aAc = "actorArn";
const _aB = "afterBlob";
const _aBI = "afterBlobId";
const _aCI = "afterCommitId";
const _aCS = "afterCommitSpecifier";
const _aD = "additionalData";
const _aI = "accountId";
const _aN = "authorName";
const _aP = "absolutePath";
const _aPf = "afterPath";
const _aR = "approvalRule";
const _aRC = "approvalRuleContent";
const _aREM = "approvalRuleEventMetadata";
const _aRI = "approvalRuleId";
const _aRN = "approvalRuleName";
const _aRNS = "approvalRulesNotSatisfied";
const _aRNs = "associatedRepositoryNames";
const _aROEM = "approvalRuleOverriddenEventMetadata";
const _aRS = "approvalRulesSatisfied";
const _aRT = "approvalRuleTemplate";
const _aRTC = "approvalRuleTemplateContent";
const _aRTD = "approvalRuleTemplateDescription";
const _aRTI = "approvalRuleTemplateId";
const _aRTN = "approvalRuleTemplateName";
const _aRTNp = "approvalRuleTemplateNames";
const _aRp = "approvalRules";
const _aS = "approvalState";
const _aSCEM = "approvalStateChangedEventMetadata";
const _aSp = "approvalStatus";
const _ap = "approved";
const _app = "approvals";
const _b = "base";
const _bB = "beforeBlob";
const _bBI = "beforeBlobId";
const _bCI = "baseCommitId";
const _bCIe = "beforeCommitId";
const _bCS = "beforeCommitSpecifier";
const _bI = "blobId";
const _bN = "branchName";
const _bP = "beforePath";
const _br = "branch";
const _bra = "branches";
const _c = "client";
const _cC = "contentConflict";
const _cD = "creationDate";
const _cDL = "conflictDetailLevel";
const _cDu = "customData";
const _cFCCD = "commentsForComparedCommitData";
const _cFPRD = "commentsForPullRequestData";
const _cI = "commitId";
const _cIo = "commitIds";
const _cIom = "commentId";
const _cIon = "configurationId";
const _cM = "conflictMetadata";
const _cML = "conflictMetadataList";
const _cMo = "commitMessage";
const _cR = "callerReactions";
const _cRS = "conflictResolutionStrategy";
const _cRT = "clientRequestToken";
const _cRo = "conflictResolution";
const _cS = "commitSpecifier";
const _cT = "changeType";
const _cUH = "cloneUrlHttp";
const _cUS = "cloneUrlSsh";
const _co = "conflicts";
const _com = "commits";
const _comm = "comments";
const _comme = "comment";
const _commi = "committer";
const _commit = "commit";
const _con = "content";
const _d = "deleted";
const _dA = "destinationArn";
const _dB = "deletedBranch";
const _dBN = "defaultBranchName";
const _dBe = "defaultBranch";
const _dC = "destinationCommit";
const _dCI = "destinationCommitId";
const _dCS = "destinationCommitSpecifier";
const _dF = "deleteFiles";
const _dR = "destinationReference";
const _dRN = "disassociatedRepositoryNames";
const _da = "date";
const _de = "description";
const _des = "destination";
const _di = "differences";
const _e = "error";
const _eC = "errorCode";
const _eD = "eventDate";
const _eL = "endLine";
const _eM = "errorMessage";
const _eN = "exceptionName";
const _eRCS = "existingRuleContentSha256";
const _em = "email";
const _emo = "emoji";
const _er = "errors";
const _ev = "evaluation";
const _eve = "events";
const _f = "files";
const _fA = "filesAdded";
const _fC = "fileContent";
const _fD = "filesDeleted";
const _fE = "failedExecutions";
const _fM = "fileModes";
const _fMC = "fileModeConflict";
const _fMa = "failureMessage";
const _fMi = "fileMode";
const _fP = "filePath";
const _fPi = "filePaths";
const _fPil = "filePosition";
const _fPo = "folderPath";
const _fS = "fileSizes";
const _fSi = "fileSize";
const _fU = "filesUpdated";
const _hC = "hunkContent";
const _iBF = "isBinaryFile";
const _iC = "isConflict";
const _iM = "isMerged";
const _iMs = "isMove";
const _iRT = "inReplyTo";
const _kEF = "keepEmptyFolders";
const _kKI = "kmsKeyId";
const _l = "location";
const _lAD = "lastActivityDate";
const _lMD = "lastModifiedDate";
const _lMU = "lastModifiedUser";
const _m = "message";
const _mB = "mergedBy";
const _mBe = "mergeBase";
const _mCF = "maxConflictFiles";
const _mCI = "mergedCommitId";
const _mCIe = "mergeCommitId";
const _mH = "mergeHunks";
const _mM = "mergeMetadata";
const _mMH = "maxMergeHunks";
const _mO = "mergeOption";
const _mOe = "mergeOperations";
const _mOer = "mergeOptions";
const _mR = "maxResults";
const _me = "mergeable";
const _mo = "mode";
const _n = "name";
const _nARTN = "newApprovalRuleTemplateName";
const _nN = "newName";
const _nOC = "numberOfConflicts";
const _nRC = "newRuleContent";
const _nT = "nextToken";
const _o = "overridden";
const _oART = "originApprovalRuleTemplate";
const _oARTN = "oldApprovalRuleTemplateName";
const _oKKI = "originalKmsKeyId";
const _oN = "oldName";
const _oS = "overrideStatus";
const _oT = "objectTypes";
const _oTC = "objectTypeConflict";
const _or = "order";
const _ov = "overrider";
const _p = "path";
const _pCI = "parentCommitId";
const _pF = "putFiles";
const _pR = "pullRequest";
const _pRCEM = "pullRequestCreatedEventMetadata";
const _pRE = "pullRequestEvents";
const _pRET = "pullRequestEventType";
const _pRI = "pullRequestId";
const _pRIu = "pullRequestIds";
const _pRMSCEM = "pullRequestMergedStateChangedEventMetadata";
const _pRS = "pullRequestStatus";
const _pRSCEM = "pullRequestStatusChangedEventMetadata";
const _pRSRUEM = "pullRequestSourceReferenceUpdatedEventMetadata";
const _pRT = "pullRequestTargets";
const _pa = "parents";
const _r = "repositories";
const _rA = "resourceArn";
const _rC = "reactionCounts";
const _rCS = "ruleContentSha256";
const _rCe = "replaceContents";
const _rCev = "revisionChildren";
const _rD = "repositoryDescription";
const _rDe = "revisionDag";
const _rFC = "reactionsForComment";
const _rFDUC = "reactionsFromDeletedUsersCount";
const _rFV = "relativeFileVersion";
const _rI = "revisionId";
const _rIe = "repositoryId";
const _rM = "repositoryMetadata";
const _rN = "repositoryName";
const _rNF = "repositoriesNotFound";
const _rNe = "repositoryNames";
const _rP = "relativePath";
const _rT = "replacementType";
const _rU = "reactionUsers";
const _rUA = "reactionUserArn";
const _rV = "reactionValue";
const _re = "reaction";
const _s = "server";
const _sB = "sortBy";
const _sC = "sourceCommit";
const _sCI = "sourceCommitId";
const _sCS = "sourceCommitSpecifier";
const _sCh = "shortCode";
const _sE = "successfulExecutions";
const _sF = "subFolders";
const _sFM = "setFileModes";
const _sFo = "sourceFile";
const _sL = "symbolicLinks";
const _sLt = "startLine";
const _sM = "subModules";
const _sR = "sourceReference";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.codecommit";
const _so = "source";
const _t = "title";
const _tB = "targetBranch";
const _tI = "treeId";
const _tK = "tagKeys";
const _ta = "targets";
const _tag = "tags";
const _tr = "triggers";
const _tri = "trigger";
const _u = "unicode";
const _uA = "userArn";
const n0 = "com.amazonaws.codecommit";
import { TypeRegistry } from "@smithy/core/schema";
import { CodeCommitServiceException as __CodeCommitServiceException } from "../models/CodeCommitServiceException";
import { ActorDoesNotExistException as __ActorDoesNotExistException, ApprovalRuleContentRequiredException as __ApprovalRuleContentRequiredException, ApprovalRuleDoesNotExistException as __ApprovalRuleDoesNotExistException, ApprovalRuleNameAlreadyExistsException as __ApprovalRuleNameAlreadyExistsException, ApprovalRuleNameRequiredException as __ApprovalRuleNameRequiredException, ApprovalRuleTemplateContentRequiredException as __ApprovalRuleTemplateContentRequiredException, ApprovalRuleTemplateDoesNotExistException as __ApprovalRuleTemplateDoesNotExistException, ApprovalRuleTemplateInUseException as __ApprovalRuleTemplateInUseException, ApprovalRuleTemplateNameAlreadyExistsException as __ApprovalRuleTemplateNameAlreadyExistsException, ApprovalRuleTemplateNameRequiredException as __ApprovalRuleTemplateNameRequiredException, ApprovalStateRequiredException as __ApprovalStateRequiredException, AuthorDoesNotExistException as __AuthorDoesNotExistException, BeforeCommitIdAndAfterCommitIdAreSameException as __BeforeCommitIdAndAfterCommitIdAreSameException, BlobIdDoesNotExistException as __BlobIdDoesNotExistException, BlobIdRequiredException as __BlobIdRequiredException, BranchDoesNotExistException as __BranchDoesNotExistException, BranchNameExistsException as __BranchNameExistsException, BranchNameIsTagNameException as __BranchNameIsTagNameException, BranchNameRequiredException as __BranchNameRequiredException, CannotDeleteApprovalRuleFromTemplateException as __CannotDeleteApprovalRuleFromTemplateException, CannotModifyApprovalRuleFromTemplateException as __CannotModifyApprovalRuleFromTemplateException, ClientRequestTokenRequiredException as __ClientRequestTokenRequiredException, CommentContentRequiredException as __CommentContentRequiredException, CommentContentSizeLimitExceededException as __CommentContentSizeLimitExceededException, CommentDeletedException as __CommentDeletedException, CommentDoesNotExistException as __CommentDoesNotExistException, CommentIdRequiredException as __CommentIdRequiredException, CommentNotCreatedByCallerException as __CommentNotCreatedByCallerException, CommitDoesNotExistException as __CommitDoesNotExistException, CommitIdDoesNotExistException as __CommitIdDoesNotExistException, CommitIdRequiredException as __CommitIdRequiredException, CommitIdsLimitExceededException as __CommitIdsLimitExceededException, CommitIdsListRequiredException as __CommitIdsListRequiredException, CommitMessageLengthExceededException as __CommitMessageLengthExceededException, CommitRequiredException as __CommitRequiredException, ConcurrentReferenceUpdateException as __ConcurrentReferenceUpdateException, DefaultBranchCannotBeDeletedException as __DefaultBranchCannotBeDeletedException, DirectoryNameConflictsWithFileNameException as __DirectoryNameConflictsWithFileNameException, EncryptionIntegrityChecksFailedException as __EncryptionIntegrityChecksFailedException, EncryptionKeyAccessDeniedException as __EncryptionKeyAccessDeniedException, EncryptionKeyDisabledException as __EncryptionKeyDisabledException, EncryptionKeyInvalidIdException as __EncryptionKeyInvalidIdException, EncryptionKeyInvalidUsageException as __EncryptionKeyInvalidUsageException, EncryptionKeyNotFoundException as __EncryptionKeyNotFoundException, EncryptionKeyRequiredException as __EncryptionKeyRequiredException, EncryptionKeyUnavailableException as __EncryptionKeyUnavailableException, FileContentAndSourceFileSpecifiedException as __FileContentAndSourceFileSpecifiedException, FileContentRequiredException as __FileContentRequiredException, FileContentSizeLimitExceededException as __FileContentSizeLimitExceededException, FileDoesNotExistException as __FileDoesNotExistException, FileEntryRequiredException as __FileEntryRequiredException, FileModeRequiredException as __FileModeRequiredException, FileNameConflictsWithDirectoryNameException as __FileNameConflictsWithDirectoryNameException, FilePathConflictsWithSubmodulePathException as __FilePathConflictsWithSubmodulePathException, FileTooLargeException as __FileTooLargeException, FolderContentSizeLimitExceededException as __FolderContentSizeLimitExceededException, FolderDoesNotExistException as __FolderDoesNotExistException, IdempotencyParameterMismatchException as __IdempotencyParameterMismatchException, InvalidActorArnException as __InvalidActorArnException, InvalidApprovalRuleContentException as __InvalidApprovalRuleContentException, InvalidApprovalRuleNameException as __InvalidApprovalRuleNameException, InvalidApprovalRuleTemplateContentException as __InvalidApprovalRuleTemplateContentException, InvalidApprovalRuleTemplateDescriptionException as __InvalidApprovalRuleTemplateDescriptionException, InvalidApprovalRuleTemplateNameException as __InvalidApprovalRuleTemplateNameException, InvalidApprovalStateException as __InvalidApprovalStateException, InvalidAuthorArnException as __InvalidAuthorArnException, InvalidBlobIdException as __InvalidBlobIdException, InvalidBranchNameException as __InvalidBranchNameException, InvalidClientRequestTokenException as __InvalidClientRequestTokenException, InvalidCommentIdException as __InvalidCommentIdException, InvalidCommitException as __InvalidCommitException, InvalidCommitIdException as __InvalidCommitIdException, InvalidConflictDetailLevelException as __InvalidConflictDetailLevelException, InvalidConflictResolutionException as __InvalidConflictResolutionException, InvalidConflictResolutionStrategyException as __InvalidConflictResolutionStrategyException, InvalidContinuationTokenException as __InvalidContinuationTokenException, InvalidDeletionParameterException as __InvalidDeletionParameterException, InvalidDescriptionException as __InvalidDescriptionException, InvalidDestinationCommitSpecifierException as __InvalidDestinationCommitSpecifierException, InvalidEmailException as __InvalidEmailException, InvalidFileLocationException as __InvalidFileLocationException, InvalidFileModeException as __InvalidFileModeException, InvalidFilePositionException as __InvalidFilePositionException, InvalidMaxConflictFilesException as __InvalidMaxConflictFilesException, InvalidMaxMergeHunksException as __InvalidMaxMergeHunksException, InvalidMaxResultsException as __InvalidMaxResultsException, InvalidMergeOptionException as __InvalidMergeOptionException, InvalidOrderException as __InvalidOrderException, InvalidOverrideStatusException as __InvalidOverrideStatusException, InvalidParentCommitIdException as __InvalidParentCommitIdException, InvalidPathException as __InvalidPathException, InvalidPullRequestEventTypeException as __InvalidPullRequestEventTypeException, InvalidPullRequestIdException as __InvalidPullRequestIdException, InvalidPullRequestStatusException as __InvalidPullRequestStatusException, InvalidPullRequestStatusUpdateException as __InvalidPullRequestStatusUpdateException, InvalidReactionUserArnException as __InvalidReactionUserArnException, InvalidReactionValueException as __InvalidReactionValueException, InvalidReferenceNameException as __InvalidReferenceNameException, InvalidRelativeFileVersionEnumException as __InvalidRelativeFileVersionEnumException, InvalidReplacementContentException as __InvalidReplacementContentException, InvalidReplacementTypeException as __InvalidReplacementTypeException, InvalidRepositoryDescriptionException as __InvalidRepositoryDescriptionException, InvalidRepositoryNameException as __InvalidRepositoryNameException, InvalidRepositoryTriggerBranchNameException as __InvalidRepositoryTriggerBranchNameException, InvalidRepositoryTriggerCustomDataException as __InvalidRepositoryTriggerCustomDataException, InvalidRepositoryTriggerDestinationArnException as __InvalidRepositoryTriggerDestinationArnException, InvalidRepositoryTriggerEventsException as __InvalidRepositoryTriggerEventsException, InvalidRepositoryTriggerNameException as __InvalidRepositoryTriggerNameException, InvalidRepositoryTriggerRegionException as __InvalidRepositoryTriggerRegionException, InvalidResourceArnException as __InvalidResourceArnException, InvalidRevisionIdException as __InvalidRevisionIdException, InvalidRuleContentSha256Exception as __InvalidRuleContentSha256Exception, InvalidSortByException as __InvalidSortByException, InvalidSourceCommitSpecifierException as __InvalidSourceCommitSpecifierException, InvalidSystemTagUsageException as __InvalidSystemTagUsageException, InvalidTagKeysListException as __InvalidTagKeysListException, InvalidTagsMapException as __InvalidTagsMapException, InvalidTargetBranchException as __InvalidTargetBranchException, InvalidTargetException as __InvalidTargetException, InvalidTargetsException as __InvalidTargetsException, InvalidTitleException as __InvalidTitleException, ManualMergeRequiredException as __ManualMergeRequiredException, MaximumBranchesExceededException as __MaximumBranchesExceededException, MaximumConflictResolutionEntriesExceededException as __MaximumConflictResolutionEntriesExceededException, MaximumFileContentToLoadExceededException as __MaximumFileContentToLoadExceededException, MaximumFileEntriesExceededException as __MaximumFileEntriesExceededException, MaximumItemsToCompareExceededException as __MaximumItemsToCompareExceededException, MaximumNumberOfApprovalsExceededException as __MaximumNumberOfApprovalsExceededException, MaximumOpenPullRequestsExceededException as __MaximumOpenPullRequestsExceededException, MaximumRepositoryNamesExceededException as __MaximumRepositoryNamesExceededException, MaximumRepositoryTriggersExceededException as __MaximumRepositoryTriggersExceededException, MaximumRuleTemplatesAssociatedWithRepositoryException as __MaximumRuleTemplatesAssociatedWithRepositoryException, MergeOptionRequiredException as __MergeOptionRequiredException, MultipleConflictResolutionEntriesException as __MultipleConflictResolutionEntriesException, MultipleRepositoriesInPullRequestException as __MultipleRepositoriesInPullRequestException, NameLengthExceededException as __NameLengthExceededException, NoChangeException as __NoChangeException, NumberOfRulesExceededException as __NumberOfRulesExceededException, NumberOfRuleTemplatesExceededException as __NumberOfRuleTemplatesExceededException, OperationNotAllowedException as __OperationNotAllowedException, OverrideAlreadySetException as __OverrideAlreadySetException, OverrideStatusRequiredException as __OverrideStatusRequiredException, ParentCommitDoesNotExistException as __ParentCommitDoesNotExistException, ParentCommitIdOutdatedException as __ParentCommitIdOutdatedException, ParentCommitIdRequiredException as __ParentCommitIdRequiredException, PathDoesNotExistException as __PathDoesNotExistException, PathRequiredException as __PathRequiredException, PullRequestAlreadyClosedException as __PullRequestAlreadyClosedException, PullRequestApprovalRulesNotSatisfiedException as __PullRequestApprovalRulesNotSatisfiedException, PullRequestCannotBeApprovedByAuthorException as __PullRequestCannotBeApprovedByAuthorException, PullRequestDoesNotExistException as __PullRequestDoesNotExistException, PullRequestIdRequiredException as __PullRequestIdRequiredException, PullRequestStatusRequiredException as __PullRequestStatusRequiredException, PutFileEntryConflictException as __PutFileEntryConflictException, ReactionLimitExceededException as __ReactionLimitExceededException, ReactionValueRequiredException as __ReactionValueRequiredException, ReferenceDoesNotExistException as __ReferenceDoesNotExistException, ReferenceNameRequiredException as __ReferenceNameRequiredException, ReferenceTypeNotSupportedException as __ReferenceTypeNotSupportedException, ReplacementContentRequiredException as __ReplacementContentRequiredException, ReplacementTypeRequiredException as __ReplacementTypeRequiredException, RepositoryDoesNotExistException as __RepositoryDoesNotExistException, RepositoryLimitExceededException as __RepositoryLimitExceededException, RepositoryNameExistsException as __RepositoryNameExistsException, RepositoryNameRequiredException as __RepositoryNameRequiredException, RepositoryNamesRequiredException as __RepositoryNamesRequiredException, RepositoryNotAssociatedWithPullRequestException as __RepositoryNotAssociatedWithPullRequestException, RepositoryTriggerBranchNameListRequiredException as __RepositoryTriggerBranchNameListRequiredException, RepositoryTriggerDestinationArnRequiredException as __RepositoryTriggerDestinationArnRequiredException, RepositoryTriggerEventsListRequiredException as __RepositoryTriggerEventsListRequiredException, RepositoryTriggerNameRequiredException as __RepositoryTriggerNameRequiredException, RepositoryTriggersListRequiredException as __RepositoryTriggersListRequiredException, ResourceArnRequiredException as __ResourceArnRequiredException, RestrictedSourceFileException as __RestrictedSourceFileException, RevisionIdRequiredException as __RevisionIdRequiredException, RevisionNotCurrentException as __RevisionNotCurrentException, SameFileContentException as __SameFileContentException, SamePathRequestException as __SamePathRequestException, SourceAndDestinationAreSameException as __SourceAndDestinationAreSameException, SourceFileOrContentRequiredException as __SourceFileOrContentRequiredException, TagKeysListRequiredException as __TagKeysListRequiredException, TagPolicyException as __TagPolicyException, TagsMapRequiredException as __TagsMapRequiredException, TargetRequiredException as __TargetRequiredException, TargetsRequiredException as __TargetsRequiredException, TipOfSourceReferenceIsDifferentException as __TipOfSourceReferenceIsDifferentException, TipsDivergenceExceededException as __TipsDivergenceExceededException, TitleRequiredException as __TitleRequiredException, TooManyTagsException as __TooManyTagsException, } from "../models/index";
export var ActorDoesNotExistException = [
    -3,
    n0,
    _ADNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ActorDoesNotExistException, __ActorDoesNotExistException);
export var Approval = [3, n0, _A, 0, [_uA, _aS], [0, 0]];
export var ApprovalRule = [
    3,
    n0,
    _AR,
    0,
    [_aRI, _aRN, _aRC, _rCS, _lMD, _cD, _lMU, _oART],
    [0, 0, 0, 0, 4, 4, 0, () => OriginApprovalRuleTemplate],
];
export var ApprovalRuleContentRequiredException = [
    -3,
    n0,
    _ARCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleContentRequiredException, __ApprovalRuleContentRequiredException);
export var ApprovalRuleDoesNotExistException = [
    -3,
    n0,
    _ARDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleDoesNotExistException, __ApprovalRuleDoesNotExistException);
export var ApprovalRuleEventMetadata = [3, n0, _AREM, 0, [_aRN, _aRI, _aRC], [0, 0, 0]];
export var ApprovalRuleNameAlreadyExistsException = [
    -3,
    n0,
    _ARNAEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleNameAlreadyExistsException, __ApprovalRuleNameAlreadyExistsException);
export var ApprovalRuleNameRequiredException = [
    -3,
    n0,
    _ARNRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleNameRequiredException, __ApprovalRuleNameRequiredException);
export var ApprovalRuleOverriddenEventMetadata = [3, n0, _AROEM, 0, [_rI, _oS], [0, 0]];
export var ApprovalRuleTemplate = [
    3,
    n0,
    _ART,
    0,
    [_aRTI, _aRTN, _aRTD, _aRTC, _rCS, _lMD, _cD, _lMU],
    [0, 0, 0, 0, 0, 4, 4, 0],
];
export var ApprovalRuleTemplateContentRequiredException = [
    -3,
    n0,
    _ARTCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleTemplateContentRequiredException, __ApprovalRuleTemplateContentRequiredException);
export var ApprovalRuleTemplateDoesNotExistException = [
    -3,
    n0,
    _ARTDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleTemplateDoesNotExistException, __ApprovalRuleTemplateDoesNotExistException);
export var ApprovalRuleTemplateInUseException = [
    -3,
    n0,
    _ARTIUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleTemplateInUseException, __ApprovalRuleTemplateInUseException);
export var ApprovalRuleTemplateNameAlreadyExistsException = [
    -3,
    n0,
    _ARTNAEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleTemplateNameAlreadyExistsException, __ApprovalRuleTemplateNameAlreadyExistsException);
export var ApprovalRuleTemplateNameRequiredException = [
    -3,
    n0,
    _ARTNRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalRuleTemplateNameRequiredException, __ApprovalRuleTemplateNameRequiredException);
export var ApprovalStateChangedEventMetadata = [3, n0, _ASCEM, 0, [_rI, _aSp], [0, 0]];
export var ApprovalStateRequiredException = [
    -3,
    n0,
    _ASRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ApprovalStateRequiredException, __ApprovalStateRequiredException);
export var AssociateApprovalRuleTemplateWithRepositoryInput = [
    3,
    n0,
    _AARTWRI,
    0,
    [_aRTN, _rN],
    [0, 0],
];
export var AuthorDoesNotExistException = [
    -3,
    n0,
    _ADNEEu,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(AuthorDoesNotExistException, __AuthorDoesNotExistException);
export var BatchAssociateApprovalRuleTemplateWithRepositoriesError = [
    3,
    n0,
    _BAARTWRE,
    0,
    [_rN, _eC, _eM],
    [0, 0, 0],
];
export var BatchAssociateApprovalRuleTemplateWithRepositoriesInput = [
    3,
    n0,
    _BAARTWRI,
    0,
    [_aRTN, _rNe],
    [0, 64 | 0],
];
export var BatchAssociateApprovalRuleTemplateWithRepositoriesOutput = [
    3,
    n0,
    _BAARTWRO,
    0,
    [_aRNs, _er],
    [64 | 0, () => BatchAssociateApprovalRuleTemplateWithRepositoriesErrorsList],
];
export var BatchDescribeMergeConflictsError = [3, n0, _BDMCE, 0, [_fP, _eN, _m], [0, 0, 0]];
export var BatchDescribeMergeConflictsInput = [
    3,
    n0,
    _BDMCI,
    0,
    [_rN, _dCS, _sCS, _mO, _mMH, _mCF, _fPi, _cDL, _cRS, _nT],
    [0, 0, 0, 0, 1, 1, 64 | 0, 0, 0, 0],
];
export var BatchDescribeMergeConflictsOutput = [
    3,
    n0,
    _BDMCO,
    0,
    [_co, _nT, _er, _dCI, _sCI, _bCI],
    [() => Conflicts, 0, () => BatchDescribeMergeConflictsErrors, 0, 0, 0],
];
export var BatchDisassociateApprovalRuleTemplateFromRepositoriesError = [
    3,
    n0,
    _BDARTFRE,
    0,
    [_rN, _eC, _eM],
    [0, 0, 0],
];
export var BatchDisassociateApprovalRuleTemplateFromRepositoriesInput = [
    3,
    n0,
    _BDARTFRI,
    0,
    [_aRTN, _rNe],
    [0, 64 | 0],
];
export var BatchDisassociateApprovalRuleTemplateFromRepositoriesOutput = [
    3,
    n0,
    _BDARTFRO,
    0,
    [_dRN, _er],
    [64 | 0, () => BatchDisassociateApprovalRuleTemplateFromRepositoriesErrorsList],
];
export var BatchGetCommitsError = [3, n0, _BGCE, 0, [_cI, _eC, _eM], [0, 0, 0]];
export var BatchGetCommitsInput = [3, n0, _BGCI, 0, [_cIo, _rN], [64 | 0, 0]];
export var BatchGetCommitsOutput = [
    3,
    n0,
    _BGCO,
    0,
    [_com, _er],
    [() => CommitObjectsList, () => BatchGetCommitsErrorsList],
];
export var BatchGetRepositoriesError = [3, n0, _BGRE, 0, [_rIe, _rN, _eC, _eM], [0, 0, 0, 0]];
export var BatchGetRepositoriesInput = [3, n0, _BGRI, 0, [_rNe], [64 | 0]];
export var BatchGetRepositoriesOutput = [
    3,
    n0,
    _BGRO,
    0,
    [_r, _rNF, _er],
    [() => RepositoryMetadataList, 64 | 0, () => BatchGetRepositoriesErrorsList],
];
export var BeforeCommitIdAndAfterCommitIdAreSameException = [
    -3,
    n0,
    _BCIAACIASE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BeforeCommitIdAndAfterCommitIdAreSameException, __BeforeCommitIdAndAfterCommitIdAreSameException);
export var BlobIdDoesNotExistException = [
    -3,
    n0,
    _BIDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BlobIdDoesNotExistException, __BlobIdDoesNotExistException);
export var BlobIdRequiredException = [
    -3,
    n0,
    _BIRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BlobIdRequiredException, __BlobIdRequiredException);
export var BlobMetadata = [3, n0, _BM, 0, [_bI, _p, _mo], [0, 0, 0]];
export var BranchDoesNotExistException = [
    -3,
    n0,
    _BDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BranchDoesNotExistException, __BranchDoesNotExistException);
export var BranchInfo = [3, n0, _BI, 0, [_bN, _cI], [0, 0]];
export var BranchNameExistsException = [
    -3,
    n0,
    _BNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BranchNameExistsException, __BranchNameExistsException);
export var BranchNameIsTagNameException = [
    -3,
    n0,
    _BNITNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BranchNameIsTagNameException, __BranchNameIsTagNameException);
export var BranchNameRequiredException = [
    -3,
    n0,
    _BNRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BranchNameRequiredException, __BranchNameRequiredException);
export var CannotDeleteApprovalRuleFromTemplateException = [
    -3,
    n0,
    _CDARFTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CannotDeleteApprovalRuleFromTemplateException, __CannotDeleteApprovalRuleFromTemplateException);
export var CannotModifyApprovalRuleFromTemplateException = [
    -3,
    n0,
    _CMARFTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CannotModifyApprovalRuleFromTemplateException, __CannotModifyApprovalRuleFromTemplateException);
export var ClientRequestTokenRequiredException = [
    -3,
    n0,
    _CRTRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ClientRequestTokenRequiredException, __ClientRequestTokenRequiredException);
export var Comment = [
    3,
    n0,
    _C,
    0,
    [_cIom, _con, _iRT, _cD, _lMD, _aA, _d, _cRT, _cR, _rC],
    [0, 0, 0, 4, 4, 0, 2, 0, 64 | 0, 128 | 1],
];
export var CommentContentRequiredException = [
    -3,
    n0,
    _CCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentContentRequiredException, __CommentContentRequiredException);
export var CommentContentSizeLimitExceededException = [
    -3,
    n0,
    _CCSLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentContentSizeLimitExceededException, __CommentContentSizeLimitExceededException);
export var CommentDeletedException = [
    -3,
    n0,
    _CDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentDeletedException, __CommentDeletedException);
export var CommentDoesNotExistException = [
    -3,
    n0,
    _CDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentDoesNotExistException, __CommentDoesNotExistException);
export var CommentIdRequiredException = [
    -3,
    n0,
    _CIRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentIdRequiredException, __CommentIdRequiredException);
export var CommentNotCreatedByCallerException = [
    -3,
    n0,
    _CNCBCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommentNotCreatedByCallerException, __CommentNotCreatedByCallerException);
export var CommentsForComparedCommit = [
    3,
    n0,
    _CFCC,
    0,
    [_rN, _bCIe, _aCI, _bBI, _aBI, _l, _comm],
    [0, 0, 0, 0, 0, () => Location, () => Comments],
];
export var CommentsForPullRequest = [
    3,
    n0,
    _CFPR,
    0,
    [_pRI, _rN, _bCIe, _aCI, _bBI, _aBI, _l, _comm],
    [0, 0, 0, 0, 0, 0, () => Location, () => Comments],
];
export var Commit = [
    3,
    n0,
    _Co,
    0,
    [_cI, _tI, _pa, _m, _a, _commi, _aD],
    [0, 0, 64 | 0, 0, () => UserInfo, () => UserInfo, 0],
];
export var CommitDoesNotExistException = [
    -3,
    n0,
    _CDNEEo,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitDoesNotExistException, __CommitDoesNotExistException);
export var CommitIdDoesNotExistException = [
    -3,
    n0,
    _CIDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitIdDoesNotExistException, __CommitIdDoesNotExistException);
export var CommitIdRequiredException = [
    -3,
    n0,
    _CIREo,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitIdRequiredException, __CommitIdRequiredException);
export var CommitIdsLimitExceededException = [
    -3,
    n0,
    _CILEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitIdsLimitExceededException, __CommitIdsLimitExceededException);
export var CommitIdsListRequiredException = [
    -3,
    n0,
    _CILRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitIdsListRequiredException, __CommitIdsListRequiredException);
export var CommitMessageLengthExceededException = [
    -3,
    n0,
    _CMLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitMessageLengthExceededException, __CommitMessageLengthExceededException);
export var CommitRequiredException = [
    -3,
    n0,
    _CRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(CommitRequiredException, __CommitRequiredException);
export var ConcurrentReferenceUpdateException = [
    -3,
    n0,
    _CRUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ConcurrentReferenceUpdateException, __ConcurrentReferenceUpdateException);
export var Conflict = [3, n0, _Con, 0, [_cM, _mH], [() => ConflictMetadata, () => MergeHunks]];
export var ConflictMetadata = [
    3,
    n0,
    _CM,
    0,
    [_fP, _fS, _fM, _oT, _nOC, _iBF, _cC, _fMC, _oTC, _mOe],
    [0, () => FileSizes, () => FileModes, () => ObjectTypes, 1, () => IsBinaryFile, 2, 2, 2, () => MergeOperations],
];
export var ConflictResolution = [
    3,
    n0,
    _CR,
    0,
    [_rCe, _dF, _sFM],
    [() => ReplaceContentEntries, () => DeleteFileEntries, () => SetFileModeEntries],
];
export var CreateApprovalRuleTemplateInput = [
    3,
    n0,
    _CARTI,
    0,
    [_aRTN, _aRTC, _aRTD],
    [0, 0, 0],
];
export var CreateApprovalRuleTemplateOutput = [
    3,
    n0,
    _CARTO,
    0,
    [_aRT],
    [() => ApprovalRuleTemplate],
];
export var CreateBranchInput = [3, n0, _CBI, 0, [_rN, _bN, _cI], [0, 0, 0]];
export var CreateCommitInput = [
    3,
    n0,
    _CCI,
    0,
    [_rN, _bN, _pCI, _aN, _em, _cMo, _kEF, _pF, _dF, _sFM],
    [0, 0, 0, 0, 0, 0, 2, () => PutFileEntries, () => DeleteFileEntries, () => SetFileModeEntries],
];
export var CreateCommitOutput = [
    3,
    n0,
    _CCO,
    0,
    [_cI, _tI, _fA, _fU, _fD],
    [0, 0, () => FilesMetadata, () => FilesMetadata, () => FilesMetadata],
];
export var CreatePullRequestApprovalRuleInput = [
    3,
    n0,
    _CPRARI,
    0,
    [_pRI, _aRN, _aRC],
    [0, 0, 0],
];
export var CreatePullRequestApprovalRuleOutput = [
    3,
    n0,
    _CPRARO,
    0,
    [_aR],
    [() => ApprovalRule],
];
export var CreatePullRequestInput = [
    3,
    n0,
    _CPRI,
    0,
    [_t, _de, _ta, _cRT],
    [0, 0, () => TargetList, [0, 4]],
];
export var CreatePullRequestOutput = [3, n0, _CPRO, 0, [_pR], [() => PullRequest]];
export var CreateRepositoryInput = [3, n0, _CRI, 0, [_rN, _rD, _tag, _kKI], [0, 0, 128 | 0, 0]];
export var CreateRepositoryOutput = [3, n0, _CRO, 0, [_rM], [() => RepositoryMetadata]];
export var CreateUnreferencedMergeCommitInput = [
    3,
    n0,
    _CUMCI,
    0,
    [_rN, _sCS, _dCS, _mO, _cDL, _cRS, _aN, _em, _cMo, _kEF, _cRo],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 2, () => ConflictResolution],
];
export var CreateUnreferencedMergeCommitOutput = [3, n0, _CUMCO, 0, [_cI, _tI], [0, 0]];
export var DefaultBranchCannotBeDeletedException = [
    -3,
    n0,
    _DBCBDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(DefaultBranchCannotBeDeletedException, __DefaultBranchCannotBeDeletedException);
export var DeleteApprovalRuleTemplateInput = [3, n0, _DARTI, 0, [_aRTN], [0]];
export var DeleteApprovalRuleTemplateOutput = [3, n0, _DARTO, 0, [_aRTI], [0]];
export var DeleteBranchInput = [3, n0, _DBI, 0, [_rN, _bN], [0, 0]];
export var DeleteBranchOutput = [3, n0, _DBO, 0, [_dB], [() => BranchInfo]];
export var DeleteCommentContentInput = [3, n0, _DCCI, 0, [_cIom], [0]];
export var DeleteCommentContentOutput = [3, n0, _DCCO, 0, [_comme], [() => Comment]];
export var DeleteFileEntry = [3, n0, _DFE, 0, [_fP], [0]];
export var DeleteFileInput = [
    3,
    n0,
    _DFI,
    0,
    [_rN, _bN, _fP, _pCI, _kEF, _cMo, _n, _em],
    [0, 0, 0, 0, 2, 0, 0, 0],
];
export var DeleteFileOutput = [3, n0, _DFO, 0, [_cI, _bI, _tI, _fP], [0, 0, 0, 0]];
export var DeletePullRequestApprovalRuleInput = [3, n0, _DPRARI, 0, [_pRI, _aRN], [0, 0]];
export var DeletePullRequestApprovalRuleOutput = [3, n0, _DPRARO, 0, [_aRI], [0]];
export var DeleteRepositoryInput = [3, n0, _DRI, 0, [_rN], [0]];
export var DeleteRepositoryOutput = [3, n0, _DRO, 0, [_rIe], [0]];
export var DescribeMergeConflictsInput = [
    3,
    n0,
    _DMCI,
    0,
    [_rN, _dCS, _sCS, _mO, _mMH, _fP, _cDL, _cRS, _nT],
    [0, 0, 0, 0, 1, 0, 0, 0, 0],
];
export var DescribeMergeConflictsOutput = [
    3,
    n0,
    _DMCO,
    0,
    [_cM, _mH, _nT, _dCI, _sCI, _bCI],
    [() => ConflictMetadata, () => MergeHunks, 0, 0, 0, 0],
];
export var DescribePullRequestEventsInput = [
    3,
    n0,
    _DPREI,
    0,
    [_pRI, _pRET, _aAc, _nT, _mR],
    [0, 0, 0, 0, 1],
];
export var DescribePullRequestEventsOutput = [
    3,
    n0,
    _DPREO,
    0,
    [_pRE, _nT],
    [() => PullRequestEventList, 0],
];
export var Difference = [
    3,
    n0,
    _D,
    0,
    [_bB, _aB, _cT],
    [() => BlobMetadata, () => BlobMetadata, 0],
];
export var DirectoryNameConflictsWithFileNameException = [
    -3,
    n0,
    _DNCWFNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(DirectoryNameConflictsWithFileNameException, __DirectoryNameConflictsWithFileNameException);
export var DisassociateApprovalRuleTemplateFromRepositoryInput = [
    3,
    n0,
    _DARTFRI,
    0,
    [_aRTN, _rN],
    [0, 0],
];
export var EncryptionIntegrityChecksFailedException = [
    -3,
    n0,
    _EICFE,
    {
        [_e]: _s,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionIntegrityChecksFailedException, __EncryptionIntegrityChecksFailedException);
export var EncryptionKeyAccessDeniedException = [
    -3,
    n0,
    _EKADE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyAccessDeniedException, __EncryptionKeyAccessDeniedException);
export var EncryptionKeyDisabledException = [
    -3,
    n0,
    _EKDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyDisabledException, __EncryptionKeyDisabledException);
export var EncryptionKeyInvalidIdException = [
    -3,
    n0,
    _EKIIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyInvalidIdException, __EncryptionKeyInvalidIdException);
export var EncryptionKeyInvalidUsageException = [
    -3,
    n0,
    _EKIUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyInvalidUsageException, __EncryptionKeyInvalidUsageException);
export var EncryptionKeyNotFoundException = [
    -3,
    n0,
    _EKNFE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyNotFoundException, __EncryptionKeyNotFoundException);
export var EncryptionKeyRequiredException = [
    -3,
    n0,
    _EKRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyRequiredException, __EncryptionKeyRequiredException);
export var EncryptionKeyUnavailableException = [
    -3,
    n0,
    _EKUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(EncryptionKeyUnavailableException, __EncryptionKeyUnavailableException);
export var EvaluatePullRequestApprovalRulesInput = [3, n0, _EPRARI, 0, [_pRI, _rI], [0, 0]];
export var EvaluatePullRequestApprovalRulesOutput = [
    3,
    n0,
    _EPRARO,
    0,
    [_ev],
    [() => Evaluation],
];
export var Evaluation = [3, n0, _E, 0, [_ap, _o, _aRS, _aRNS], [2, 2, 64 | 0, 64 | 0]];
export var File = [3, n0, _F, 0, [_bI, _aP, _rP, _fMi], [0, 0, 0, 0]];
export var FileContentAndSourceFileSpecifiedException = [
    -3,
    n0,
    _FCASFSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileContentAndSourceFileSpecifiedException, __FileContentAndSourceFileSpecifiedException);
export var FileContentRequiredException = [
    -3,
    n0,
    _FCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileContentRequiredException, __FileContentRequiredException);
export var FileContentSizeLimitExceededException = [
    -3,
    n0,
    _FCSLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileContentSizeLimitExceededException, __FileContentSizeLimitExceededException);
export var FileDoesNotExistException = [
    -3,
    n0,
    _FDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileDoesNotExistException, __FileDoesNotExistException);
export var FileEntryRequiredException = [
    -3,
    n0,
    _FERE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileEntryRequiredException, __FileEntryRequiredException);
export var FileMetadata = [3, n0, _FM, 0, [_aP, _bI, _fMi], [0, 0, 0]];
export var FileModeRequiredException = [
    -3,
    n0,
    _FMRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileModeRequiredException, __FileModeRequiredException);
export var FileModes = [3, n0, _FMi, 0, [_so, _des, _b], [0, 0, 0]];
export var FileNameConflictsWithDirectoryNameException = [
    -3,
    n0,
    _FNCWDNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileNameConflictsWithDirectoryNameException, __FileNameConflictsWithDirectoryNameException);
export var FilePathConflictsWithSubmodulePathException = [
    -3,
    n0,
    _FPCWSPE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FilePathConflictsWithSubmodulePathException, __FilePathConflictsWithSubmodulePathException);
export var FileSizes = [3, n0, _FS, 0, [_so, _des, _b], [1, 1, 1]];
export var FileTooLargeException = [
    -3,
    n0,
    _FTLE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FileTooLargeException, __FileTooLargeException);
export var FileVersion = [
    3,
    n0,
    _FV,
    0,
    [_commit, _bI, _p, _rCev],
    [() => Commit, 0, 0, 64 | 0],
];
export var Folder = [3, n0, _Fo, 0, [_tI, _aP, _rP], [0, 0, 0]];
export var FolderContentSizeLimitExceededException = [
    -3,
    n0,
    _FCSLEEo,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FolderContentSizeLimitExceededException, __FolderContentSizeLimitExceededException);
export var FolderDoesNotExistException = [
    -3,
    n0,
    _FDNEEo,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(FolderDoesNotExistException, __FolderDoesNotExistException);
export var GetApprovalRuleTemplateInput = [3, n0, _GARTI, 0, [_aRTN], [0]];
export var GetApprovalRuleTemplateOutput = [
    3,
    n0,
    _GARTO,
    0,
    [_aRT],
    [() => ApprovalRuleTemplate],
];
export var GetBlobInput = [3, n0, _GBI, 0, [_rN, _bI], [0, 0]];
export var GetBlobOutput = [3, n0, _GBO, 0, [_con], [21]];
export var GetBranchInput = [3, n0, _GBIe, 0, [_rN, _bN], [0, 0]];
export var GetBranchOutput = [3, n0, _GBOe, 0, [_br], [() => BranchInfo]];
export var GetCommentInput = [3, n0, _GCI, 0, [_cIom], [0]];
export var GetCommentOutput = [3, n0, _GCO, 0, [_comme], [() => Comment]];
export var GetCommentReactionsInput = [3, n0, _GCRI, 0, [_cIom, _rUA, _nT, _mR], [0, 0, 0, 1]];
export var GetCommentReactionsOutput = [
    3,
    n0,
    _GCRO,
    0,
    [_rFC, _nT],
    [() => ReactionsForCommentList, 0],
];
export var GetCommentsForComparedCommitInput = [
    3,
    n0,
    _GCFCCI,
    0,
    [_rN, _bCIe, _aCI, _nT, _mR],
    [0, 0, 0, 0, 1],
];
export var GetCommentsForComparedCommitOutput = [
    3,
    n0,
    _GCFCCO,
    0,
    [_cFCCD, _nT],
    [() => CommentsForComparedCommitData, 0],
];
export var GetCommentsForPullRequestInput = [
    3,
    n0,
    _GCFPRI,
    0,
    [_pRI, _rN, _bCIe, _aCI, _nT, _mR],
    [0, 0, 0, 0, 0, 1],
];
export var GetCommentsForPullRequestOutput = [
    3,
    n0,
    _GCFPRO,
    0,
    [_cFPRD, _nT],
    [() => CommentsForPullRequestData, 0],
];
export var GetCommitInput = [3, n0, _GCIe, 0, [_rN, _cI], [0, 0]];
export var GetCommitOutput = [3, n0, _GCOe, 0, [_commit], [() => Commit]];
export var GetDifferencesInput = [
    3,
    n0,
    _GDI,
    0,
    [_rN, _bCS, _aCS, _bP, _aPf, _MR, _NT],
    [0, 0, 0, 0, 0, 1, 0],
];
export var GetDifferencesOutput = [3, n0, _GDO, 0, [_di, _NT], [() => DifferenceList, 0]];
export var GetFileInput = [3, n0, _GFI, 0, [_rN, _cS, _fP], [0, 0, 0]];
export var GetFileOutput = [
    3,
    n0,
    _GFO,
    0,
    [_cI, _bI, _fP, _fMi, _fSi, _fC],
    [0, 0, 0, 0, 1, 21],
];
export var GetFolderInput = [3, n0, _GFIe, 0, [_rN, _cS, _fPo], [0, 0, 0]];
export var GetFolderOutput = [
    3,
    n0,
    _GFOe,
    0,
    [_cI, _fPo, _tI, _sF, _f, _sL, _sM],
    [0, 0, 0, () => FolderList, () => FileList, () => SymbolicLinkList, () => SubModuleList],
];
export var GetMergeCommitInput = [
    3,
    n0,
    _GMCI,
    0,
    [_rN, _sCS, _dCS, _cDL, _cRS],
    [0, 0, 0, 0, 0],
];
export var GetMergeCommitOutput = [3, n0, _GMCO, 0, [_sCI, _dCI, _bCI, _mCI], [0, 0, 0, 0]];
export var GetMergeConflictsInput = [
    3,
    n0,
    _GMCIe,
    0,
    [_rN, _dCS, _sCS, _mO, _cDL, _mCF, _cRS, _nT],
    [0, 0, 0, 0, 0, 1, 0, 0],
];
export var GetMergeConflictsOutput = [
    3,
    n0,
    _GMCOe,
    0,
    [_me, _dCI, _sCI, _bCI, _cML, _nT],
    [2, 0, 0, 0, () => ConflictMetadataList, 0],
];
export var GetMergeOptionsInput = [
    3,
    n0,
    _GMOI,
    0,
    [_rN, _sCS, _dCS, _cDL, _cRS],
    [0, 0, 0, 0, 0],
];
export var GetMergeOptionsOutput = [
    3,
    n0,
    _GMOO,
    0,
    [_mOer, _sCI, _dCI, _bCI],
    [64 | 0, 0, 0, 0],
];
export var GetPullRequestApprovalStatesInput = [3, n0, _GPRASI, 0, [_pRI, _rI], [0, 0]];
export var GetPullRequestApprovalStatesOutput = [
    3,
    n0,
    _GPRASO,
    0,
    [_app],
    [() => ApprovalList],
];
export var GetPullRequestInput = [3, n0, _GPRI, 0, [_pRI], [0]];
export var GetPullRequestOutput = [3, n0, _GPRO, 0, [_pR], [() => PullRequest]];
export var GetPullRequestOverrideStateInput = [3, n0, _GPROSI, 0, [_pRI, _rI], [0, 0]];
export var GetPullRequestOverrideStateOutput = [3, n0, _GPROSO, 0, [_o, _ov], [2, 0]];
export var GetRepositoryInput = [3, n0, _GRI, 0, [_rN], [0]];
export var GetRepositoryOutput = [3, n0, _GRO, 0, [_rM], [() => RepositoryMetadata]];
export var GetRepositoryTriggersInput = [3, n0, _GRTI, 0, [_rN], [0]];
export var GetRepositoryTriggersOutput = [
    3,
    n0,
    _GRTO,
    0,
    [_cIon, _tr],
    [0, () => RepositoryTriggersList],
];
export var IdempotencyParameterMismatchException = [
    -3,
    n0,
    _IPME,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(IdempotencyParameterMismatchException, __IdempotencyParameterMismatchException);
export var InvalidActorArnException = [
    -3,
    n0,
    _IAAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidActorArnException, __InvalidActorArnException);
export var InvalidApprovalRuleContentException = [
    -3,
    n0,
    _IARCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalRuleContentException, __InvalidApprovalRuleContentException);
export var InvalidApprovalRuleNameException = [
    -3,
    n0,
    _IARNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalRuleNameException, __InvalidApprovalRuleNameException);
export var InvalidApprovalRuleTemplateContentException = [
    -3,
    n0,
    _IARTCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalRuleTemplateContentException, __InvalidApprovalRuleTemplateContentException);
export var InvalidApprovalRuleTemplateDescriptionException = [
    -3,
    n0,
    _IARTDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalRuleTemplateDescriptionException, __InvalidApprovalRuleTemplateDescriptionException);
export var InvalidApprovalRuleTemplateNameException = [
    -3,
    n0,
    _IARTNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalRuleTemplateNameException, __InvalidApprovalRuleTemplateNameException);
export var InvalidApprovalStateException = [
    -3,
    n0,
    _IASE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidApprovalStateException, __InvalidApprovalStateException);
export var InvalidAuthorArnException = [
    -3,
    n0,
    _IAAEn,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidAuthorArnException, __InvalidAuthorArnException);
export var InvalidBlobIdException = [
    -3,
    n0,
    _IBIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidBlobIdException, __InvalidBlobIdException);
export var InvalidBranchNameException = [
    -3,
    n0,
    _IBNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidBranchNameException, __InvalidBranchNameException);
export var InvalidClientRequestTokenException = [
    -3,
    n0,
    _ICRTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidClientRequestTokenException, __InvalidClientRequestTokenException);
export var InvalidCommentIdException = [
    -3,
    n0,
    _ICIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidCommentIdException, __InvalidCommentIdException);
export var InvalidCommitException = [
    -3,
    n0,
    _ICE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidCommitException, __InvalidCommitException);
export var InvalidCommitIdException = [
    -3,
    n0,
    _ICIEn,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidCommitIdException, __InvalidCommitIdException);
export var InvalidConflictDetailLevelException = [
    -3,
    n0,
    _ICDLE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidConflictDetailLevelException, __InvalidConflictDetailLevelException);
export var InvalidConflictResolutionException = [
    -3,
    n0,
    _ICRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidConflictResolutionException, __InvalidConflictResolutionException);
export var InvalidConflictResolutionStrategyException = [
    -3,
    n0,
    _ICRSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidConflictResolutionStrategyException, __InvalidConflictResolutionStrategyException);
export var InvalidContinuationTokenException = [
    -3,
    n0,
    _ICTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidContinuationTokenException, __InvalidContinuationTokenException);
export var InvalidDeletionParameterException = [
    -3,
    n0,
    _IDPE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidDeletionParameterException, __InvalidDeletionParameterException);
export var InvalidDescriptionException = [
    -3,
    n0,
    _IDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidDescriptionException, __InvalidDescriptionException);
export var InvalidDestinationCommitSpecifierException = [
    -3,
    n0,
    _IDCSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidDestinationCommitSpecifierException, __InvalidDestinationCommitSpecifierException);
export var InvalidEmailException = [
    -3,
    n0,
    _IEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidEmailException, __InvalidEmailException);
export var InvalidFileLocationException = [
    -3,
    n0,
    _IFLE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidFileLocationException, __InvalidFileLocationException);
export var InvalidFileModeException = [
    -3,
    n0,
    _IFME,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidFileModeException, __InvalidFileModeException);
export var InvalidFilePositionException = [
    -3,
    n0,
    _IFPE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidFilePositionException, __InvalidFilePositionException);
export var InvalidMaxConflictFilesException = [
    -3,
    n0,
    _IMCFE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidMaxConflictFilesException, __InvalidMaxConflictFilesException);
export var InvalidMaxMergeHunksException = [
    -3,
    n0,
    _IMMHE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidMaxMergeHunksException, __InvalidMaxMergeHunksException);
export var InvalidMaxResultsException = [
    -3,
    n0,
    _IMRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidMaxResultsException, __InvalidMaxResultsException);
export var InvalidMergeOptionException = [
    -3,
    n0,
    _IMOE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidMergeOptionException, __InvalidMergeOptionException);
export var InvalidOrderException = [
    -3,
    n0,
    _IOE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidOrderException, __InvalidOrderException);
export var InvalidOverrideStatusException = [
    -3,
    n0,
    _IOSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidOverrideStatusException, __InvalidOverrideStatusException);
export var InvalidParentCommitIdException = [
    -3,
    n0,
    _IPCIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidParentCommitIdException, __InvalidParentCommitIdException);
export var InvalidPathException = [
    -3,
    n0,
    _IPE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidPathException, __InvalidPathException);
export var InvalidPullRequestEventTypeException = [
    -3,
    n0,
    _IPRETE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidPullRequestEventTypeException, __InvalidPullRequestEventTypeException);
export var InvalidPullRequestIdException = [
    -3,
    n0,
    _IPRIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidPullRequestIdException, __InvalidPullRequestIdException);
export var InvalidPullRequestStatusException = [
    -3,
    n0,
    _IPRSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidPullRequestStatusException, __InvalidPullRequestStatusException);
export var InvalidPullRequestStatusUpdateException = [
    -3,
    n0,
    _IPRSUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidPullRequestStatusUpdateException, __InvalidPullRequestStatusUpdateException);
export var InvalidReactionUserArnException = [
    -3,
    n0,
    _IRUAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidReactionUserArnException, __InvalidReactionUserArnException);
export var InvalidReactionValueException = [
    -3,
    n0,
    _IRVE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidReactionValueException, __InvalidReactionValueException);
export var InvalidReferenceNameException = [
    -3,
    n0,
    _IRNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidReferenceNameException, __InvalidReferenceNameException);
export var InvalidRelativeFileVersionEnumException = [
    -3,
    n0,
    _IRFVEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRelativeFileVersionEnumException, __InvalidRelativeFileVersionEnumException);
export var InvalidReplacementContentException = [
    -3,
    n0,
    _IRCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidReplacementContentException, __InvalidReplacementContentException);
export var InvalidReplacementTypeException = [
    -3,
    n0,
    _IRTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidReplacementTypeException, __InvalidReplacementTypeException);
export var InvalidRepositoryDescriptionException = [
    -3,
    n0,
    _IRDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryDescriptionException, __InvalidRepositoryDescriptionException);
export var InvalidRepositoryNameException = [
    -3,
    n0,
    _IRNEn,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryNameException, __InvalidRepositoryNameException);
export var InvalidRepositoryTriggerBranchNameException = [
    -3,
    n0,
    _IRTBNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerBranchNameException, __InvalidRepositoryTriggerBranchNameException);
export var InvalidRepositoryTriggerCustomDataException = [
    -3,
    n0,
    _IRTCDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerCustomDataException, __InvalidRepositoryTriggerCustomDataException);
export var InvalidRepositoryTriggerDestinationArnException = [
    -3,
    n0,
    _IRTDAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerDestinationArnException, __InvalidRepositoryTriggerDestinationArnException);
export var InvalidRepositoryTriggerEventsException = [
    -3,
    n0,
    _IRTEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerEventsException, __InvalidRepositoryTriggerEventsException);
export var InvalidRepositoryTriggerNameException = [
    -3,
    n0,
    _IRTNE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerNameException, __InvalidRepositoryTriggerNameException);
export var InvalidRepositoryTriggerRegionException = [
    -3,
    n0,
    _IRTRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRepositoryTriggerRegionException, __InvalidRepositoryTriggerRegionException);
export var InvalidResourceArnException = [
    -3,
    n0,
    _IRAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidResourceArnException, __InvalidResourceArnException);
export var InvalidRevisionIdException = [
    -3,
    n0,
    _IRIE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRevisionIdException, __InvalidRevisionIdException);
export var InvalidRuleContentSha256Exception = [
    -3,
    n0,
    _IRCSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidRuleContentSha256Exception, __InvalidRuleContentSha256Exception);
export var InvalidSortByException = [
    -3,
    n0,
    _ISBE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidSortByException, __InvalidSortByException);
export var InvalidSourceCommitSpecifierException = [
    -3,
    n0,
    _ISCSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidSourceCommitSpecifierException, __InvalidSourceCommitSpecifierException);
export var InvalidSystemTagUsageException = [
    -3,
    n0,
    _ISTUE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidSystemTagUsageException, __InvalidSystemTagUsageException);
export var InvalidTagKeysListException = [
    -3,
    n0,
    _ITKLE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTagKeysListException, __InvalidTagKeysListException);
export var InvalidTagsMapException = [
    -3,
    n0,
    _ITME,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTagsMapException, __InvalidTagsMapException);
export var InvalidTargetBranchException = [
    -3,
    n0,
    _ITBE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTargetBranchException, __InvalidTargetBranchException);
export var InvalidTargetException = [
    -3,
    n0,
    _ITE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTargetException, __InvalidTargetException);
export var InvalidTargetsException = [
    -3,
    n0,
    _ITEn,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTargetsException, __InvalidTargetsException);
export var InvalidTitleException = [
    -3,
    n0,
    _ITEnv,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(InvalidTitleException, __InvalidTitleException);
export var IsBinaryFile = [3, n0, _IBF, 0, [_so, _des, _b], [2, 2, 2]];
export var ListApprovalRuleTemplatesInput = [3, n0, _LARTI, 0, [_nT, _mR], [0, 1]];
export var ListApprovalRuleTemplatesOutput = [3, n0, _LARTO, 0, [_aRTNp, _nT], [64 | 0, 0]];
export var ListAssociatedApprovalRuleTemplatesForRepositoryInput = [
    3,
    n0,
    _LAARTFRI,
    0,
    [_rN, _nT, _mR],
    [0, 0, 1],
];
export var ListAssociatedApprovalRuleTemplatesForRepositoryOutput = [
    3,
    n0,
    _LAARTFRO,
    0,
    [_aRTNp, _nT],
    [64 | 0, 0],
];
export var ListBranchesInput = [3, n0, _LBI, 0, [_rN, _nT], [0, 0]];
export var ListBranchesOutput = [3, n0, _LBO, 0, [_bra, _nT], [64 | 0, 0]];
export var ListFileCommitHistoryRequest = [
    3,
    n0,
    _LFCHR,
    0,
    [_rN, _cS, _fP, _mR, _nT],
    [0, 0, 0, 1, 0],
];
export var ListFileCommitHistoryResponse = [
    3,
    n0,
    _LFCHRi,
    0,
    [_rDe, _nT],
    [() => RevisionDag, 0],
];
export var ListPullRequestsInput = [
    3,
    n0,
    _LPRI,
    0,
    [_rN, _aA, _pRS, _nT, _mR],
    [0, 0, 0, 0, 1],
];
export var ListPullRequestsOutput = [3, n0, _LPRO, 0, [_pRIu, _nT], [64 | 0, 0]];
export var ListRepositoriesForApprovalRuleTemplateInput = [
    3,
    n0,
    _LRFARTI,
    0,
    [_aRTN, _nT, _mR],
    [0, 0, 1],
];
export var ListRepositoriesForApprovalRuleTemplateOutput = [
    3,
    n0,
    _LRFARTO,
    0,
    [_rNe, _nT],
    [64 | 0, 0],
];
export var ListRepositoriesInput = [3, n0, _LRI, 0, [_nT, _sB, _or], [0, 0, 0]];
export var ListRepositoriesOutput = [
    3,
    n0,
    _LRO,
    0,
    [_r, _nT],
    [() => RepositoryNameIdPairList, 0],
];
export var ListTagsForResourceInput = [3, n0, _LTFRI, 0, [_rA, _nT], [0, 0]];
export var ListTagsForResourceOutput = [3, n0, _LTFRO, 0, [_tag, _nT], [128 | 0, 0]];
export var Location = [3, n0, _L, 0, [_fP, _fPil, _rFV], [0, 1, 0]];
export var ManualMergeRequiredException = [
    -3,
    n0,
    _MMRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ManualMergeRequiredException, __ManualMergeRequiredException);
export var MaximumBranchesExceededException = [
    -3,
    n0,
    _MBEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumBranchesExceededException, __MaximumBranchesExceededException);
export var MaximumConflictResolutionEntriesExceededException = [
    -3,
    n0,
    _MCREEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumConflictResolutionEntriesExceededException, __MaximumConflictResolutionEntriesExceededException);
export var MaximumFileContentToLoadExceededException = [
    -3,
    n0,
    _MFCTLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumFileContentToLoadExceededException, __MaximumFileContentToLoadExceededException);
export var MaximumFileEntriesExceededException = [
    -3,
    n0,
    _MFEEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumFileEntriesExceededException, __MaximumFileEntriesExceededException);
export var MaximumItemsToCompareExceededException = [
    -3,
    n0,
    _MITCEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumItemsToCompareExceededException, __MaximumItemsToCompareExceededException);
export var MaximumNumberOfApprovalsExceededException = [
    -3,
    n0,
    _MNOAEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumNumberOfApprovalsExceededException, __MaximumNumberOfApprovalsExceededException);
export var MaximumOpenPullRequestsExceededException = [
    -3,
    n0,
    _MOPREE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumOpenPullRequestsExceededException, __MaximumOpenPullRequestsExceededException);
export var MaximumRepositoryNamesExceededException = [
    -3,
    n0,
    _MRNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumRepositoryNamesExceededException, __MaximumRepositoryNamesExceededException);
export var MaximumRepositoryTriggersExceededException = [
    -3,
    n0,
    _MRTEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumRepositoryTriggersExceededException, __MaximumRepositoryTriggersExceededException);
export var MaximumRuleTemplatesAssociatedWithRepositoryException = [
    -3,
    n0,
    _MRTAWRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MaximumRuleTemplatesAssociatedWithRepositoryException, __MaximumRuleTemplatesAssociatedWithRepositoryException);
export var MergeBranchesByFastForwardInput = [
    3,
    n0,
    _MBBFFI,
    0,
    [_rN, _sCS, _dCS, _tB],
    [0, 0, 0, 0],
];
export var MergeBranchesByFastForwardOutput = [3, n0, _MBBFFO, 0, [_cI, _tI], [0, 0]];
export var MergeBranchesBySquashInput = [
    3,
    n0,
    _MBBSI,
    0,
    [_rN, _sCS, _dCS, _tB, _cDL, _cRS, _aN, _em, _cMo, _kEF, _cRo],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 2, () => ConflictResolution],
];
export var MergeBranchesBySquashOutput = [3, n0, _MBBSO, 0, [_cI, _tI], [0, 0]];
export var MergeBranchesByThreeWayInput = [
    3,
    n0,
    _MBBTWI,
    0,
    [_rN, _sCS, _dCS, _tB, _cDL, _cRS, _aN, _em, _cMo, _kEF, _cRo],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 2, () => ConflictResolution],
];
export var MergeBranchesByThreeWayOutput = [3, n0, _MBBTWO, 0, [_cI, _tI], [0, 0]];
export var MergeHunk = [
    3,
    n0,
    _MH,
    0,
    [_iC, _so, _des, _b],
    [2, () => MergeHunkDetail, () => MergeHunkDetail, () => MergeHunkDetail],
];
export var MergeHunkDetail = [3, n0, _MHD, 0, [_sLt, _eL, _hC], [1, 1, 0]];
export var MergeMetadata = [3, n0, _MM, 0, [_iM, _mB, _mCIe, _mO], [2, 0, 0, 0]];
export var MergeOperations = [3, n0, _MO, 0, [_so, _des], [0, 0]];
export var MergeOptionRequiredException = [
    -3,
    n0,
    _MORE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MergeOptionRequiredException, __MergeOptionRequiredException);
export var MergePullRequestByFastForwardInput = [
    3,
    n0,
    _MPRBFFI,
    0,
    [_pRI, _rN, _sCI],
    [0, 0, 0],
];
export var MergePullRequestByFastForwardOutput = [
    3,
    n0,
    _MPRBFFO,
    0,
    [_pR],
    [() => PullRequest],
];
export var MergePullRequestBySquashInput = [
    3,
    n0,
    _MPRBSI,
    0,
    [_pRI, _rN, _sCI, _cDL, _cRS, _cMo, _aN, _em, _kEF, _cRo],
    [0, 0, 0, 0, 0, 0, 0, 0, 2, () => ConflictResolution],
];
export var MergePullRequestBySquashOutput = [3, n0, _MPRBSO, 0, [_pR], [() => PullRequest]];
export var MergePullRequestByThreeWayInput = [
    3,
    n0,
    _MPRBTWI,
    0,
    [_pRI, _rN, _sCI, _cDL, _cRS, _cMo, _aN, _em, _kEF, _cRo],
    [0, 0, 0, 0, 0, 0, 0, 0, 2, () => ConflictResolution],
];
export var MergePullRequestByThreeWayOutput = [3, n0, _MPRBTWO, 0, [_pR], [() => PullRequest]];
export var MultipleConflictResolutionEntriesException = [
    -3,
    n0,
    _MCREE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MultipleConflictResolutionEntriesException, __MultipleConflictResolutionEntriesException);
export var MultipleRepositoriesInPullRequestException = [
    -3,
    n0,
    _MRIPRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(MultipleRepositoriesInPullRequestException, __MultipleRepositoriesInPullRequestException);
export var NameLengthExceededException = [
    -3,
    n0,
    _NLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(NameLengthExceededException, __NameLengthExceededException);
export var NoChangeException = [
    -3,
    n0,
    _NCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(NoChangeException, __NoChangeException);
export var NumberOfRulesExceededException = [
    -3,
    n0,
    _NOREE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(NumberOfRulesExceededException, __NumberOfRulesExceededException);
export var NumberOfRuleTemplatesExceededException = [
    -3,
    n0,
    _NORTEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(NumberOfRuleTemplatesExceededException, __NumberOfRuleTemplatesExceededException);
export var ObjectTypes = [3, n0, _OT, 0, [_so, _des, _b], [0, 0, 0]];
export var OperationNotAllowedException = [
    -3,
    n0,
    _ONAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(OperationNotAllowedException, __OperationNotAllowedException);
export var OriginApprovalRuleTemplate = [3, n0, _OART, 0, [_aRTI, _aRTN], [0, 0]];
export var OverrideAlreadySetException = [
    -3,
    n0,
    _OASE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(OverrideAlreadySetException, __OverrideAlreadySetException);
export var OverridePullRequestApprovalRulesInput = [
    3,
    n0,
    _OPRARI,
    0,
    [_pRI, _rI, _oS],
    [0, 0, 0],
];
export var OverrideStatusRequiredException = [
    -3,
    n0,
    _OSRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(OverrideStatusRequiredException, __OverrideStatusRequiredException);
export var ParentCommitDoesNotExistException = [
    -3,
    n0,
    _PCDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ParentCommitDoesNotExistException, __ParentCommitDoesNotExistException);
export var ParentCommitIdOutdatedException = [
    -3,
    n0,
    _PCIOE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ParentCommitIdOutdatedException, __ParentCommitIdOutdatedException);
export var ParentCommitIdRequiredException = [
    -3,
    n0,
    _PCIRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ParentCommitIdRequiredException, __ParentCommitIdRequiredException);
export var PathDoesNotExistException = [
    -3,
    n0,
    _PDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PathDoesNotExistException, __PathDoesNotExistException);
export var PathRequiredException = [
    -3,
    n0,
    _PRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PathRequiredException, __PathRequiredException);
export var PostCommentForComparedCommitInput = [
    3,
    n0,
    _PCFCCI,
    0,
    [_rN, _bCIe, _aCI, _l, _con, _cRT],
    [0, 0, 0, () => Location, 0, [0, 4]],
];
export var PostCommentForComparedCommitOutput = [
    3,
    n0,
    _PCFCCO,
    0,
    [_rN, _bCIe, _aCI, _bBI, _aBI, _l, _comme],
    [0, 0, 0, 0, 0, () => Location, () => Comment],
];
export var PostCommentForPullRequestInput = [
    3,
    n0,
    _PCFPRI,
    0,
    [_pRI, _rN, _bCIe, _aCI, _l, _con, _cRT],
    [0, 0, 0, 0, () => Location, 0, [0, 4]],
];
export var PostCommentForPullRequestOutput = [
    3,
    n0,
    _PCFPRO,
    0,
    [_rN, _pRI, _bCIe, _aCI, _bBI, _aBI, _l, _comme],
    [0, 0, 0, 0, 0, 0, () => Location, () => Comment],
];
export var PostCommentReplyInput = [3, n0, _PCRI, 0, [_iRT, _cRT, _con], [0, [0, 4], 0]];
export var PostCommentReplyOutput = [3, n0, _PCRO, 0, [_comme], [() => Comment]];
export var PullRequest = [
    3,
    n0,
    _PR,
    0,
    [_pRI, _t, _de, _lAD, _cD, _pRS, _aA, _pRT, _cRT, _rI, _aRp],
    [0, 0, 0, 4, 4, 0, 0, () => PullRequestTargetList, 0, 0, () => ApprovalRulesList],
];
export var PullRequestAlreadyClosedException = [
    -3,
    n0,
    _PRACE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestAlreadyClosedException, __PullRequestAlreadyClosedException);
export var PullRequestApprovalRulesNotSatisfiedException = [
    -3,
    n0,
    _PRARNSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestApprovalRulesNotSatisfiedException, __PullRequestApprovalRulesNotSatisfiedException);
export var PullRequestCannotBeApprovedByAuthorException = [
    -3,
    n0,
    _PRCBABAE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestCannotBeApprovedByAuthorException, __PullRequestCannotBeApprovedByAuthorException);
export var PullRequestCreatedEventMetadata = [
    3,
    n0,
    _PRCEM,
    0,
    [_rN, _sCI, _dCI, _mBe],
    [0, 0, 0, 0],
];
export var PullRequestDoesNotExistException = [
    -3,
    n0,
    _PRDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestDoesNotExistException, __PullRequestDoesNotExistException);
export var PullRequestEvent = [
    3,
    n0,
    _PREu,
    0,
    [_pRI, _eD, _pRET, _aAc, _pRCEM, _pRSCEM, _pRSRUEM, _pRMSCEM, _aREM, _aSCEM, _aROEM],
    [
        0,
        4,
        0,
        0,
        () => PullRequestCreatedEventMetadata,
        () => PullRequestStatusChangedEventMetadata,
        () => PullRequestSourceReferenceUpdatedEventMetadata,
        () => PullRequestMergedStateChangedEventMetadata,
        () => ApprovalRuleEventMetadata,
        () => ApprovalStateChangedEventMetadata,
        () => ApprovalRuleOverriddenEventMetadata,
    ],
];
export var PullRequestIdRequiredException = [
    -3,
    n0,
    _PRIRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestIdRequiredException, __PullRequestIdRequiredException);
export var PullRequestMergedStateChangedEventMetadata = [
    3,
    n0,
    _PRMSCEM,
    0,
    [_rN, _dR, _mM],
    [0, 0, () => MergeMetadata],
];
export var PullRequestSourceReferenceUpdatedEventMetadata = [
    3,
    n0,
    _PRSRUEM,
    0,
    [_rN, _bCIe, _aCI, _mBe],
    [0, 0, 0, 0],
];
export var PullRequestStatusChangedEventMetadata = [3, n0, _PRSCEM, 0, [_pRS], [0]];
export var PullRequestStatusRequiredException = [
    -3,
    n0,
    _PRSRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PullRequestStatusRequiredException, __PullRequestStatusRequiredException);
export var PullRequestTarget = [
    3,
    n0,
    _PRT,
    0,
    [_rN, _sR, _dR, _dC, _sC, _mBe, _mM],
    [0, 0, 0, 0, 0, 0, () => MergeMetadata],
];
export var PutCommentReactionInput = [3, n0, _PCRIu, 0, [_cIom, _rV], [0, 0]];
export var PutFileEntry = [
    3,
    n0,
    _PFE,
    0,
    [_fP, _fMi, _fC, _sFo],
    [0, 0, 21, () => SourceFileSpecifier],
];
export var PutFileEntryConflictException = [
    -3,
    n0,
    _PFECE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(PutFileEntryConflictException, __PutFileEntryConflictException);
export var PutFileInput = [
    3,
    n0,
    _PFI,
    0,
    [_rN, _bN, _fC, _fP, _fMi, _pCI, _cMo, _n, _em],
    [0, 0, 21, 0, 0, 0, 0, 0, 0],
];
export var PutFileOutput = [3, n0, _PFO, 0, [_cI, _bI, _tI], [0, 0, 0]];
export var PutRepositoryTriggersInput = [
    3,
    n0,
    _PRTI,
    0,
    [_rN, _tr],
    [0, () => RepositoryTriggersList],
];
export var PutRepositoryTriggersOutput = [3, n0, _PRTO, 0, [_cIon], [0]];
export var ReactionForComment = [
    3,
    n0,
    _RFC,
    0,
    [_re, _rU, _rFDUC],
    [() => ReactionValueFormats, 64 | 0, 1],
];
export var ReactionLimitExceededException = [
    -3,
    n0,
    _RLEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReactionLimitExceededException, __ReactionLimitExceededException);
export var ReactionValueFormats = [3, n0, _RVF, 0, [_emo, _sCh, _u], [0, 0, 0]];
export var ReactionValueRequiredException = [
    -3,
    n0,
    _RVRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReactionValueRequiredException, __ReactionValueRequiredException);
export var ReferenceDoesNotExistException = [
    -3,
    n0,
    _RDNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReferenceDoesNotExistException, __ReferenceDoesNotExistException);
export var ReferenceNameRequiredException = [
    -3,
    n0,
    _RNRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReferenceNameRequiredException, __ReferenceNameRequiredException);
export var ReferenceTypeNotSupportedException = [
    -3,
    n0,
    _RTNSE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReferenceTypeNotSupportedException, __ReferenceTypeNotSupportedException);
export var ReplaceContentEntry = [3, n0, _RCE, 0, [_fP, _rT, _con, _fMi], [0, 0, 21, 0]];
export var ReplacementContentRequiredException = [
    -3,
    n0,
    _RCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReplacementContentRequiredException, __ReplacementContentRequiredException);
export var ReplacementTypeRequiredException = [
    -3,
    n0,
    _RTRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ReplacementTypeRequiredException, __ReplacementTypeRequiredException);
export var RepositoryDoesNotExistException = [
    -3,
    n0,
    _RDNEEe,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryDoesNotExistException, __RepositoryDoesNotExistException);
export var RepositoryLimitExceededException = [
    -3,
    n0,
    _RLEEe,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryLimitExceededException, __RepositoryLimitExceededException);
export var RepositoryMetadata = [
    3,
    n0,
    _RM,
    0,
    [_aI, _rIe, _rN, _rD, _dBe, _lMD, _cD, _cUH, _cUS, _Ar, _kKI],
    [0, 0, 0, 0, 0, 4, 4, 0, 0, 0, 0],
];
export var RepositoryNameExistsException = [
    -3,
    n0,
    _RNEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryNameExistsException, __RepositoryNameExistsException);
export var RepositoryNameIdPair = [3, n0, _RNIP, 0, [_rN, _rIe], [0, 0]];
export var RepositoryNameRequiredException = [
    -3,
    n0,
    _RNREe,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryNameRequiredException, __RepositoryNameRequiredException);
export var RepositoryNamesRequiredException = [
    -3,
    n0,
    _RNREep,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryNamesRequiredException, __RepositoryNamesRequiredException);
export var RepositoryNotAssociatedWithPullRequestException = [
    -3,
    n0,
    _RNAWPRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryNotAssociatedWithPullRequestException, __RepositoryNotAssociatedWithPullRequestException);
export var RepositoryTrigger = [
    3,
    n0,
    _RT,
    0,
    [_n, _dA, _cDu, _bra, _eve],
    [0, 0, 0, 64 | 0, 64 | 0],
];
export var RepositoryTriggerBranchNameListRequiredException = [
    -3,
    n0,
    _RTBNLRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryTriggerBranchNameListRequiredException, __RepositoryTriggerBranchNameListRequiredException);
export var RepositoryTriggerDestinationArnRequiredException = [
    -3,
    n0,
    _RTDARE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryTriggerDestinationArnRequiredException, __RepositoryTriggerDestinationArnRequiredException);
export var RepositoryTriggerEventsListRequiredException = [
    -3,
    n0,
    _RTELRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryTriggerEventsListRequiredException, __RepositoryTriggerEventsListRequiredException);
export var RepositoryTriggerExecutionFailure = [3, n0, _RTEF, 0, [_tri, _fMa], [0, 0]];
export var RepositoryTriggerNameRequiredException = [
    -3,
    n0,
    _RTNRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryTriggerNameRequiredException, __RepositoryTriggerNameRequiredException);
export var RepositoryTriggersListRequiredException = [
    -3,
    n0,
    _RTLRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RepositoryTriggersListRequiredException, __RepositoryTriggersListRequiredException);
export var ResourceArnRequiredException = [
    -3,
    n0,
    _RARE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ResourceArnRequiredException, __ResourceArnRequiredException);
export var RestrictedSourceFileException = [
    -3,
    n0,
    _RSFE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RestrictedSourceFileException, __RestrictedSourceFileException);
export var RevisionIdRequiredException = [
    -3,
    n0,
    _RIRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RevisionIdRequiredException, __RevisionIdRequiredException);
export var RevisionNotCurrentException = [
    -3,
    n0,
    _RNCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(RevisionNotCurrentException, __RevisionNotCurrentException);
export var SameFileContentException = [
    -3,
    n0,
    _SFCE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(SameFileContentException, __SameFileContentException);
export var SamePathRequestException = [
    -3,
    n0,
    _SPRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(SamePathRequestException, __SamePathRequestException);
export var SetFileModeEntry = [3, n0, _SFME, 0, [_fP, _fMi], [0, 0]];
export var SourceAndDestinationAreSameException = [
    -3,
    n0,
    _SADASE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(SourceAndDestinationAreSameException, __SourceAndDestinationAreSameException);
export var SourceFileOrContentRequiredException = [
    -3,
    n0,
    _SFOCRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(SourceFileOrContentRequiredException, __SourceFileOrContentRequiredException);
export var SourceFileSpecifier = [3, n0, _SFS, 0, [_fP, _iMs], [0, 2]];
export var SubModule = [3, n0, _SM, 0, [_cI, _aP, _rP], [0, 0, 0]];
export var SymbolicLink = [3, n0, _SL, 0, [_bI, _aP, _rP, _fMi], [0, 0, 0, 0]];
export var TagKeysListRequiredException = [
    -3,
    n0,
    _TKLRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TagKeysListRequiredException, __TagKeysListRequiredException);
export var TagPolicyException = [
    -3,
    n0,
    _TPE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TagPolicyException, __TagPolicyException);
export var TagResourceInput = [3, n0, _TRI, 0, [_rA, _tag], [0, 128 | 0]];
export var TagsMapRequiredException = [
    -3,
    n0,
    _TMRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TagsMapRequiredException, __TagsMapRequiredException);
export var Target = [3, n0, _T, 0, [_rN, _sR, _dR], [0, 0, 0]];
export var TargetRequiredException = [
    -3,
    n0,
    _TRE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TargetRequiredException, __TargetRequiredException);
export var TargetsRequiredException = [
    -3,
    n0,
    _TREa,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TargetsRequiredException, __TargetsRequiredException);
export var TestRepositoryTriggersInput = [
    3,
    n0,
    _TRTI,
    0,
    [_rN, _tr],
    [0, () => RepositoryTriggersList],
];
export var TestRepositoryTriggersOutput = [
    3,
    n0,
    _TRTO,
    0,
    [_sE, _fE],
    [64 | 0, () => RepositoryTriggerExecutionFailureList],
];
export var TipOfSourceReferenceIsDifferentException = [
    -3,
    n0,
    _TOSRIDE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TipOfSourceReferenceIsDifferentException, __TipOfSourceReferenceIsDifferentException);
export var TipsDivergenceExceededException = [
    -3,
    n0,
    _TDEE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TipsDivergenceExceededException, __TipsDivergenceExceededException);
export var TitleRequiredException = [
    -3,
    n0,
    _TREi,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TitleRequiredException, __TitleRequiredException);
export var TooManyTagsException = [
    -3,
    n0,
    _TMTE,
    {
        [_e]: _c,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(TooManyTagsException, __TooManyTagsException);
export var UntagResourceInput = [3, n0, _URI, 0, [_rA, _tK], [0, 64 | 0]];
export var UpdateApprovalRuleTemplateContentInput = [
    3,
    n0,
    _UARTCI,
    0,
    [_aRTN, _nRC, _eRCS],
    [0, 0, 0],
];
export var UpdateApprovalRuleTemplateContentOutput = [
    3,
    n0,
    _UARTCO,
    0,
    [_aRT],
    [() => ApprovalRuleTemplate],
];
export var UpdateApprovalRuleTemplateDescriptionInput = [
    3,
    n0,
    _UARTDI,
    0,
    [_aRTN, _aRTD],
    [0, 0],
];
export var UpdateApprovalRuleTemplateDescriptionOutput = [
    3,
    n0,
    _UARTDO,
    0,
    [_aRT],
    [() => ApprovalRuleTemplate],
];
export var UpdateApprovalRuleTemplateNameInput = [3, n0, _UARTNI, 0, [_oARTN, _nARTN], [0, 0]];
export var UpdateApprovalRuleTemplateNameOutput = [
    3,
    n0,
    _UARTNO,
    0,
    [_aRT],
    [() => ApprovalRuleTemplate],
];
export var UpdateCommentInput = [3, n0, _UCI, 0, [_cIom, _con], [0, 0]];
export var UpdateCommentOutput = [3, n0, _UCO, 0, [_comme], [() => Comment]];
export var UpdateDefaultBranchInput = [3, n0, _UDBI, 0, [_rN, _dBN], [0, 0]];
export var UpdatePullRequestApprovalRuleContentInput = [
    3,
    n0,
    _UPRARCI,
    0,
    [_pRI, _aRN, _eRCS, _nRC],
    [0, 0, 0, 0],
];
export var UpdatePullRequestApprovalRuleContentOutput = [
    3,
    n0,
    _UPRARCO,
    0,
    [_aR],
    [() => ApprovalRule],
];
export var UpdatePullRequestApprovalStateInput = [
    3,
    n0,
    _UPRASI,
    0,
    [_pRI, _rI, _aS],
    [0, 0, 0],
];
export var UpdatePullRequestDescriptionInput = [3, n0, _UPRDI, 0, [_pRI, _de], [0, 0]];
export var UpdatePullRequestDescriptionOutput = [3, n0, _UPRDO, 0, [_pR], [() => PullRequest]];
export var UpdatePullRequestStatusInput = [3, n0, _UPRSI, 0, [_pRI, _pRS], [0, 0]];
export var UpdatePullRequestStatusOutput = [3, n0, _UPRSO, 0, [_pR], [() => PullRequest]];
export var UpdatePullRequestTitleInput = [3, n0, _UPRTI, 0, [_pRI, _t], [0, 0]];
export var UpdatePullRequestTitleOutput = [3, n0, _UPRTO, 0, [_pR], [() => PullRequest]];
export var UpdateRepositoryDescriptionInput = [3, n0, _URDI, 0, [_rN, _rD], [0, 0]];
export var UpdateRepositoryEncryptionKeyInput = [3, n0, _UREKI, 0, [_rN, _kKI], [0, 0]];
export var UpdateRepositoryEncryptionKeyOutput = [
    3,
    n0,
    _UREKO,
    0,
    [_rIe, _kKI, _oKKI],
    [0, 0, 0],
];
export var UpdateRepositoryNameInput = [3, n0, _URNI, 0, [_oN, _nN], [0, 0]];
export var UserInfo = [3, n0, _UI, 0, [_n, _em, _da], [0, 0, 0]];
export var __Unit = "unit";
export var CodeCommitServiceException = [-3, _sm, "CodeCommitServiceException", 0, [], []];
TypeRegistry.for(_sm).registerError(CodeCommitServiceException, __CodeCommitServiceException);
export var ApprovalList = [1, n0, _AL, 0, () => Approval];
export var ApprovalRulesList = [1, n0, _ARL, 0, () => ApprovalRule];
export var ApprovalRulesNotSatisfiedList = 64 | 0;
export var ApprovalRulesSatisfiedList = 64 | 0;
export var ApprovalRuleTemplateNameList = 64 | 0;
export var BatchAssociateApprovalRuleTemplateWithRepositoriesErrorsList = [
    1,
    n0,
    _BAARTWREL,
    0,
    () => BatchAssociateApprovalRuleTemplateWithRepositoriesError,
];
export var BatchDescribeMergeConflictsErrors = [
    1,
    n0,
    _BDMCEa,
    0,
    () => BatchDescribeMergeConflictsError,
];
export var BatchDisassociateApprovalRuleTemplateFromRepositoriesErrorsList = [
    1,
    n0,
    _BDARTFREL,
    0,
    () => BatchDisassociateApprovalRuleTemplateFromRepositoriesError,
];
export var BatchGetCommitsErrorsList = [1, n0, _BGCEL, 0, () => BatchGetCommitsError];
export var BatchGetRepositoriesErrorsList = [1, n0, _BGREL, 0, () => BatchGetRepositoriesError];
export var BranchNameList = 64 | 0;
export var CallerReactions = 64 | 0;
export var Comments = [1, n0, _Com, 0, () => Comment];
export var CommentsForComparedCommitData = [1, n0, _CFCCD, 0, () => CommentsForComparedCommit];
export var CommentsForPullRequestData = [1, n0, _CFPRD, 0, () => CommentsForPullRequest];
export var CommitIdsInputList = 64 | 0;
export var CommitObjectsList = [1, n0, _COL, 0, () => Commit];
export var ConflictMetadataList = [1, n0, _CML, 0, () => ConflictMetadata];
export var Conflicts = [1, n0, _Conf, 0, () => Conflict];
export var DeleteFileEntries = [1, n0, _DFEe, 0, () => DeleteFileEntry];
export var DifferenceList = [1, n0, _DL, 0, () => Difference];
export var FileList = [1, n0, _FL, 0, () => File];
export var FilePaths = 64 | 0;
export var FilesMetadata = [1, n0, _FMil, 0, () => FileMetadata];
export var FolderList = [1, n0, _FLo, 0, () => Folder];
export var MergeHunks = [1, n0, _MHe, 0, () => MergeHunk];
export var MergeOptions = 64 | 0;
export var ParentList = 64 | 0;
export var PullRequestEventList = [1, n0, _PREL, 0, () => PullRequestEvent];
export var PullRequestIdList = 64 | 0;
export var PullRequestTargetList = [1, n0, _PRTL, 0, () => PullRequestTarget];
export var PutFileEntries = [1, n0, _PFEu, 0, () => PutFileEntry];
export var ReactionsForCommentList = [1, n0, _RFCL, 0, () => ReactionForComment];
export var ReactionUsersList = 64 | 0;
export var ReplaceContentEntries = [1, n0, _RCEe, 0, () => ReplaceContentEntry];
export var RepositoryMetadataList = [1, n0, _RML, 0, () => RepositoryMetadata];
export var RepositoryNameIdPairList = [1, n0, _RNIPL, 0, () => RepositoryNameIdPair];
export var RepositoryNameList = 64 | 0;
export var RepositoryNotFoundList = 64 | 0;
export var RepositoryTriggerEventList = 64 | 0;
export var RepositoryTriggerExecutionFailureList = [
    1,
    n0,
    _RTEFL,
    0,
    () => RepositoryTriggerExecutionFailure,
];
export var RepositoryTriggerNameList = 64 | 0;
export var RepositoryTriggersList = [1, n0, _RTL, 0, () => RepositoryTrigger];
export var RevisionChildren = 64 | 0;
export var RevisionDag = [1, n0, _RD, 0, () => FileVersion];
export var SetFileModeEntries = [1, n0, _SFMEe, 0, () => SetFileModeEntry];
export var SubModuleList = [1, n0, _SML, 0, () => SubModule];
export var SymbolicLinkList = [1, n0, _SLL, 0, () => SymbolicLink];
export var TagKeysList = 64 | 0;
export var TargetList = [1, n0, _TL, 0, () => Target];
export var ReactionCountsMap = 128 | 1;
export var TagsMap = 128 | 0;
export var AssociateApprovalRuleTemplateWithRepository = [
    9,
    n0,
    _AARTWR,
    0,
    () => AssociateApprovalRuleTemplateWithRepositoryInput,
    () => __Unit,
];
export var BatchAssociateApprovalRuleTemplateWithRepositories = [
    9,
    n0,
    _BAARTWR,
    0,
    () => BatchAssociateApprovalRuleTemplateWithRepositoriesInput,
    () => BatchAssociateApprovalRuleTemplateWithRepositoriesOutput,
];
export var BatchDescribeMergeConflicts = [
    9,
    n0,
    _BDMC,
    0,
    () => BatchDescribeMergeConflictsInput,
    () => BatchDescribeMergeConflictsOutput,
];
export var BatchDisassociateApprovalRuleTemplateFromRepositories = [
    9,
    n0,
    _BDARTFR,
    0,
    () => BatchDisassociateApprovalRuleTemplateFromRepositoriesInput,
    () => BatchDisassociateApprovalRuleTemplateFromRepositoriesOutput,
];
export var BatchGetCommits = [
    9,
    n0,
    _BGC,
    0,
    () => BatchGetCommitsInput,
    () => BatchGetCommitsOutput,
];
export var BatchGetRepositories = [
    9,
    n0,
    _BGR,
    0,
    () => BatchGetRepositoriesInput,
    () => BatchGetRepositoriesOutput,
];
export var CreateApprovalRuleTemplate = [
    9,
    n0,
    _CART,
    0,
    () => CreateApprovalRuleTemplateInput,
    () => CreateApprovalRuleTemplateOutput,
];
export var CreateBranch = [9, n0, _CB, 0, () => CreateBranchInput, () => __Unit];
export var CreateCommit = [9, n0, _CC, 0, () => CreateCommitInput, () => CreateCommitOutput];
export var CreatePullRequest = [
    9,
    n0,
    _CPR,
    0,
    () => CreatePullRequestInput,
    () => CreatePullRequestOutput,
];
export var CreatePullRequestApprovalRule = [
    9,
    n0,
    _CPRAR,
    0,
    () => CreatePullRequestApprovalRuleInput,
    () => CreatePullRequestApprovalRuleOutput,
];
export var CreateRepository = [
    9,
    n0,
    _CRr,
    0,
    () => CreateRepositoryInput,
    () => CreateRepositoryOutput,
];
export var CreateUnreferencedMergeCommit = [
    9,
    n0,
    _CUMC,
    0,
    () => CreateUnreferencedMergeCommitInput,
    () => CreateUnreferencedMergeCommitOutput,
];
export var DeleteApprovalRuleTemplate = [
    9,
    n0,
    _DART,
    0,
    () => DeleteApprovalRuleTemplateInput,
    () => DeleteApprovalRuleTemplateOutput,
];
export var DeleteBranch = [9, n0, _DB, 0, () => DeleteBranchInput, () => DeleteBranchOutput];
export var DeleteCommentContent = [
    9,
    n0,
    _DCC,
    0,
    () => DeleteCommentContentInput,
    () => DeleteCommentContentOutput,
];
export var DeleteFile = [9, n0, _DF, 0, () => DeleteFileInput, () => DeleteFileOutput];
export var DeletePullRequestApprovalRule = [
    9,
    n0,
    _DPRAR,
    0,
    () => DeletePullRequestApprovalRuleInput,
    () => DeletePullRequestApprovalRuleOutput,
];
export var DeleteRepository = [
    9,
    n0,
    _DR,
    0,
    () => DeleteRepositoryInput,
    () => DeleteRepositoryOutput,
];
export var DescribeMergeConflicts = [
    9,
    n0,
    _DMC,
    0,
    () => DescribeMergeConflictsInput,
    () => DescribeMergeConflictsOutput,
];
export var DescribePullRequestEvents = [
    9,
    n0,
    _DPRE,
    0,
    () => DescribePullRequestEventsInput,
    () => DescribePullRequestEventsOutput,
];
export var DisassociateApprovalRuleTemplateFromRepository = [
    9,
    n0,
    _DARTFR,
    0,
    () => DisassociateApprovalRuleTemplateFromRepositoryInput,
    () => __Unit,
];
export var EvaluatePullRequestApprovalRules = [
    9,
    n0,
    _EPRAR,
    0,
    () => EvaluatePullRequestApprovalRulesInput,
    () => EvaluatePullRequestApprovalRulesOutput,
];
export var GetApprovalRuleTemplate = [
    9,
    n0,
    _GART,
    0,
    () => GetApprovalRuleTemplateInput,
    () => GetApprovalRuleTemplateOutput,
];
export var GetBlob = [9, n0, _GB, 0, () => GetBlobInput, () => GetBlobOutput];
export var GetBranch = [9, n0, _GBe, 0, () => GetBranchInput, () => GetBranchOutput];
export var GetComment = [9, n0, _GC, 0, () => GetCommentInput, () => GetCommentOutput];
export var GetCommentReactions = [
    9,
    n0,
    _GCR,
    0,
    () => GetCommentReactionsInput,
    () => GetCommentReactionsOutput,
];
export var GetCommentsForComparedCommit = [
    9,
    n0,
    _GCFCC,
    0,
    () => GetCommentsForComparedCommitInput,
    () => GetCommentsForComparedCommitOutput,
];
export var GetCommentsForPullRequest = [
    9,
    n0,
    _GCFPR,
    0,
    () => GetCommentsForPullRequestInput,
    () => GetCommentsForPullRequestOutput,
];
export var GetCommit = [9, n0, _GCe, 0, () => GetCommitInput, () => GetCommitOutput];
export var GetDifferences = [
    9,
    n0,
    _GD,
    0,
    () => GetDifferencesInput,
    () => GetDifferencesOutput,
];
export var GetFile = [9, n0, _GF, 0, () => GetFileInput, () => GetFileOutput];
export var GetFolder = [9, n0, _GFe, 0, () => GetFolderInput, () => GetFolderOutput];
export var GetMergeCommit = [
    9,
    n0,
    _GMC,
    0,
    () => GetMergeCommitInput,
    () => GetMergeCommitOutput,
];
export var GetMergeConflicts = [
    9,
    n0,
    _GMCe,
    0,
    () => GetMergeConflictsInput,
    () => GetMergeConflictsOutput,
];
export var GetMergeOptions = [
    9,
    n0,
    _GMO,
    0,
    () => GetMergeOptionsInput,
    () => GetMergeOptionsOutput,
];
export var GetPullRequest = [
    9,
    n0,
    _GPR,
    0,
    () => GetPullRequestInput,
    () => GetPullRequestOutput,
];
export var GetPullRequestApprovalStates = [
    9,
    n0,
    _GPRAS,
    0,
    () => GetPullRequestApprovalStatesInput,
    () => GetPullRequestApprovalStatesOutput,
];
export var GetPullRequestOverrideState = [
    9,
    n0,
    _GPROS,
    0,
    () => GetPullRequestOverrideStateInput,
    () => GetPullRequestOverrideStateOutput,
];
export var GetRepository = [9, n0, _GR, 0, () => GetRepositoryInput, () => GetRepositoryOutput];
export var GetRepositoryTriggers = [
    9,
    n0,
    _GRT,
    0,
    () => GetRepositoryTriggersInput,
    () => GetRepositoryTriggersOutput,
];
export var ListApprovalRuleTemplates = [
    9,
    n0,
    _LART,
    0,
    () => ListApprovalRuleTemplatesInput,
    () => ListApprovalRuleTemplatesOutput,
];
export var ListAssociatedApprovalRuleTemplatesForRepository = [
    9,
    n0,
    _LAARTFR,
    0,
    () => ListAssociatedApprovalRuleTemplatesForRepositoryInput,
    () => ListAssociatedApprovalRuleTemplatesForRepositoryOutput,
];
export var ListBranches = [9, n0, _LB, 0, () => ListBranchesInput, () => ListBranchesOutput];
export var ListFileCommitHistory = [
    9,
    n0,
    _LFCH,
    0,
    () => ListFileCommitHistoryRequest,
    () => ListFileCommitHistoryResponse,
];
export var ListPullRequests = [
    9,
    n0,
    _LPR,
    0,
    () => ListPullRequestsInput,
    () => ListPullRequestsOutput,
];
export var ListRepositories = [
    9,
    n0,
    _LR,
    0,
    () => ListRepositoriesInput,
    () => ListRepositoriesOutput,
];
export var ListRepositoriesForApprovalRuleTemplate = [
    9,
    n0,
    _LRFART,
    0,
    () => ListRepositoriesForApprovalRuleTemplateInput,
    () => ListRepositoriesForApprovalRuleTemplateOutput,
];
export var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    0,
    () => ListTagsForResourceInput,
    () => ListTagsForResourceOutput,
];
export var MergeBranchesByFastForward = [
    9,
    n0,
    _MBBFF,
    0,
    () => MergeBranchesByFastForwardInput,
    () => MergeBranchesByFastForwardOutput,
];
export var MergeBranchesBySquash = [
    9,
    n0,
    _MBBS,
    0,
    () => MergeBranchesBySquashInput,
    () => MergeBranchesBySquashOutput,
];
export var MergeBranchesByThreeWay = [
    9,
    n0,
    _MBBTW,
    0,
    () => MergeBranchesByThreeWayInput,
    () => MergeBranchesByThreeWayOutput,
];
export var MergePullRequestByFastForward = [
    9,
    n0,
    _MPRBFF,
    0,
    () => MergePullRequestByFastForwardInput,
    () => MergePullRequestByFastForwardOutput,
];
export var MergePullRequestBySquash = [
    9,
    n0,
    _MPRBS,
    0,
    () => MergePullRequestBySquashInput,
    () => MergePullRequestBySquashOutput,
];
export var MergePullRequestByThreeWay = [
    9,
    n0,
    _MPRBTW,
    0,
    () => MergePullRequestByThreeWayInput,
    () => MergePullRequestByThreeWayOutput,
];
export var OverridePullRequestApprovalRules = [
    9,
    n0,
    _OPRAR,
    0,
    () => OverridePullRequestApprovalRulesInput,
    () => __Unit,
];
export var PostCommentForComparedCommit = [
    9,
    n0,
    _PCFCC,
    2,
    () => PostCommentForComparedCommitInput,
    () => PostCommentForComparedCommitOutput,
];
export var PostCommentForPullRequest = [
    9,
    n0,
    _PCFPR,
    2,
    () => PostCommentForPullRequestInput,
    () => PostCommentForPullRequestOutput,
];
export var PostCommentReply = [
    9,
    n0,
    _PCR,
    2,
    () => PostCommentReplyInput,
    () => PostCommentReplyOutput,
];
export var PutCommentReaction = [9, n0, _PCRu, 0, () => PutCommentReactionInput, () => __Unit];
export var PutFile = [9, n0, _PF, 0, () => PutFileInput, () => PutFileOutput];
export var PutRepositoryTriggers = [
    9,
    n0,
    _PRTu,
    0,
    () => PutRepositoryTriggersInput,
    () => PutRepositoryTriggersOutput,
];
export var TagResource = [9, n0, _TR, 0, () => TagResourceInput, () => __Unit];
export var TestRepositoryTriggers = [
    9,
    n0,
    _TRT,
    0,
    () => TestRepositoryTriggersInput,
    () => TestRepositoryTriggersOutput,
];
export var UntagResource = [9, n0, _UR, 0, () => UntagResourceInput, () => __Unit];
export var UpdateApprovalRuleTemplateContent = [
    9,
    n0,
    _UARTC,
    0,
    () => UpdateApprovalRuleTemplateContentInput,
    () => UpdateApprovalRuleTemplateContentOutput,
];
export var UpdateApprovalRuleTemplateDescription = [
    9,
    n0,
    _UARTD,
    0,
    () => UpdateApprovalRuleTemplateDescriptionInput,
    () => UpdateApprovalRuleTemplateDescriptionOutput,
];
export var UpdateApprovalRuleTemplateName = [
    9,
    n0,
    _UARTN,
    0,
    () => UpdateApprovalRuleTemplateNameInput,
    () => UpdateApprovalRuleTemplateNameOutput,
];
export var UpdateComment = [9, n0, _UC, 0, () => UpdateCommentInput, () => UpdateCommentOutput];
export var UpdateDefaultBranch = [9, n0, _UDB, 0, () => UpdateDefaultBranchInput, () => __Unit];
export var UpdatePullRequestApprovalRuleContent = [
    9,
    n0,
    _UPRARC,
    0,
    () => UpdatePullRequestApprovalRuleContentInput,
    () => UpdatePullRequestApprovalRuleContentOutput,
];
export var UpdatePullRequestApprovalState = [
    9,
    n0,
    _UPRAS,
    0,
    () => UpdatePullRequestApprovalStateInput,
    () => __Unit,
];
export var UpdatePullRequestDescription = [
    9,
    n0,
    _UPRD,
    0,
    () => UpdatePullRequestDescriptionInput,
    () => UpdatePullRequestDescriptionOutput,
];
export var UpdatePullRequestStatus = [
    9,
    n0,
    _UPRS,
    0,
    () => UpdatePullRequestStatusInput,
    () => UpdatePullRequestStatusOutput,
];
export var UpdatePullRequestTitle = [
    9,
    n0,
    _UPRT,
    0,
    () => UpdatePullRequestTitleInput,
    () => UpdatePullRequestTitleOutput,
];
export var UpdateRepositoryDescription = [
    9,
    n0,
    _URD,
    0,
    () => UpdateRepositoryDescriptionInput,
    () => __Unit,
];
export var UpdateRepositoryEncryptionKey = [
    9,
    n0,
    _UREK,
    0,
    () => UpdateRepositoryEncryptionKeyInput,
    () => UpdateRepositoryEncryptionKeyOutput,
];
export var UpdateRepositoryName = [
    9,
    n0,
    _URN,
    0,
    () => UpdateRepositoryNameInput,
    () => __Unit,
];
