// @backstage/plugin-scaffolder-backend-module-notifications/dist/actions/sendNotification.cjs.js
'use strict';

var pluginScaffolderNode = require('@backstage/plugin-scaffolder-node');
var sendNotification_examples = require('./sendNotification.examples.cjs.js');

function createSendNotificationAction(options) {
  const { notifications } = options;
  return pluginScaffolderNode.createTemplateAction({
    id: "notification:send",
    description: "Sends a notification using NotificationService",
    examples: sendNotification_examples.examples,
    schema: {
      input: {
        recipients: (z) => z.enum(["broadcast", "entity"]).describe(
          "The recipient of the notification, either broadcast or entity. If using entity, also entityRef must be provided"
        ),
        entityRefs: (z) => z.array(z.string()).optional().describe(
          "The entity references to send the notification to, required if using recipient of entity"
        ),
        title: (z) => z.string().describe("Notification title"),
        info: (z) => z.string().optional().describe("Notification description"),
        link: (z) => z.string().optional().describe("Notification link"),
        severity: (z) => z.enum(["low", "normal", "high", "critical"]).optional().describe("Notification severity"),
        scope: (z) => z.string().optional().describe("Notification scope"),
        topic: (z) => z.string().optional().describe("Notification topic"),
        optional: (z) => z.boolean().optional().describe(
          "Do not fail the action if the notification sending fails"
        )
      }
    },
    async handler(ctx) {
      const {
        recipients,
        entityRefs,
        title,
        info,
        link,
        severity,
        topic,
        scope,
        optional
      } = ctx.input;
      ctx.logger.info(`Sending notification to ${recipients}`);
      if (recipients === "entity" && !entityRefs) {
        if (optional !== true) {
          throw new Error("Entity references must be provided");
        }
        return;
      }
      const notificationRecipients = recipients === "broadcast" ? { type: "broadcast" } : { type: "entity", entityRef: entityRefs };
      const payload = {
        title,
        description: info,
        link,
        severity,
        topic,
        scope
      };
      try {
        await ctx.checkpoint({
          key: `send.notification.${payload.title}`,
          fn: async () => {
            await notifications.send({
              recipients: notificationRecipients,
              payload
            });
          }
        });
      } catch (e) {
        ctx.logger.error(`Failed to send notification: ${e}`);
        if (optional !== true) {
          throw e;
        }
      }
    }
  });
}

exports.createSendNotificationAction = createSendNotificationAction;
