// @backstage/plugin-catalog-backend/dist/actions/createValidateEntityAction.cjs.js
'use strict';

var yaml = require('yaml');

function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e : { default: e }; }

var yaml__default = /*#__PURE__*/_interopDefaultCompat(yaml);

const createValidateEntityAction = (options) => {
  const { actionsRegistry, catalog } = options;
  actionsRegistry.register({
    name: "validate-entity",
    title: "Validate Catalog Entity",
    description: `
      This action can be used to validate catalog-info.yaml file contents meant to be used with the software catalog.
      It checks both that the YAML syntax is correct, and that the entity content is valid according to the catalog's rules.
      `,
    attributes: {
      readOnly: true,
      idempotent: true,
      destructive: false
    },
    schema: {
      input: (z) => z.object({
        entity: z.string().describe("Entity YAML content to validate"),
        location: z.string().url().optional().describe("Location to validate")
      }),
      output: (z) => z.object({
        isValid: z.boolean().describe("Whether the entity is valid"),
        isValidYaml: z.boolean().describe("Whether the YAML syntax is valid"),
        errors: z.array(z.string()).describe("Array of validation errors"),
        entity: z.record(z.any()).optional().describe("Parsed entity object if valid")
      })
    },
    action: async ({ input, credentials }) => {
      const { entity: content, location } = input;
      try {
        let entity;
        try {
          entity = yaml__default.default.parse(content);
        } catch (yamlError) {
          return {
            output: {
              isValid: false,
              isValidYaml: false,
              errors: [`YAML parsing error: ${yamlError.message}`]
            }
          };
        }
        const resp = await catalog.validateEntity(
          entity,
          location ?? "url:https://localhost/entity-validator",
          { credentials }
        );
        return {
          output: {
            isValid: resp.valid,
            isValidYaml: true,
            errors: resp.valid ? [] : resp.errors.map((e) => e.message),
            entity: resp.valid ? entity : void 0
          }
        };
      } catch (error) {
        return {
          output: {
            isValid: false,
            isValidYaml: false,
            errors: [`Validation error: ${error.message}`]
          }
        };
      }
    }
  });
};

exports.createValidateEntityAction = createValidateEntityAction;
