// @backstage/frontend-plugin-api/dist/wiring/resolveExtensionDefinition.esm.js
import { OpaqueExtensionDefinition } from '../frontend-internal/src/wiring/InternalExtensionDefinition.esm.js';
import { OpaqueExtensionInput } from '../frontend-internal/src/wiring/InternalExtensionInput.esm.js';

function resolveExtensionId(kind, namespace, name) {
  const namePart = name && namespace ? `${namespace}/${name}` : namespace || name;
  if (!namePart) {
    throw new Error(
      `Extension must declare an explicit namespace or name as it could not be resolved from context, kind=${kind} namespace=${namespace} name=${name}`
    );
  }
  return kind ? `${kind}:${namePart}` : namePart;
}
function resolveAttachTo(attachTo, namespace) {
  const resolveSpec = (spec) => {
    if (OpaqueExtensionInput.isType(spec)) {
      const { context } = OpaqueExtensionInput.toInternal(spec);
      if (!context) {
        throw new Error(
          "Invalid input object without a parent extension used as attachment point"
        );
      }
      return {
        id: resolveExtensionId(context.kind, namespace, context.name),
        input: context.input
      };
    }
    if ("relative" in spec && spec.relative) {
      return {
        id: resolveExtensionId(
          spec.relative.kind,
          namespace,
          spec.relative.name
        ),
        input: spec.input
      };
    }
    if ("id" in spec) {
      return { id: spec.id, input: spec.input };
    }
    throw new Error("Invalid attachment point specification");
  };
  if (Array.isArray(attachTo)) {
    return attachTo.map(resolveSpec);
  }
  return resolveSpec(attachTo);
}
function resolveExtensionDefinition(definition, context) {
  const internalDefinition = OpaqueExtensionDefinition.toInternal(definition);
  const {
    name,
    kind,
    namespace: internalNamespace,
    override: _skip2,
    attachTo,
    ...rest
  } = internalDefinition;
  const namespace = internalNamespace ?? context?.namespace;
  const id = resolveExtensionId(kind, namespace, name);
  return {
    ...rest,
    attachTo: resolveAttachTo(attachTo, namespace),
    $$type: "@backstage/Extension",
    version: internalDefinition.version,
    id,
    toString() {
      return `Extension{id=${id}}`;
    }
  };
}

export { resolveExtensionDefinition };
