// @backstage/frontend-plugin-api/dist/routing/ExternalRouteRef.esm.js
import { describeParentCallSite } from './describeParentCallSite.esm.js';
import { OpaqueExternalRouteRef } from '../frontend-internal/src/routing/OpaqueExternalRouteRef.esm.js';
import '../frontend-internal/src/routing/OpaqueRouteRef.esm.js';
import '../frontend-internal/src/routing/OpaqueSubRouteRef.esm.js';

function createExternalRouteRef(config) {
  const params = config?.params ?? [];
  const creationSite = describeParentCallSite();
  let id = void 0;
  return OpaqueExternalRouteRef.createInstance("v1", {
    T: void 0,
    getParams() {
      return params;
    },
    getDescription() {
      if (id) {
        return id;
      }
      return `created at '${creationSite}'`;
    },
    getDefaultTarget() {
      return config?.defaultTarget;
    },
    setId(newId) {
      if (!newId) {
        throw new Error(`ExternalRouteRef id must be a non-empty string`);
      }
      if (id && id !== newId) {
        throw new Error(
          `ExternalRouteRef was referenced twice as both '${id}' and '${newId}'`
        );
      }
      id = newId;
    },
    toString() {
      return `externalRouteRef{id=${id},at='${creationSite}'}`;
    }
  });
}

export { createExternalRouteRef };
