// @backstage/frontend-plugin-api/dist/routing/SubRouteRef.esm.js
import { OpaqueSubRouteRef } from '../frontend-internal/src/routing/OpaqueSubRouteRef.esm.js';
import { OpaqueRouteRef } from '../frontend-internal/src/routing/OpaqueRouteRef.esm.js';
import '../frontend-internal/src/routing/OpaqueExternalRouteRef.esm.js';

const PARAM_PATTERN = /^\w+$/;
function createSubRouteRef(config) {
  const { path, parent } = config;
  const internalParent = OpaqueRouteRef.toInternal(parent);
  const parentParams = internalParent.getParams();
  const parentDescription = internalParent.getDescription();
  const pathParams = path.split("/").filter((p) => p.startsWith(":")).map((p) => p.substring(1));
  const params = [...parentParams, ...pathParams];
  if (parentParams.some((p) => pathParams.includes(p))) {
    throw new Error(
      "SubRouteRef may not have params that overlap with its parent"
    );
  }
  if (!path.startsWith("/")) {
    throw new Error(`SubRouteRef path must start with '/', got '${path}'`);
  }
  if (path.endsWith("/")) {
    throw new Error(`SubRouteRef path must not end with '/', got '${path}'`);
  }
  for (const param of pathParams) {
    if (!PARAM_PATTERN.test(param)) {
      throw new Error(`SubRouteRef path has invalid param, got '${param}'`);
    }
  }
  return OpaqueSubRouteRef.createInstance("v1", {
    T: void 0,
    path,
    getParams() {
      return params;
    },
    getParent() {
      return parent;
    },
    getDescription() {
      return `at ${path} with parent ${parentDescription}`;
    },
    toString() {
      return `subRouteRef{path='${path}',parent=${parent}}`;
    }
  });
}

export { createSubRouteRef };
