// @backstage-community/plugin-tech-insights-react/dist/components/ResultLinksMenu/ResultLinksMenu.esm.js
import { jsx } from 'react/jsx-runtime';
import { useMemo, useState, useEffect, useCallback } from 'react';
import { useApi } from '@backstage/core-plugin-api';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { techInsightsApiRef } from '../../api/TechInsightsApi.esm.js';
import '@backstage-community/plugin-tech-insights-common/client';
import '@material-ui/icons/CheckCircleOutline';
import '@material-ui/icons/ErrorOutline';
import { stringifyEntityRef } from '@backstage/catalog-model';

const ResultLinksMenu = (props) => {
  const { result, entity, setMenu } = props;
  const api = useApi(techInsightsApiRef);
  const links = useMemo(
    () => entity ? api.getLinksForEntity(result, entity, {
      includeStaticLinks: true
    }) : result.check.links ?? [],
    [api, result, entity]
  );
  const menuId = `menu-${result.check.id}-${entity ? stringifyEntityRef(entity) : "unknown"}`;
  const [anchorEl, setAnchorEl] = useState(void 0);
  useEffect(() => {
    if (links.length === 0) {
      setMenu(void 0);
      return;
    }
    setMenu({
      open: (elem) => {
        setAnchorEl(elem);
      }
    });
  }, [setMenu, links]);
  const handleClose = useCallback(() => {
    setAnchorEl(void 0);
  }, [setAnchorEl]);
  if (links.length === 0) {
    return null;
  }
  return /* @__PURE__ */ jsx(
    Menu,
    {
      id: menuId,
      anchorEl: anchorEl ?? null,
      keepMounted: true,
      open: Boolean(anchorEl),
      onClose: handleClose,
      children: links.map((link, i) => /* @__PURE__ */ jsx(
        MenuItem,
        {
          button: true,
          component: "a",
          href: link.url,
          target: link.url.startsWith("/") ? void 0 : "_blank",
          onClick: handleClose,
          children: link.title
        },
        `${i}-${link.url}`
      ))
    }
  );
};

export { ResultLinksMenu };
