// @backstage-community/plugin-tech-insights-common/dist/permissions.cjs.js
'use strict';

var pluginPermissionCommon = require('@backstage/plugin-permission-common');

const techInsightsCheckReadPermission = pluginPermissionCommon.createPermission({
  name: "tech-insights.check.read",
  attributes: {
    action: "read"
  }
});
const techInsightsCheckUpdatePermission = pluginPermissionCommon.createPermission({
  name: "tech-insights.check.run",
  attributes: {
    action: "update"
  }
});
const techInsightsFactRetrieverReadPermission = pluginPermissionCommon.createPermission({
  name: "tech-insights.fact-retriever.read",
  attributes: {
    action: "read"
  }
});
const techInsightsPermissions = [
  techInsightsCheckReadPermission,
  techInsightsCheckUpdatePermission,
  techInsightsFactRetrieverReadPermission
];

exports.techInsightsCheckReadPermission = techInsightsCheckReadPermission;
exports.techInsightsCheckUpdatePermission = techInsightsCheckUpdatePermission;
exports.techInsightsFactRetrieverReadPermission = techInsightsFactRetrieverReadPermission;
exports.techInsightsPermissions = techInsightsPermissions;
