// @backstage-community/plugin-tech-insights-backend/dist/plugin/plugin.cjs.js
'use strict';

var backendPluginApi = require('@backstage/backend-plugin-api');
var pluginTechInsightsCommon = require('@backstage-community/plugin-tech-insights-common');
var pluginTechInsightsNode = require('@backstage-community/plugin-tech-insights-node');
var entityMetadataFactRetriever = require('../service/fact/factRetrievers/entityMetadataFactRetriever.cjs.js');
var entityOwnershipFactRetriever = require('../service/fact/factRetrievers/entityOwnershipFactRetriever.cjs.js');
var techdocsFactRetriever = require('../service/fact/factRetrievers/techdocsFactRetriever.cjs.js');
require('../service/persistence/persistenceContext.cjs.js');
var router = require('../service/router.cjs.js');
var techInsightsContextBuilder = require('../service/techInsightsContextBuilder.cjs.js');
var config = require('./config.cjs.js');

const techInsightsPlugin = backendPluginApi.createBackendPlugin({
  pluginId: "tech-insights",
  register(env) {
    let factCheckerFactory = void 0;
    env.registerExtensionPoint(pluginTechInsightsNode.techInsightsFactCheckerFactoryExtensionPoint, {
      setFactCheckerFactory(factory) {
        factCheckerFactory = factory;
      }
    });
    let factRetrieverRegistry = void 0;
    env.registerExtensionPoint(
      pluginTechInsightsNode.techInsightsFactRetrieverRegistryExtensionPoint,
      {
        setFactRetrieverRegistry(registry) {
          factRetrieverRegistry = registry;
        }
      }
    );
    const addedFactRetrievers = {
      entityMetadataFactRetriever: entityMetadataFactRetriever.entityMetadataFactRetriever,
      entityOwnershipFactRetriever: entityOwnershipFactRetriever.entityOwnershipFactRetriever,
      techdocsFactRetriever: techdocsFactRetriever.techdocsFactRetriever
    };
    env.registerExtensionPoint(pluginTechInsightsNode.techInsightsFactRetrieversExtensionPoint, {
      addFactRetrievers(factRetrievers) {
        Object.entries(factRetrievers).forEach(([key, value]) => {
          addedFactRetrievers[key] = value;
        });
      }
    });
    let persistenceContext = void 0;
    env.registerExtensionPoint(pluginTechInsightsNode.techInsightsPersistenceContextExtensionPoint, {
      setPersistenceContext(context) {
        persistenceContext = context;
      }
    });
    env.registerInit({
      deps: {
        config: backendPluginApi.coreServices.rootConfig,
        database: backendPluginApi.coreServices.database,
        discovery: backendPluginApi.coreServices.discovery,
        httpRouter: backendPluginApi.coreServices.httpRouter,
        logger: backendPluginApi.coreServices.logger,
        scheduler: backendPluginApi.coreServices.scheduler,
        auth: backendPluginApi.coreServices.auth,
        urlReader: backendPluginApi.coreServices.urlReader,
        httpAuth: backendPluginApi.coreServices.httpAuth,
        permissions: backendPluginApi.coreServices.permissions,
        permissionsRegistry: backendPluginApi.coreServices.permissionsRegistry
      },
      async init({
        config: config$1,
        database,
        discovery,
        httpRouter,
        logger,
        scheduler,
        auth,
        urlReader,
        httpAuth,
        permissions,
        permissionsRegistry
      }) {
        permissionsRegistry.addPermissions(pluginTechInsightsCommon.techInsightsPermissions);
        const factRetrievers = Object.entries(
          addedFactRetrievers
        ).map(
          ([name, factRetriever]) => config.createFactRetrieverRegistrationFromConfig(
            config$1,
            name,
            factRetriever
          )
        ).filter((registration) => registration);
        const context = await techInsightsContextBuilder.buildTechInsightsContext({
          config: config$1,
          database,
          discovery,
          factCheckerFactory,
          factRetrieverRegistry,
          factRetrievers,
          logger,
          persistenceContext,
          scheduler,
          auth,
          urlReader
        });
        httpRouter.use(
          await router.createRouter({
            ...context,
            config: config$1,
            logger,
            permissions,
            httpAuth
          })
        );
      }
    });
  }
});

exports.techInsightsPlugin = techInsightsPlugin;
