// @roadiehq/backstage-plugin-github-insights/dist/components/Widgets/ContributorsCard/ContributorsCard.esm.js
import { jsx } from 'react/jsx-runtime';
import { makeStyles } from '@material-ui/core/styles';
import Alert from '@material-ui/lab/Alert';
import { MissingAnnotationEmptyState, Progress, InfoCard } from '@backstage/core-components';
import ContributorsList from './components/ContributorsList/ContributorsList.esm.js';
import { useRequest } from '../../../hooks/useRequest.esm.js';
import { useEntityGithubScmIntegration } from '../../../hooks/useEntityGithubScmIntegration.esm.js';
import { useProjectEntity } from '../../../hooks/useProjectEntity.esm.js';
import { isGithubInsightsAvailable, GITHUB_INSIGHTS_ANNOTATION } from '../../utils/isGithubInsightsAvailable.esm.js';
import { useEntity } from '@backstage/plugin-catalog-react';
import { getHostname } from '../../utils/githubUtils.esm.js';
import { GitHubAuthorizationWrapper } from '@roadiehq/github-auth-utils-react';

const useStyles = makeStyles((theme) => ({
  infoCard: {
    marginBottom: "10px",
    "& + .MuiAlert-root": {
      marginTop: theme.spacing(3)
    }
  }
}));
const ContributorsCardContent = () => {
  const { entity } = useEntity();
  const classes = useStyles();
  const { value, loading, error } = useRequest(entity, "contributors", 10);
  const { hostname } = useEntityGithubScmIntegration(entity);
  const projectAlert = isGithubInsightsAvailable(entity);
  const { owner, repo } = useProjectEntity(entity);
  if (!projectAlert) {
    return /* @__PURE__ */ jsx(MissingAnnotationEmptyState, { annotation: GITHUB_INSIGHTS_ANNOTATION });
  }
  if (loading) {
    return /* @__PURE__ */ jsx(Progress, {});
  } else if (error) {
    return /* @__PURE__ */ jsx(Alert, { severity: "error", className: classes.infoCard, children: error.message });
  }
  return /* @__PURE__ */ jsx(
    InfoCard,
    {
      title: "Contributors",
      deepLink: {
        link: `https://${hostname}/${owner}/${repo}/graphs/contributors`,
        title: "People",
        onClick: (e) => {
          e.preventDefault();
          window.open(
            `https://${hostname}/${owner}/${repo}/graphs/contributors`
          );
        }
      },
      className: classes.infoCard,
      children: /* @__PURE__ */ jsx(ContributorsList, { contributors: value || [] })
    }
  );
};
const ContributorsCard = () => {
  const { entity } = useEntity();
  const hostname = getHostname(entity);
  return /* @__PURE__ */ jsx(GitHubAuthorizationWrapper, { title: "Contributors", hostname, children: /* @__PURE__ */ jsx(ContributorsCardContent, {}) });
};

export { ContributorsCard as default };
