// @backstage/plugin-home/dist/homePageComponents/VisitedByType/Context.esm.js
import { jsx } from 'react/jsx-runtime';
import { useState, useMemo, createContext, useContext as useContext$1 } from 'react';

const defaultContextValueOnly = {
  collapsed: true,
  numVisitsOpen: 3,
  numVisitsTotal: 8,
  visits: [],
  loading: true,
  kind: "recent"
};
const defaultContextValue = {
  ...defaultContextValueOnly,
  setCollapsed: () => {
  },
  setNumVisitsOpen: () => {
  },
  setNumVisitsTotal: () => {
  },
  setVisits: () => {
  },
  setLoading: () => {
  },
  setKind: () => {
  },
  setContext: () => {
  }
};
const Context = createContext(defaultContextValue);
const getFilteredSet = (setContext, contextKey) => (e) => setContext((state) => ({
  ...state,
  [contextKey]: typeof e === "function" ? e(state[contextKey]) : e
}));
const ContextProvider = ({ children }) => {
  const [context, setContext] = useState({
    ...defaultContextValueOnly
  });
  const {
    setCollapsed,
    setNumVisitsOpen,
    setNumVisitsTotal,
    setVisits,
    setLoading,
    setKind
  } = useMemo(
    () => ({
      setCollapsed: getFilteredSet(setContext, "collapsed"),
      setNumVisitsOpen: getFilteredSet(setContext, "numVisitsOpen"),
      setNumVisitsTotal: getFilteredSet(setContext, "numVisitsTotal"),
      setVisits: getFilteredSet(setContext, "visits"),
      setLoading: getFilteredSet(setContext, "loading"),
      setKind: getFilteredSet(setContext, "kind")
    }),
    [setContext]
  );
  const value = {
    ...context,
    setContext,
    setCollapsed,
    setNumVisitsOpen,
    setNumVisitsTotal,
    setVisits,
    setLoading,
    setKind
  };
  return /* @__PURE__ */ jsx(Context.Provider, { value, children });
};
const useContext = () => {
  const value = useContext$1(Context);
  if (value === void 0)
    throw new Error(
      "VisitedByType useContext found undefined ContextValue, <ContextProvider/> could be missing"
    );
  return value;
};

export { Context, ContextProvider, Context as default, defaultContextValue, useContext };
