// @stagecentral/plugin-artifactory/dist/esm/index-b81d2320.esm.js
import React, { useState, useEffect } from 'react';
import { useEntity } from '@backstage/plugin-catalog-react';
import { useApi, configApiRef } from '@backstage/core-plugin-api';
import { ArtifactoryPluginApiRef } from '../index.esm.js';
import { Table } from '@backstage/core-components';

const useArtifactoryObjects = (entity) => {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);
  const [artifacts, setArtifacts] = useState([]);
  const myPluginApi = useApi(ArtifactoryPluginApiRef);
  const getObjects = async () => {
    try {
      const artifacts2 = await myPluginApi.getArtifacts(entity);
      setArtifacts(artifacts2);
    } catch (e) {
      setError(true);
    } finally {
      setLoading(false);
    }
  };
  useEffect(() => {
    getObjects();
  });
  return {
    error,
    loading,
    artifacts
  };
};

const DenseTable = ({ artifacts }) => {
  const configApi = useApi(configApiRef);
  const jfrogUrl = configApi.getString("artifactory.url") + "/artifactory";
  const columns = [
    { title: "Name", field: "name" },
    { title: "path", field: "path" },
    { title: "repo", field: "repo" },
    { title: "url", field: "url", render: (row) => /* @__PURE__ */ React.createElement("a", { href: `${jfrogUrl}/${row.repo}/${row.path}/${row.name}` }, `${jfrogUrl}/${row.repo}/${row.path}/${row.name}`) }
  ];
  const data = artifacts.map((artifact) => {
    return {
      name: artifact.name,
      path: artifact.path,
      repo: artifact.repo,
      url: `${jfrogUrl}/artifactory/${artifact.repo}/${artifact.path}/${artifact.name}`
    };
  });
  return /* @__PURE__ */ React.createElement(
    Table,
    {
      title: `Artifact List (fetching data from ${jfrogUrl})`,
      options: { search: false, paging: false },
      columns,
      data
    }
  );
};
const ArtifactoryComponent = () => {
  const { entity } = useEntity();
  const { error, loading, artifacts } = useArtifactoryObjects(entity);
  if (loading) {
    return /* @__PURE__ */ React.createElement("div", null, "Loading");
  }
  if (error) {
    return /* @__PURE__ */ React.createElement("div", null, "Error");
  }
  return /* @__PURE__ */ React.createElement(DenseTable, { artifacts: artifacts || [] });
};

export { ArtifactoryComponent };
