// @stagecentral/plugin-artifactory/dist/index.esm.js
import { createApiRef, createRouteRef, createPlugin, createApiFactory, discoveryApiRef, createRoutableExtension } from '@backstage/core-plugin-api';

class ArtifactoryPluginBackendClient {
  constructor(options) {
    this.discoveryApi = options.discoveryApi;
  }
  async handleResponse(response) {
    if (!response.ok) {
      throw new Error();
    }
    const data = await response.json();
    if (!data.artifacts) {
      return data;
    }
    return data.artifacts;
  }
  async getHealth() {
    const url = `${await this.discoveryApi.getBaseUrl("artifactory")}/health`;
    const response = await fetch(url, {
      method: "GET"
    });
    return await this.handleResponse(response);
  }
  async getArtifacts(entity) {
    const url = `${await this.discoveryApi.getBaseUrl("artifactory")}/artifacts`;
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "puka": "puka"
      },
      body: `{"entityRef": "${entity.kind}:${entity.metadata.namespace}/${entity.metadata.name}" }`
    });
    return await this.handleResponse(response);
  }
}

const ArtifactoryPluginApiRef = createApiRef({
  id: "plugin.artifactoryplugin.service"
});

const rootRouteRef = createRouteRef({
  id: "stagecentral-artifactory"
});

const ArtifactoryPlugin = createPlugin({
  id: "stagecentral-artifactory",
  apis: [
    createApiFactory({
      api: ArtifactoryPluginApiRef,
      deps: {
        discoveryApi: discoveryApiRef
      },
      factory: ({ discoveryApi }) => new ArtifactoryPluginBackendClient({ discoveryApi })
    })
  ],
  routes: {
    root: rootRouteRef
  }
});
const EntityArtifactoryContent = ArtifactoryPlugin.provide(
  createRoutableExtension({
    name: "EntityArtifactoryContent",
    component: () => import('./esm/index-b81d2320.esm.js').then((m) => m.ArtifactoryComponent),
    mountPoint: rootRouteRef
  })
);

export { ArtifactoryPlugin, ArtifactoryPluginApiRef, EntityArtifactoryContent };
