// @backstage/ui/dist/components/HeaderPage/HeaderPage.esm.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import { Text } from '../Text/Text.esm.js';
import { RiArrowRightSLine } from '@remixicon/react';
import { Tabs, TabList, Tab } from '../Tabs/Tabs.esm.js';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { HeaderPageDefinition } from './definition.esm.js';
import { Container } from '../Container/Container.esm.js';
import { Link } from '../Link/Link.esm.js';
import styles from './HeaderPage.module.css.esm.js';
import clsx from 'clsx';

const HeaderPage = (props) => {
  const { classNames, cleanedProps } = useStyles(HeaderPageDefinition, props);
  const { className, title, tabs, customActions, breadcrumbs } = cleanedProps;
  return /* @__PURE__ */ jsxs(
    Container,
    {
      className: clsx(classNames.root, styles[classNames.root], className),
      children: [
        /* @__PURE__ */ jsxs("div", { className: clsx(classNames.content, styles[classNames.content]), children: [
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: clsx(
                classNames.breadcrumbs,
                styles[classNames.breadcrumbs]
              ),
              children: [
                breadcrumbs && breadcrumbs.map((breadcrumb) => /* @__PURE__ */ jsxs(Fragment, { children: [
                  /* @__PURE__ */ jsx(
                    Link,
                    {
                      href: breadcrumb.href,
                      variant: "title-small",
                      weight: "bold",
                      color: "secondary",
                      truncate: true,
                      style: { maxWidth: "240px" },
                      children: breadcrumb.label
                    }
                  ),
                  /* @__PURE__ */ jsx(RiArrowRightSLine, { size: 16, color: "var(--bui-fg-secondary)" })
                ] }, breadcrumb.label)),
                /* @__PURE__ */ jsx(Text, { variant: "title-small", weight: "bold", as: "h2", children: title })
              ]
            }
          ),
          /* @__PURE__ */ jsx("div", { className: clsx(classNames.controls, styles[classNames.controls]), children: customActions })
        ] }),
        tabs && /* @__PURE__ */ jsx(
          "div",
          {
            className: clsx(
              classNames.tabsWrapper,
              styles[classNames.tabsWrapper]
            ),
            children: /* @__PURE__ */ jsx(Tabs, { children: /* @__PURE__ */ jsx(TabList, { children: tabs.map((tab) => /* @__PURE__ */ jsx(
              Tab,
              {
                id: tab.id,
                href: tab.href,
                matchStrategy: tab.matchStrategy,
                children: tab.label
              },
              tab.id
            )) }) })
          }
        )
      ]
    }
  );
};

export { HeaderPage };
