// @backstage-community/plugin-jfrog-artifactory/dist/plugin.esm.js
import { createPlugin, createApiFactory, identityApiRef, configApiRef, discoveryApiRef, createComponentExtension } from '@backstage/core-plugin-api';
import { JfrogArtifactoryApiClient, jfrogArtifactoryApiRef } from './api/index.esm.js';
import { JFROG_ARTIFACTORY_ANNOTATION_IMAGE_NAME } from './components/useJfrogArtifactoryAppData.esm.js';
import { rootRouteRef } from './routes.esm.js';

const jfrogArtifactoryPlugin = createPlugin({
  id: "jfrog-artifactory",
  routes: {
    root: rootRouteRef
  },
  apis: [
    createApiFactory({
      api: jfrogArtifactoryApiRef,
      deps: {
        discoveryApi: discoveryApiRef,
        configApi: configApiRef,
        identityApi: identityApiRef
      },
      factory: ({ discoveryApi, configApi, identityApi }) => new JfrogArtifactoryApiClient({ discoveryApi, configApi, identityApi })
    })
  ]
});
const JfrogArtifactoryPage = jfrogArtifactoryPlugin.provide(
  createComponentExtension({
    name: "JfrogArtifactoryPage",
    component: {
      lazy: () => import('./components/JfrogArtifactoryDashboardPage/index.esm.js').then(
        (m) => m.JfrogArtifactoryDashboardPage
      )
    }
  })
);
const isJfrogArtifactoryAvailable = (entity) => Boolean(
  entity?.metadata.annotations?.[JFROG_ARTIFACTORY_ANNOTATION_IMAGE_NAME]
);

export { JfrogArtifactoryPage, isJfrogArtifactoryAvailable, jfrogArtifactoryPlugin };
