// @backstage/ui/dist/components/Select/Select.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { forwardRef, useEffect } from 'react';
import { Select as Select$1, Popover } from 'react-aria-components';
import clsx from 'clsx';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { SelectDefinition } from './definition.esm.js';
import { PopoverDefinition } from '../Popover/definition.esm.js';
import { FieldLabel } from '../FieldLabel/FieldLabel.esm.js';
import { FieldError } from '../FieldError/FieldError.esm.js';
import styles from './Select.module.css.esm.js';
import stylesPopover from '../Popover/Popover.module.css.esm.js';
import { SelectTrigger } from './SelectTrigger.esm.js';
import { SelectContent } from './SelectContent.esm.js';

const Select = forwardRef((props, ref) => {
  const { classNames: popoverClassNames } = useStyles(PopoverDefinition);
  const { classNames, dataAttributes, cleanedProps } = useStyles(
    SelectDefinition,
    {
      size: "small",
      placeholder: "Select an option",
      ...props
    }
  );
  const {
    className,
    label,
    description,
    options,
    icon,
    searchable,
    searchPlaceholder,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledBy,
    isRequired,
    secondaryLabel,
    ...rest
  } = cleanedProps;
  useEffect(() => {
    if (!label && !ariaLabel && !ariaLabelledBy) {
      console.warn(
        "Select requires either a visible label, aria-label, or aria-labelledby for accessibility"
      );
    }
  }, [label, ariaLabel, ariaLabelledBy]);
  const secondaryLabelText = secondaryLabel || (isRequired ? "Required" : null);
  return /* @__PURE__ */ jsxs(
    Select$1,
    {
      className: clsx(classNames.root, styles[classNames.root], className),
      ...dataAttributes,
      ref,
      "aria-label": ariaLabel,
      "aria-labelledby": ariaLabelledBy,
      ...rest,
      children: [
        /* @__PURE__ */ jsx(
          FieldLabel,
          {
            label,
            secondaryLabel: secondaryLabelText,
            description
          }
        ),
        /* @__PURE__ */ jsx(SelectTrigger, { icon }),
        /* @__PURE__ */ jsx(FieldError, {}),
        /* @__PURE__ */ jsx(
          Popover,
          {
            className: clsx(
              popoverClassNames.root,
              stylesPopover[popoverClassNames.root],
              classNames.popover,
              styles[classNames.popover]
            ),
            ...dataAttributes,
            children: /* @__PURE__ */ jsx(
              SelectContent,
              {
                searchable,
                searchPlaceholder,
                options
              }
            )
          }
        )
      ]
    }
  );
});
Select.displayName = "Select";

export { Select };
