// @backstage/ui/dist/components/Select/SelectListBox.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { ListBox, ListBoxItem, Text } from 'react-aria-components';
import { RiCheckLine } from '@remixicon/react';
import clsx from 'clsx';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { SelectDefinition } from './definition.esm.js';
import styles from './Select.module.css.esm.js';

const NoResults = () => {
  const { classNames } = useStyles(SelectDefinition);
  return /* @__PURE__ */ jsx("div", { className: clsx(classNames.noResults, styles[classNames.noResults]), children: "No results found." });
};
function SelectListBox({ options, ...props }) {
  const { classNames } = useStyles(SelectDefinition, props);
  return /* @__PURE__ */ jsx(
    ListBox,
    {
      className: clsx(classNames.list, styles[classNames.list]),
      renderEmptyState: () => /* @__PURE__ */ jsx(NoResults, {}),
      children: options?.map((option) => /* @__PURE__ */ jsxs(
        ListBoxItem,
        {
          id: option.value,
          textValue: option.label,
          className: clsx(classNames.item, styles[classNames.item]),
          isDisabled: option.disabled,
          children: [
            /* @__PURE__ */ jsx(
              "div",
              {
                className: clsx(
                  classNames.itemIndicator,
                  styles[classNames.itemIndicator]
                ),
                children: /* @__PURE__ */ jsx(RiCheckLine, {})
              }
            ),
            /* @__PURE__ */ jsx(
              Text,
              {
                slot: "label",
                className: clsx(classNames.itemLabel, styles[classNames.itemLabel]),
                children: option.label
              }
            )
          ]
        },
        option.value
      ))
    }
  );
}

export { SelectListBox };
