// @backstage/plugin-home/dist/homePageComponents/Toolkit/Content.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { Link } from '@backstage/core-components';
import List from '@material-ui/core/List';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
import { useToolkit } from './Context.esm.js';

const useStyles = makeStyles((theme) => ({
  toolkit: {
    display: "flex",
    flexWrap: "wrap",
    textAlign: "center"
  },
  tool: {
    margin: theme.spacing(0.5, 1)
  },
  label: {
    marginTop: theme.spacing(1),
    width: "72px",
    fontSize: "0.9em",
    lineHeight: "1.25",
    overflowWrap: "break-word",
    color: theme.palette.text.secondary
  },
  icon: {
    width: "64px",
    height: "64px",
    borderRadius: "50px",
    justifyContent: "center",
    alignItems: "center",
    boxShadow: theme.shadows[1],
    backgroundColor: theme.palette.background.default
  }
}));
const Content = (props) => {
  const classes = useStyles();
  const toolkit = useToolkit();
  const tools = toolkit?.tools ?? props.tools;
  return /* @__PURE__ */ jsx(List, { className: classes.toolkit, children: tools.map((tool) => /* @__PURE__ */ jsxs(Link, { to: tool.url, className: classes.tool, children: [
    /* @__PURE__ */ jsx(ListItemIcon, { className: classes.icon, children: tool.icon }),
    /* @__PURE__ */ jsx(
      ListItemText,
      {
        secondaryTypographyProps: { className: classes.label },
        secondary: tool.label
      }
    )
  ] }, tool.url)) });
};

export { Content };
