// @backstage/frontend-test-utils/dist/app/renderTestApp.esm.js
import { jsx } from 'react/jsx-runtime';
import { createSpecializedApp } from '@backstage/frontend-app-api';
import { RouterBlueprint, createFrontendPlugin, coreExtensionData } from '@backstage/frontend-plugin-api';
import { render } from '@testing-library/react';
import appPlugin from '@backstage/plugin-app';
import { ConfigReader } from '@backstage/config';
import { MemoryRouter } from 'react-router-dom';

const DEFAULT_MOCK_CONFIG = {
  app: { baseUrl: "http://localhost:3000" },
  backend: { baseUrl: "http://localhost:7007" }
};
const appPluginOverride = appPlugin.withOverrides({
  extensions: [
    appPlugin.getExtension("sign-in-page:app").override({
      disabled: true
    })
  ]
});
function renderTestApp(options) {
  const extensions = [
    RouterBlueprint.make({
      params: {
        component: ({ children }) => /* @__PURE__ */ jsx(MemoryRouter, { initialEntries: options.initialRouteEntries, children })
      }
    }),
    ...options.extensions ?? []
  ];
  const features = [
    createFrontendPlugin({
      pluginId: "test",
      extensions
    }),
    appPluginOverride
  ];
  if (options.features) {
    features.push(...options.features);
  }
  const app = createSpecializedApp({
    features,
    config: ConfigReader.fromConfigs([
      {
        context: "render-config",
        data: options?.config ?? DEFAULT_MOCK_CONFIG
      }
    ])
  });
  return render(
    app.tree.root.instance.getData(coreExtensionData.reactElement)
  );
}

export { renderTestApp };
