// @backstage/ui/dist/components/Accordion/Accordion.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { Disclosure, Heading, Button, DisclosurePanel, DisclosureGroup } from 'react-aria-components';
import { RiArrowDownSLine } from '@remixicon/react';
import clsx from 'clsx';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { AccordionDefinition } from './definition.esm.js';
import styles from './Accordion.module.css.esm.js';
import { Flex } from '../Flex/Flex.esm.js';

const Accordion = forwardRef(({ className, ...props }, ref) => {
  const { classNames, cleanedProps } = useStyles(AccordionDefinition, props);
  return /* @__PURE__ */ jsx(
    Disclosure,
    {
      ref,
      className: clsx(classNames.root, styles[classNames.root], className),
      ...cleanedProps
    }
  );
});
Accordion.displayName = "Accordion";
const AccordionTrigger = forwardRef(({ className, title, subtitle, children, ...props }, ref) => {
  const { classNames, cleanedProps } = useStyles(AccordionDefinition, props);
  return /* @__PURE__ */ jsx(
    Heading,
    {
      ref,
      className: clsx(
        classNames.trigger,
        styles[classNames.trigger],
        className
      ),
      ...cleanedProps,
      children: /* @__PURE__ */ jsxs(
        Button,
        {
          slot: "trigger",
          className: clsx(
            classNames.triggerButton,
            styles[classNames.triggerButton]
          ),
          children: [
            children ? children : /* @__PURE__ */ jsxs(Flex, { gap: "2", align: "center", children: [
              /* @__PURE__ */ jsx(
                "span",
                {
                  className: clsx(
                    classNames.triggerTitle,
                    styles[classNames.triggerTitle]
                  ),
                  children: title
                }
              ),
              subtitle && /* @__PURE__ */ jsx(
                "span",
                {
                  className: clsx(
                    classNames.triggerSubtitle,
                    styles[classNames.triggerSubtitle]
                  ),
                  children: subtitle
                }
              )
            ] }),
            /* @__PURE__ */ jsx(
              RiArrowDownSLine,
              {
                className: clsx(
                  classNames.triggerIcon,
                  styles[classNames.triggerIcon]
                ),
                size: 16
              }
            )
          ]
        }
      )
    }
  );
});
AccordionTrigger.displayName = "AccordionTrigger";
const AccordionPanel = forwardRef(({ className, ...props }, ref) => {
  const { classNames, cleanedProps } = useStyles(AccordionDefinition, props);
  return /* @__PURE__ */ jsx(
    DisclosurePanel,
    {
      ref,
      className: clsx(classNames.panel, styles[classNames.panel], className),
      ...cleanedProps
    }
  );
});
AccordionPanel.displayName = "AccordionPanel";
const AccordionGroup = forwardRef(({ className, allowsMultiple = false, ...props }, ref) => {
  const { classNames, cleanedProps } = useStyles(AccordionDefinition, props);
  return /* @__PURE__ */ jsx(
    DisclosureGroup,
    {
      ref,
      allowsMultipleExpanded: allowsMultiple,
      className: clsx(classNames.group, styles[classNames.group], className),
      ...cleanedProps
    }
  );
});
AccordionGroup.displayName = "AccordionGroup";

export { Accordion, AccordionGroup, AccordionPanel, AccordionTrigger };
