// @backstage/core-plugin-api/dist/apis/definitions/auth.esm.js
import 'react/jsx-runtime';
import '@backstage/version-bridge';
import '@backstage/errors';
import { createApiRef } from '../system/ApiRef.esm.js';

const SessionState = {
  /**
   * User signed in.
   */
  SignedIn: "SignedIn",
  /**
   * User not signed in.
   */
  SignedOut: "SignedOut"
};
const googleAuthApiRef = createApiRef({
  id: "core.auth.google"
});
const githubAuthApiRef = createApiRef({
  id: "core.auth.github"
});
const oktaAuthApiRef = createApiRef({
  id: "core.auth.okta"
});
const gitlabAuthApiRef = createApiRef({
  id: "core.auth.gitlab"
});
const microsoftAuthApiRef = createApiRef({
  id: "core.auth.microsoft"
});
const oneloginAuthApiRef = createApiRef({
  id: "core.auth.onelogin"
});
const bitbucketAuthApiRef = createApiRef({
  id: "core.auth.bitbucket"
});
const bitbucketServerAuthApiRef = createApiRef({
  id: "core.auth.bitbucket-server"
});
const atlassianAuthApiRef = createApiRef({
  id: "core.auth.atlassian"
});
const vmwareCloudAuthApiRef = createApiRef({
  id: "core.auth.vmware-cloud"
});
const openshiftAuthApiRef = createApiRef({
  id: "core.auth.openshift"
});

export { SessionState, atlassianAuthApiRef, bitbucketAuthApiRef, bitbucketServerAuthApiRef, githubAuthApiRef, gitlabAuthApiRef, googleAuthApiRef, microsoftAuthApiRef, oktaAuthApiRef, oneloginAuthApiRef, openshiftAuthApiRef, vmwareCloudAuthApiRef };
