// @backstage/plugin-kubernetes-backend/dist/service/KubernetesInitializer.cjs.js
'use strict';

var KubernetesFetcher = require('./KubernetesFetcher.cjs.js');
var buildDefaultAuthStrategyMap = require('../auth/buildDefaultAuthStrategyMap.cjs.js');
var luxon = require('luxon');
var index = require('../cluster-locator/index.cjs.js');
var DispatchStrategy = require('../auth/DispatchStrategy.cjs.js');
var buildDefaultServiceLocator = require('../service-locator/buildDefaultServiceLocator.cjs.js');
var KubernetesFanOutHandler = require('./KubernetesFanOutHandler.cjs.js');

class KubernetesInitializer {
  opts;
  constructor(opts) {
    this.opts = opts;
  }
  static create(opts) {
    return new KubernetesInitializer(opts);
  }
  async defaultFetcher() {
    return new KubernetesFetcher.KubernetesClientBasedFetcher({
      logger: this.opts.logger
    });
  }
  async defaultAuthStrategy() {
    return buildDefaultAuthStrategyMap.buildDefaultAuthStrategyMap({
      logger: this.opts.logger,
      config: this.opts.config
    });
  }
  async defaultClusterSupplier(opts) {
    const refreshInterval = luxon.Duration.fromObject({
      minutes: 60
    });
    return index.getCombinedClusterSupplier(
      this.opts.config,
      this.opts.catalog,
      new DispatchStrategy.DispatchStrategy({
        authStrategyMap: Object.fromEntries(opts.authStrategyMap.entries())
      }),
      this.opts.logger,
      refreshInterval,
      this.opts.auth
    );
  }
  async defaultServiceLocator(opts) {
    return buildDefaultServiceLocator.buildDefaultServiceLocator({
      config: this.opts.config,
      clusterSupplier: opts.clusterSupplier
    });
  }
  async defaultObjectsProvider(opts) {
    return new KubernetesFanOutHandler.KubernetesFanOutHandler({
      logger: this.opts.logger,
      config: this.opts.config,
      fetcher: opts.fetcher,
      serviceLocator: opts.serviceLocator,
      customResources: opts.customResources,
      objectTypesToFetch: opts.objectTypesToFetch,
      authStrategy: new DispatchStrategy.DispatchStrategy({
        authStrategyMap: Object.fromEntries(opts.authStrategyMap.entries())
      })
    });
  }
  async defaultObjectsProviderOptions() {
    const customResources = (this.opts.config.getOptionalConfigArray("kubernetes.customResources") ?? []).map(
      (c) => ({
        group: c.getString("group"),
        apiVersion: c.getString("apiVersion"),
        plural: c.getString("plural"),
        objectType: "customresources"
      })
    );
    const objectTypesToFetchStrings = this.opts.config.getOptionalStringArray(
      "kubernetes.objectTypes"
    );
    const apiVersionOverrides = this.opts.config.getOptionalConfig(
      "kubernetes.apiVersionOverrides"
    );
    let objectTypesToFetch = void 0;
    if (objectTypesToFetchStrings) {
      objectTypesToFetch = KubernetesFanOutHandler.ALL_OBJECTS.filter(
        (obj) => objectTypesToFetchStrings.includes(obj.objectType)
      );
    }
    if (apiVersionOverrides) {
      objectTypesToFetch ??= KubernetesFanOutHandler.DEFAULT_OBJECTS;
      for (const obj of objectTypesToFetch) {
        if (apiVersionOverrides.has(obj.objectType)) {
          obj.apiVersion = apiVersionOverrides.getString(obj.objectType);
        }
      }
    }
    return {
      customResources,
      objectTypesToFetch: objectTypesToFetch ?? KubernetesFanOutHandler.DEFAULT_OBJECTS
    };
  }
  async init() {
    const fetcher = await this.opts.fetcher?.({
      getDefault: () => this.defaultFetcher()
    }) ?? await this.defaultFetcher();
    const authStrategyMap = this.opts.authStrategyMap ?? await this.defaultAuthStrategy();
    const clusterSupplier = await this.opts.clusterSupplier?.({
      getDefault: () => this.defaultClusterSupplier({ authStrategyMap })
    }) ?? await this.defaultClusterSupplier({ authStrategyMap });
    const serviceLocator = await this.opts.serviceLocator?.({
      getDefault: () => this.defaultServiceLocator({ clusterSupplier }),
      clusterSupplier
    }) ?? await this.defaultServiceLocator({ clusterSupplier });
    const { customResources, objectTypesToFetch } = await this.defaultObjectsProviderOptions();
    const objectsProvider = await this.opts.objectsProvider?.({
      getDefault: () => this.defaultObjectsProvider({
        clusterSupplier,
        authStrategyMap,
        fetcher,
        serviceLocator,
        customResources,
        objectTypesToFetch
      }),
      clusterSupplier,
      serviceLocator,
      customResources,
      objectTypesToFetch,
      authStrategy: new DispatchStrategy.DispatchStrategy({
        authStrategyMap: Object.fromEntries(authStrategyMap.entries())
      })
    }) ?? await this.defaultObjectsProvider({
      clusterSupplier,
      authStrategyMap,
      fetcher,
      serviceLocator,
      customResources,
      objectTypesToFetch
    });
    return {
      fetcher,
      authStrategyMap,
      clusterSupplier,
      serviceLocator,
      objectsProvider
    };
  }
}

exports.KubernetesInitializer = KubernetesInitializer;
