// @backstage-community/plugin-jfrog-artifactory/dist/api/index.esm.js
import { createApiRef } from '@backstage/core-plugin-api';

const DEFAULT_PROXY_PATH = "/jfrog-artifactory/api";
const jfrogArtifactoryApiRef = createApiRef({
  id: "plugin.jfrog-artifactory.service"
});
class JfrogArtifactoryApiClient {
  // @ts-ignore
  discoveryApi;
  configApi;
  identityApi;
  constructor(options) {
    this.discoveryApi = options.discoveryApi;
    this.configApi = options.configApi;
    this.identityApi = options.identityApi;
  }
  async getBaseUrl(target) {
    const proxyPath = this.configApi.getOptionalString("jfrogArtifactory.proxyPath") || DEFAULT_PROXY_PATH;
    return `${await this.discoveryApi.getBaseUrl("proxy")}${target ?? proxyPath}`;
  }
  async fetcher(url, query) {
    const { token: idToken } = await this.identityApi.getCredentials();
    const response = await fetch(url, {
      headers: {
        "Content-Type": "application/json",
        ...idToken && { Authorization: `Bearer ${idToken}` }
      },
      method: "POST",
      body: query
    });
    if (!response.ok) {
      throw new Error(
        `failed to fetch data, status ${response.status}: ${response.statusText}`
      );
    }
    return await response.json();
  }
  async getTags(repo, target) {
    const proxyUrl = await this.getBaseUrl(target);
    const tagQuery = {
      query: "query ($filter: VersionFilter!, $first: Int, $orderBy: VersionOrder) { versions (filter: $filter, first: $first, orderBy: $orderBy) { edges { node { name, created, modified, package { id }, repos { name, type, leadFilePath }, licenses { name, source }, size, stats { downloadCount }, vulnerabilities { critical, high, medium, low, info, unknown, skipped }, files { name, lead, size, md5, sha1, sha256, mimeType } } } } }",
      variables: {
        filter: {
          packageId: `docker://${repo}`,
          name: "*",
          ignorePreRelease: false
        },
        first: 100,
        orderBy: {
          field: "NAME_SEMVER",
          direction: "DESC"
        }
      }
    };
    return await this.fetcher(
      `${proxyUrl}/metadata/api/v1/query`,
      JSON.stringify(tagQuery)
    );
  }
}

export { JfrogArtifactoryApiClient, jfrogArtifactoryApiRef };
