// @backstage-community/plugin-tech-insights/dist/plugin.esm.js
import { createPlugin, createApiFactory, discoveryApiRef, identityApiRef, createComponentExtension, createRoutableExtension } from '@backstage/core-plugin-api';
import { techInsightsApiRef, TechInsightsClient } from '@backstage-community/plugin-tech-insights-react';
import { rootRouteRef } from './routes.esm.js';

const techInsightsPlugin = createPlugin({
  id: "tech-insights",
  apis: [
    createApiFactory({
      api: techInsightsApiRef,
      deps: { discoveryApi: discoveryApiRef, identityApi: identityApiRef },
      factory: ({ discoveryApi, identityApi }) => new TechInsightsClient({ discoveryApi, identityApi })
    })
  ],
  routes: {
    root: rootRouteRef
  }
});
const ScorecardInfo = techInsightsPlugin.provide(
  createComponentExtension({
    name: "ScorecardInfo",
    component: {
      lazy: () => import('./components/ScorecardsInfo/index.esm.js').then((m) => m.ScorecardInfo)
    }
  })
);
const ScorecardsList = techInsightsPlugin.provide(
  createComponentExtension({
    name: "ScorecardsList",
    component: {
      lazy: () => import('./components/ScorecardsList/index.esm.js').then((m) => m.ScorecardsList)
    }
  })
);
techInsightsPlugin.provide(
  createComponentExtension({
    name: "ScorecardBadge",
    component: {
      lazy: () => import('./components/ScorecardsBadge/index.esm.js').then((m) => m.ScorecardsBadge)
    }
  })
);
techInsightsPlugin.provide(
  createComponentExtension({
    name: "ScorecardGauge",
    component: {
      lazy: () => import('./components/ScorecardsGauge/index.esm.js').then((m) => m.ScorecardsGauge)
    }
  })
);
const EntityTechInsightsScorecardContent = techInsightsPlugin.provide(
  createRoutableExtension({
    name: "EntityTechInsightsScorecardContent",
    component: () => import('./components/ScorecardsContent/index.esm.js').then((m) => m.ScorecardsContent),
    mountPoint: rootRouteRef
  })
);
const EntityTechInsightsScorecardCard = techInsightsPlugin.provide(
  createRoutableExtension({
    name: "EntityTechInsightsScorecardCard",
    component: () => import('./components/ScorecardsCard/index.esm.js').then((m) => m.ScorecardsCard),
    mountPoint: rootRouteRef
  })
);
const TechInsightsScorecardPage = techInsightsPlugin.provide(
  createRoutableExtension({
    name: "TechInsightsScorecardPage",
    component: () => import('./components/ScorecardsPage/index.esm.js').then((m) => m.ScorecardsPage),
    mountPoint: rootRouteRef
  })
);
const TechInsightsCheckIcon = techInsightsPlugin.provide(
  createComponentExtension({
    name: "TechInsightsCheckIcon",
    component: {
      lazy: () => import('@backstage-community/plugin-tech-insights-react').then(
        (m) => m.ResultCheckIcon
      )
    }
  })
);
const TechInsightsLinksMenu = techInsightsPlugin.provide(
  createComponentExtension({
    name: "TechInsightsLinksMenu",
    component: {
      lazy: () => import('@backstage-community/plugin-tech-insights-react').then(
        (m) => m.ResultLinksMenu
      )
    }
  })
);

export { EntityTechInsightsScorecardCard, EntityTechInsightsScorecardContent, ScorecardInfo, ScorecardsList, TechInsightsCheckIcon, TechInsightsLinksMenu, TechInsightsScorecardPage, techInsightsPlugin };
