// @roadiehq/backstage-plugin-github-pull-requests/dist/components/Home/YourOpenPullRequestsCard/Content.esm.js
import { jsx } from 'react/jsx-runtime';
import { SkeletonPullRequestsListView, PullRequestsListView } from '../../PullRequestsListView/PullRequestsListView.esm.js';
import { useGithubSearchPullRequest } from '../../useGithubSearchPullRequest.esm.js';
import Alert from '@material-ui/lab/Alert';
import { GitHubAuthorizationWrapper } from '@roadiehq/github-auth-utils-react';

const defaultPullRequestsQuery = "is:open is:pr author:@me archived:false";
const OpenPullRequestsContent = (props) => {
  const { query = defaultPullRequestsQuery, hostname } = props;
  const { loading, error, value } = useGithubSearchPullRequest(query, hostname);
  if (loading) return /* @__PURE__ */ jsx(SkeletonPullRequestsListView, {});
  if (error) return /* @__PURE__ */ jsx(Alert, { severity: "error", children: error.message });
  return /* @__PURE__ */ jsx(
    PullRequestsListView,
    {
      data: value,
      emptyStateText: "No open pull requests."
    }
  );
};
const Content = (props) => {
  return /* @__PURE__ */ jsx(
    GitHubAuthorizationWrapper,
    {
      title: "Your Pull Requests",
      hostname: props.hostname,
      children: /* @__PURE__ */ jsx(OpenPullRequestsContent, { ...props })
    }
  );
};

export { Content };
