// @backstage/ui/dist/components/Button/Button.esm.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import clsx from 'clsx';
import { forwardRef } from 'react';
import { Button as Button$1, ProgressBar } from 'react-aria-components';
import { RiLoader4Line } from '@remixicon/react';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { ButtonDefinition } from './definition.esm.js';
import stylesButton from './Button.module.css.esm.js';

const Button = forwardRef(
  (props, ref) => {
    const { classNames, dataAttributes, cleanedProps } = useStyles(
      ButtonDefinition,
      {
        size: "small",
        variant: "primary",
        ...props
      }
    );
    const { children, className, iconStart, iconEnd, loading, ...rest } = cleanedProps;
    return /* @__PURE__ */ jsx(
      Button$1,
      {
        className: clsx(classNames.root, stylesButton[classNames.root], className),
        ref,
        isPending: loading,
        ...dataAttributes,
        ...rest,
        children: ({ isPending }) => /* @__PURE__ */ jsxs(Fragment, { children: [
          /* @__PURE__ */ jsxs(
            "span",
            {
              className: clsx(classNames.content, stylesButton[classNames.content]),
              children: [
                iconStart,
                children,
                iconEnd
              ]
            }
          ),
          isPending && /* @__PURE__ */ jsx(
            ProgressBar,
            {
              "aria-label": "Loading",
              isIndeterminate: true,
              className: clsx(classNames.spinner, stylesButton[classNames.spinner]),
              children: /* @__PURE__ */ jsx(RiLoader4Line, { "aria-hidden": "true" })
            }
          )
        ] })
      }
    );
  }
);
Button.displayName = "Button";

export { Button };
