// @backstage/plugin-home/dist/homePageComponents/QuickStart/Content.esm.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Link } from '@backstage/core-components';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import { ContentModal } from '@backstage/plugin-home-react';
import { useStyles } from './styles.esm.js';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const Content = (props) => {
  const styles = useStyles();
  const { t } = useTranslationRef(homeTranslationRef);
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(
      ContentModal,
      {
        modalContent: props.image,
        linkContent: props.modalTitle || t("quickStart.title")
      }
    ),
    /* @__PURE__ */ jsx(Typography, { variant: "body1", paragraph: true, children: props.cardDescription || t("quickStart.description") }),
    /* @__PURE__ */ jsx(ContentModal, { modalContent: props.image, linkContent: props.image }),
    /* @__PURE__ */ jsxs(
      Grid,
      {
        container: true,
        alignItems: "center",
        className: styles.contentActionContainer,
        children: [
          props.downloadImage && /* @__PURE__ */ jsx(Grid, { item: true, children: props.downloadImage }),
          /* @__PURE__ */ jsx(Grid, { item: true, children: /* @__PURE__ */ jsx(
            Link,
            {
              to: props.docsLink || "https://backstage.io/docs/getting-started/",
              "data-testid": "quick-start-link-to-docs",
              underline: "none",
              variant: "h6",
              className: styles.link,
              children: props.docsLinkTitle || t("quickStart.learnMoreLinkTitle")
            }
          ) })
        ]
      }
    ),
    props.additionalContent && props.additionalContent || props.video && props.video
  ] });
};

export { Content };
