// @backstage/plugin-home/dist/homePageComponents/QuickStart/styles.esm.js
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
  (theme) => ({
    cardTitleIcon: {
      verticalAlign: "bottom",
      marginLeft: "-4px"
    },
    contentActionContainer: {
      marginTop: theme.spacing(1.5),
      marginBottom: theme.spacing(1.5)
    },
    contentModal: {
      position: "absolute",
      top: "50%",
      left: "50%",
      transform: "translate(-50%, -50%)",
      width: "80%",
      height: "auto"
    },
    imageSize: { width: "100%", height: "100%" },
    link: {
      display: "inline-flex",
      alignItems: "center",
      textDecoration: "none",
      color: `${theme.palette.link}`,
      "&:hover": {
        background: "transparent"
      }
    },
    linkText: {
      marginBottom: theme.spacing(1.5)
    },
    videoContainer: {
      borderRadius: "10px",
      width: "100%",
      height: "auto",
      background: `${theme.palette.background.default}`
    }
  }),
  { name: "HomeQuickStartCard" }
);

export { useStyles };
