// @backstage-community/plugin-tech-insights/dist/components/ScorecardsBadge/ScorecardsBadge.esm.js
import { jsx } from 'react/jsx-runtime';
import Chip from '@material-ui/core/Chip';
import { useApi } from '@backstage/core-plugin-api';
import { techInsightsApiRef } from '@backstage-community/plugin-tech-insights-react';
import Tooltip from '@material-ui/core/Tooltip';
import { ScorecardsList } from '../ScorecardsList/ScorecardsList.esm.js';

const ScorecardsBadge = (props) => {
  const { checkResults, entity } = props;
  const api = useApi(techInsightsApiRef);
  const types = [...new Set(checkResults.map(({ check }) => check.type))];
  const checkResultRenderers = api.getCheckResultRenderers(types);
  const checkResultsWithRenderer = checkResults.map((result) => ({
    result,
    renderer: checkResultRenderers.find(
      (renderer) => renderer.type === result.check.type
    )
  }));
  const succeeded = checkResultsWithRenderer.filter(
    ({ result, renderer }) => !renderer?.isFailed?.(result)
  ).length;
  return /* @__PURE__ */ jsx(
    Tooltip,
    {
      title: /* @__PURE__ */ jsx(
        ScorecardsList,
        {
          checkResults,
          entity,
          dense: true,
          hideDescription: true
        }
      ),
      children: /* @__PURE__ */ jsx(
        Chip,
        {
          label: `${succeeded}/${checkResults.length}`,
          size: "small",
          style: {
            backgroundColor: succeeded === checkResults.length ? "mediumseagreen" : "orangered"
          }
        }
      )
    }
  );
};

export { ScorecardsBadge };
