// @backstage/ui/dist/components/TablePagination/TablePagination.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import clsx from 'clsx';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { TablePaginationDefinition } from './definition.esm.js';
import styles from './TablePagination.module.css.esm.js';
import { RiArrowLeftSLine, RiArrowRightSLine } from '@remixicon/react';
import { Select } from '../Select/Select.esm.js';
import { ButtonIcon } from '../ButtonIcon/ButtonIcon.esm.js';
import { Text } from '../Text/Text.esm.js';

function TablePagination(props) {
  const { classNames, cleanedProps } = useStyles(TablePaginationDefinition, {
    showPageSizeOptions: true,
    ...props
  });
  const {
    className,
    offset,
    pageSize,
    rowCount,
    onNextPage,
    onPreviousPage,
    onPageSizeChange,
    setOffset,
    setPageSize,
    showPageSizeOptions,
    ...rest
  } = cleanedProps;
  const currentOffset = offset ?? 0;
  const currentPageSize = pageSize ?? 10;
  const fromCount = currentOffset + 1;
  const toCount = Math.min(currentOffset + currentPageSize, rowCount ?? 0);
  const nextPage = () => {
    const totalRows = rowCount ?? 0;
    const nextOffset = currentOffset + currentPageSize;
    if (nextOffset < totalRows) {
      onNextPage?.();
      setOffset?.(nextOffset);
    }
  };
  const previousPage = () => {
    if (currentOffset > 0) {
      onPreviousPage?.();
      const prevOffset = Math.max(0, currentOffset - currentPageSize);
      setOffset?.(prevOffset);
    }
  };
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: clsx(classNames.root, styles[classNames.root], className),
      ...rest,
      children: [
        /* @__PURE__ */ jsx("div", { className: clsx(classNames.left, styles[classNames.left]), children: showPageSizeOptions && /* @__PURE__ */ jsx(
          Select,
          {
            name: "pageSize",
            size: "small",
            placeholder: "Show 10 results",
            options: [
              { label: "Show 5 results", value: "5" },
              { label: "Show 10 results", value: "10" },
              { label: "Show 20 results", value: "20" },
              { label: "Show 30 results", value: "30" },
              { label: "Show 40 results", value: "40" },
              { label: "Show 50 results", value: "50" }
            ],
            selectedKey: pageSize?.toString(),
            onSelectionChange: (value) => {
              const newPageSize = Number(value);
              setPageSize?.(newPageSize);
              onPageSizeChange?.(newPageSize);
            },
            className: clsx(classNames.select, styles[classNames.select])
          }
        ) }),
        /* @__PURE__ */ jsxs("div", { className: clsx(classNames.right, styles[classNames.right]), children: [
          /* @__PURE__ */ jsx(
            Text,
            {
              as: "p",
              variant: "body-medium",
              children: `${fromCount} - ${toCount} of ${rowCount}`
            }
          ),
          /* @__PURE__ */ jsx(
            ButtonIcon,
            {
              variant: "secondary",
              size: "small",
              onClick: previousPage,
              isDisabled: currentOffset === 0,
              icon: /* @__PURE__ */ jsx(RiArrowLeftSLine, {}),
              "aria-label": "Previous"
            }
          ),
          /* @__PURE__ */ jsx(
            ButtonIcon,
            {
              variant: "secondary",
              size: "small",
              onClick: nextPage,
              isDisabled: rowCount !== void 0 && currentOffset + currentPageSize >= rowCount,
              icon: /* @__PURE__ */ jsx(RiArrowRightSLine, {}),
              "aria-label": "Next"
            }
          )
        ] })
      ]
    }
  );
}

export { TablePagination };
