// @backstage-community/plugin-grafana/dist/plugin.esm.js
import { createPlugin, createApiFactory, discoveryApiRef, fetchApiRef, configApiRef, createComponentExtension } from '@backstage/core-plugin-api';
import { grafanaApiRef, GrafanaApiClient, UnifiedAlertingGrafanaApiClient } from './api.esm.js';

const grafanaPlugin = createPlugin({
  id: "grafana",
  apis: [
    createApiFactory({
      api: grafanaApiRef,
      deps: {
        discoveryApi: discoveryApiRef,
        fetchApi: fetchApiRef,
        configApi: configApiRef
      },
      factory: ({ discoveryApi, fetchApi, configApi }) => {
        const unifiedAlertingEnabled = configApi.getOptionalBoolean("grafana.unifiedAlerting") || false;
        if (!unifiedAlertingEnabled) {
          return new GrafanaApiClient({
            discoveryApi,
            fetchApi,
            domain: configApi.getString("grafana.domain"),
            proxyPath: configApi.getOptionalString("grafana.proxyPath")
          });
        }
        return new UnifiedAlertingGrafanaApiClient({
          discoveryApi,
          fetchApi,
          domain: configApi.getString("grafana.domain"),
          proxyPath: configApi.getOptionalString("grafana.proxyPath")
        });
      }
    })
  ]
});
const EntityGrafanaDashboardsCard = grafanaPlugin.provide(
  createComponentExtension({
    name: "EntityGrafanaDashboardsCard",
    component: {
      lazy: () => import('./components/DashboardsCard/index.esm.js').then((m) => m.DashboardsCard)
    }
  })
);
const EntityGrafanaAlertsCard = grafanaPlugin.provide(
  createComponentExtension({
    name: "EntityGrafanaAlertsCard",
    component: {
      lazy: () => import('./components/AlertsCard/index.esm.js').then((m) => m.AlertsCard)
    }
  })
);
const EntityOverviewDashboardViewer = grafanaPlugin.provide(
  createComponentExtension({
    name: "EntityOverviewDashboardViewer",
    component: {
      lazy: () => import('./components/DashboardViewer/index.esm.js').then(
        (m) => m.EntityDashboardViewer
      )
    }
  })
);

export { EntityGrafanaAlertsCard, EntityGrafanaDashboardsCard, EntityOverviewDashboardViewer, grafanaPlugin };
