// @backstage-community/plugin-tech-insights/dist/components/ScorecardsPage/Filters.esm.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import Typography from '@material-ui/core/Typography';
import Autocomplete from '@material-ui/lab/Autocomplete';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import TextField from '@material-ui/core/TextField';
import Box from '@material-ui/core/Box';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
import CheckBoxIcon from '@material-ui/icons/CheckBox';
import Checkbox from '@material-ui/core/Checkbox';
import Tooltip from '@material-ui/core/Tooltip';
import { useApi } from '@backstage/core-plugin-api';
import useAsync from 'react-use/lib/useAsync';
import { ErrorPanel } from '@backstage/core-components';
import { techInsightsApiRef } from '@backstage-community/plugin-tech-insights-react';

const useStyles = makeStyles({
  fullWidth: { width: "100%" },
  boxLabel: {
    width: "100%",
    textOverflow: "ellipsis",
    overflow: "hidden"
  }
});
const FixedWidthFormControlLabel = withStyles((_theme) => ({
  label: {
    width: "100%"
  },
  root: {
    width: "90%"
  }
}))(FormControlLabel);
const icon = /* @__PURE__ */ jsx(CheckBoxOutlineBlankIcon, { fontSize: "small" });
const checkedIcon = /* @__PURE__ */ jsx(CheckBoxIcon, { fontSize: "small" });
function RenderOptionLabel(props) {
  const classes = useStyles();
  const { check, isSelected } = props;
  return /* @__PURE__ */ jsx(Box, { className: classes.fullWidth, children: /* @__PURE__ */ jsx(
    FixedWidthFormControlLabel,
    {
      className: classes.fullWidth,
      control: /* @__PURE__ */ jsx(
        Checkbox,
        {
          icon,
          checkedIcon,
          checked: isSelected
        }
      ),
      onClick: (event) => event.preventDefault(),
      label: /* @__PURE__ */ jsx(Tooltip, { title: check.id, children: /* @__PURE__ */ jsx(Box, { display: "flex", alignItems: "center", children: /* @__PURE__ */ jsx(Box, { className: classes.boxLabel, children: /* @__PURE__ */ jsx(Typography, { noWrap: true, children: check.name }) }) }) })
    }
  ) });
}
const withResultsOptions = [
  { label: "Yes", value: true },
  { label: "No", value: false }
];
const Filters = (props) => {
  const { checksChanged, withResultsChanged } = props;
  const api = useApi(techInsightsApiRef);
  const { value, loading, error } = useAsync(async () => {
    return api.getAllChecks();
  }, [api]);
  if (error) {
    return /* @__PURE__ */ jsx(ErrorPanel, { error });
  }
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(Box, { pb: 1, pt: 1, children: /* @__PURE__ */ jsxs(Typography, { variant: "button", component: "label", children: [
      "Checks",
      /* @__PURE__ */ jsx(
        Autocomplete,
        {
          multiple: true,
          disableCloseOnSelect: true,
          options: value ?? [],
          loading,
          getOptionLabel: (o) => o.name,
          onChange: (_, changedChecks) => {
            checksChanged(changedChecks);
          },
          filterOptions: (x) => x,
          renderOption: (check, { selected }) => {
            return /* @__PURE__ */ jsx(RenderOptionLabel, { check, isSelected: selected });
          },
          size: "small",
          popupIcon: /* @__PURE__ */ jsx(ExpandMoreIcon, {}),
          renderInput: (params) => /* @__PURE__ */ jsx(TextField, { ...params, variant: "outlined" })
        }
      )
    ] }) }),
    /* @__PURE__ */ jsx(Box, { pb: 1, pt: 1, children: /* @__PURE__ */ jsxs(Typography, { variant: "button", component: "label", children: [
      "Only with results",
      /* @__PURE__ */ jsx(
        Autocomplete,
        {
          defaultValue: withResultsOptions[0],
          options: withResultsOptions,
          getOptionLabel: (o) => o.label,
          onChange: (_, selectedItem) => {
            if (selectedItem) {
              withResultsChanged(selectedItem.value);
            }
          },
          disableClearable: true,
          size: "small",
          popupIcon: /* @__PURE__ */ jsx(ExpandMoreIcon, {}),
          renderInput: (params) => /* @__PURE__ */ jsx(TextField, { ...params, variant: "outlined" })
        }
      )
    ] }) })
  ] });
};

export { Filters };
