// @backstage-community/plugin-tech-insights/dist/components/ScorecardsPage/ScorecardsPage.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { useState, useMemo } from 'react';
import { ErrorPanel, Page, Header, HeaderLabel, Content, Table } from '@backstage/core-components';
import { useApi } from '@backstage/core-plugin-api';
import useAsync from 'react-use/lib/useAsync';
import { EntityRefLink } from '@backstage/plugin-catalog-react';
import { ScorecardsList } from '../ScorecardsList/ScorecardsList.esm.js';
import Grid from '@material-ui/core/Grid';
import { Filters } from './Filters.esm.js';
import { ExportCsv } from '@material-table/exporters';
import { techInsightsApiRef } from '@backstage-community/plugin-tech-insights-react';
import { ScorecardsBadge } from '../ScorecardsBadge/ScorecardsBadge.esm.js';

const ScorecardsPage = (props) => {
  const api = useApi(techInsightsApiRef);
  const [filterSelectedChecks, setFilterSelectedChecks] = useState([]);
  const [filterWithResults, setFilterWithResults] = useState(true);
  const tableOptions = {
    exportAllData: true,
    exportMenu: [
      {
        label: "Export CSV",
        exportFunc: (cols, datas) => ExportCsv(cols, datas, "tech-insights")
      }
    ],
    pageSize: props.badge ? 15 : 5,
    pageSizeOptions: props.badge ? [15, 30, 100] : [5, 10, 20],
    padding: `${props.dense ? "dense" : "default"}`
  };
  const { value, loading, error } = useAsync(async () => {
    const checks = await api.getAllChecks();
    const result = await api.runBulkChecks([], filterSelectedChecks);
    return {
      checks,
      result: filterWithResults ? result.filter((response) => response.results.length > 0) : result.filter(
        (response) => (response.results?.length ?? 0) === 0 || response.results.every((r) => r.result === false)
      )
    };
  }, [api, filterSelectedChecks, filterWithResults]);
  const tableColumns = useMemo(() => {
    const columns = [
      {
        field: "entity",
        title: "Entity",
        render: (row) => /* @__PURE__ */ jsx(EntityRefLink, { entityRef: row.entity })
      },
      {
        field: "results",
        title: "Results",
        render: (row) => props.badge ? /* @__PURE__ */ jsx(ScorecardsBadge, { checkResults: row.results }) : /* @__PURE__ */ jsx(ScorecardsList, { checkResults: row.results, dense: props.dense }),
        export: false
      }
    ];
    (filterSelectedChecks.length === 0 ? value?.checks || [] : filterSelectedChecks).forEach((check) => {
      columns.push({
        field: check.id,
        title: check.name,
        customExport: (row) => `${row.results.filter(
          (result) => result && result.check && result.check.id === check.id
        )[0]?.result}`,
        hidden: true,
        export: true
      });
    });
    return columns;
  }, [props, value, filterSelectedChecks]);
  if (error) {
    return /* @__PURE__ */ jsx(ErrorPanel, { error });
  }
  return /* @__PURE__ */ jsxs(Page, { themeId: "tool", children: [
    /* @__PURE__ */ jsxs(Header, { title: "Tech insights", children: [
      /* @__PURE__ */ jsx(HeaderLabel, { label: "Entities", value: value?.result.length ?? 0 }),
      /* @__PURE__ */ jsx(HeaderLabel, { label: "Checks", value: value?.checks.length ?? 0 })
    ] }),
    /* @__PURE__ */ jsx(Content, { children: /* @__PURE__ */ jsxs(Grid, { container: true, children: [
      /* @__PURE__ */ jsx(Grid, { item: true, style: { width: "300px" }, children: /* @__PURE__ */ jsx(
        Filters,
        {
          checksChanged: (checks) => setFilterSelectedChecks(checks),
          withResultsChanged: (withResults) => setFilterWithResults(withResults)
        }
      ) }),
      /* @__PURE__ */ jsx(Grid, { item: true, xs: true, children: /* @__PURE__ */ jsx(
        Table,
        {
          columns: tableColumns,
          data: value?.result ?? [],
          isLoading: loading,
          options: tableOptions
        }
      ) })
    ] }) })
  ] });
};

export { ScorecardsPage };
