// @backstage/plugin-home/dist/components/StarredEntityListItem/StarredEntityListItem.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { entityRouteRef, EntityDisplayName, entityRouteParams } from '@backstage/plugin-catalog-react';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { Link } from 'react-router-dom';
import { useRouteRef } from '@backstage/core-plugin-api';
import { FavoriteToggle } from '@backstage/core-components';
import { makeStyles } from '@material-ui/core/styles';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const useStyles = makeStyles((theme) => ({
  listItem: {
    paddingBottom: theme.spacing(0),
    paddingTop: theme.spacing(0)
  },
  secondary: {
    textTransform: "uppercase"
  }
}));
const StarredEntityListItem = ({
  entity,
  onToggleStarredEntity,
  showKind
}) => {
  const classes = useStyles();
  const catalogEntityRoute = useRouteRef(entityRouteRef);
  const { t } = useTranslationRef(homeTranslationRef);
  let secondaryText = "";
  if (showKind) {
    secondaryText += entity.kind.toLocaleLowerCase("en-US");
  }
  if (entity.spec && "type" in entity.spec) {
    if (showKind) {
      secondaryText += " \u2014 ";
    }
    secondaryText += entity.spec.type.toLocaleLowerCase(
      "en-US"
    );
  }
  return /* @__PURE__ */ jsxs(
    ListItem,
    {
      dense: true,
      className: classes.listItem,
      component: Link,
      button: true,
      to: catalogEntityRoute(entityRouteParams(entity)),
      children: [
        /* @__PURE__ */ jsx(
          ListItemIcon,
          {
            onClick: (e) => {
              e.preventDefault();
            },
            children: /* @__PURE__ */ jsx(
              FavoriteToggle,
              {
                id: `remove-favorite-${entity.metadata.uid}`,
                title: t("starredEntityListItem.removeFavoriteEntityTitle"),
                isFavorite: true,
                onToggle: () => onToggleStarredEntity(entity)
              }
            )
          }
        ),
        /* @__PURE__ */ jsx(
          ListItemText,
          {
            primary: /* @__PURE__ */ jsx(EntityDisplayName, { hideIcon: true, entityRef: entity }),
            secondary: secondaryText
          }
        )
      ]
    }
  );
};

export { StarredEntityListItem };
