// @roadiehq/backstage-plugin-github-pull-requests/dist/components/PullRequestsStatsCard/components/TooltipContent.esm.js
import { jsxs } from 'react/jsx-runtime';
import { Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useTooltipContentStyles = makeStyles((theme) => ({
  container: {
    backgroundColor: theme.palette.background.paper,
    fontSize: "1rem",
    fontWeight: Number(theme.typography.fontWeightBold),
    borderRadius: "6px",
    boxShadow: theme.shadows[10]
  },
  additions: {
    color: theme.palette.success.main
  },
  deletions: {
    color: theme.palette.error.main
  }
}));
const TooltipContent = (props) => {
  const { additions, deletions } = props;
  const classes = useTooltipContentStyles();
  return /* @__PURE__ */ jsxs(Grid, { container: true, className: classes.container, children: [
    /* @__PURE__ */ jsxs(Grid, { item: true, className: classes.additions, children: [
      "+",
      additions
    ] }),
    /* @__PURE__ */ jsxs(Grid, { item: true, className: classes.deletions, children: [
      "-",
      deletions
    ] })
  ] });
};

export { TooltipContent };
