// @backstage-community/plugin-tech-insights-backend/dist/service/fact/FactRetrieverRegistry.cjs.js
'use strict';

var errors = require('@backstage/errors');

class DefaultFactRetrieverRegistry {
  retrievers = /* @__PURE__ */ new Map();
  constructor(retrievers) {
    retrievers.forEach((r) => {
      this.registerSync(r);
    });
  }
  registerSync(registration) {
    if (this.retrievers.has(registration.factRetriever.id)) {
      throw new errors.ConflictError(
        `Tech insight fact retriever with identifier '${registration.factRetriever.id}' has already been registered`
      );
    }
    this.retrievers.set(registration.factRetriever.id, registration);
  }
  async register(registration) {
    this.registerSync(registration);
  }
  async get(retrieverReference) {
    const registration = this.retrievers.get(retrieverReference);
    if (!registration) {
      throw new errors.NotFoundError(
        `Tech insight fact retriever with identifier '${retrieverReference}' is not registered.`
      );
    }
    return registration;
  }
  async listRetrievers() {
    return [...this.retrievers.values()].map((it) => it.factRetriever);
  }
  async listRegistrations() {
    return [...this.retrievers.values()];
  }
  async getSchemas() {
    const retrievers = await this.listRetrievers();
    return retrievers.map((it) => it.schema);
  }
}

exports.DefaultFactRetrieverRegistry = DefaultFactRetrieverRegistry;
