// @backstage/ui/dist/components/Avatar/Avatar.esm.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef, useState, useEffect } from 'react';
import clsx from 'clsx';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { AvatarDefinition } from './definition.esm.js';
import styles from './Avatar.module.css.esm.js';

const Avatar = forwardRef((props, ref) => {
  const { classNames, dataAttributes, cleanedProps } = useStyles(
    AvatarDefinition,
    {
      size: "medium",
      purpose: "informative",
      ...props
    }
  );
  const { className, src, name, purpose, ...rest } = cleanedProps;
  const [imageStatus, setImageStatus] = useState("loading");
  useEffect(() => {
    setImageStatus("loading");
    const img = new Image();
    img.onload = () => setImageStatus("loaded");
    img.onerror = () => setImageStatus("error");
    img.src = src;
    return () => {
      img.onload = null;
      img.onerror = null;
    };
  }, [src]);
  const initialsCount = ["x-small", "small"].includes(cleanedProps.size) ? 1 : 2;
  const initials = name.split(" ").map((word) => word[0]).join("").toLocaleUpperCase("en-US").slice(0, initialsCount);
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      role: "img",
      "aria-label": purpose === "informative" ? name : void 0,
      "aria-hidden": purpose === "decoration" ? true : void 0,
      className: clsx(classNames.root, styles[classNames.root], className),
      ...dataAttributes,
      ...rest,
      children: imageStatus === "loaded" ? /* @__PURE__ */ jsx(
        "img",
        {
          src,
          alt: "",
          className: clsx(classNames.image, styles[classNames.image])
        }
      ) : /* @__PURE__ */ jsx(
        "div",
        {
          "aria-hidden": "true",
          className: clsx(classNames.fallback, styles[classNames.fallback]),
          children: initials
        }
      )
    }
  );
});
Avatar.displayName = "Avatar";

export { Avatar };
