// @backstage/plugin-catalog-backend/dist/processing/evictEntitiesFromOrphanedProviders.cjs.js
'use strict';

async function getOrphanedEntityProviderNames({
  db,
  providers,
  logger
}) {
  const dbProviderNames = await db.transaction(
    async (tx) => db.listReferenceSourceKeys(tx)
  );
  const providerNames = providers.map((p) => p.getProviderName());
  const orphaned = dbProviderNames.filter(
    (dbProviderName) => !providerNames.includes(dbProviderName)
  );
  if (orphaned.length) {
    const dbProviderNamesString = dbProviderNames.map((p) => `'${p}'`).join(", ");
    const providerNamesString = providerNames.map((p) => `'${p}'`).join(", ");
    const orphanedString = orphaned.map((p) => `'${p}'`).join(", ");
    logger.warn(`Found ${orphaned.length} orphaned entity provider(s)`);
    logger.warn(`Database contained providers: ${dbProviderNamesString}`);
    logger.warn(`Installed providers were: ${providerNamesString}`);
    logger.warn(`Orphaned providers were thus: ${orphanedString}`);
  }
  return orphaned;
}
async function removeEntitiesForProvider({
  db,
  providerName,
  logger
}) {
  try {
    await db.transaction(async (tx) => {
      await db.replaceUnprocessedEntities(tx, {
        sourceKey: providerName,
        type: "full",
        items: []
      });
    });
    logger.info(`Removed entities for orphaned provider ${providerName}`);
  } catch (e) {
    logger.error(
      `Failed to remove entities for orphaned provider ${providerName}`,
      e
    );
  }
}
async function evictEntitiesFromOrphanedProviders(options) {
  for (const providerName of await getOrphanedEntityProviderNames(options)) {
    await removeEntitiesForProvider({
      db: options.db,
      providerName,
      logger: options.logger
    });
  }
}

exports.evictEntitiesFromOrphanedProviders = evictEntitiesFromOrphanedProviders;
exports.getOrphanedEntityProviderNames = getOrphanedEntityProviderNames;
