// @roadiehq/backstage-plugin-github-pull-requests/dist/components/PullRequestsListView/PullRequestsListView.esm.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { Grid, Typography, Box, Link } from '@material-ui/core';
import { getStatusIconType, CommentIcon } from '../Icons.esm.js';
import { makeStyles } from '@material-ui/core/styles';
import Skeleton from '@material-ui/lab/Skeleton';

const useStyles = makeStyles((theme) => ({
  container: {
    backgroundColor: theme.palette.background.default,
    border: "1px solid grey",
    borderRadius: "6px",
    overflowY: "auto",
    maxHeight: "100%"
  },
  pullRequestRow: {
    paddingTop: "0.5rem",
    paddingBottom: "0.5rem",
    borderBottom: "1px solid grey",
    "&:hover": {
      backgroundColor: theme.palette.action.hover
    },
    "&:last-child": {
      borderBottom: "none"
    }
  },
  title: {
    fontWeight: Number(theme.typography.fontWeightBold)
  },
  link: {
    color: theme.palette.text.primary,
    "&:hover": {
      color: "#539bf5",
      fill: "#539bf5"
    },
    "&:first-child": {
      paddingRight: "4px"
    }
  },
  secondaryText: {
    color: theme.palette.text.secondary
  },
  middleColumn: {
    paddingRight: "0.5rem",
    paddingLeft: "0.5rem"
  }
}));
const SkeletonPullRequestItem = () => {
  const classes = useStyles();
  return /* @__PURE__ */ jsxs(Grid, { container: true, item: true, spacing: 0, className: classes.pullRequestRow, xs: 12, children: [
    /* @__PURE__ */ jsx(Grid, { item: true, xs: "auto", children: /* @__PURE__ */ jsx(
      Skeleton,
      {
        variant: "circle",
        width: 18,
        height: 18,
        style: { marginTop: "3px" }
      }
    ) }),
    /* @__PURE__ */ jsxs(Grid, { item: true, xs: 10, className: classes.middleColumn, children: [
      /* @__PURE__ */ jsx(Typography, { variant: "body1", children: /* @__PURE__ */ jsx(Skeleton, { variant: "text" }) }),
      /* @__PURE__ */ jsx(Typography, { variant: "caption", children: /* @__PURE__ */ jsx(Skeleton, { variant: "text", width: 130 }) })
    ] }),
    /* @__PURE__ */ jsx(
      Grid,
      {
        item: true,
        xs: 1,
        style: { flexShrink: 0, marginLeft: "auto" },
        className: classes.secondaryText,
        children: /* @__PURE__ */ jsx(Box, { display: "flex", justifyContent: "flex-end", children: /* @__PURE__ */ jsx(
          Skeleton,
          {
            variant: "circle",
            width: 18,
            height: 18,
            style: { marginTop: "3px" }
          }
        ) })
      }
    )
  ] });
};
const PullRequestItem = (props) => {
  const { pr } = props;
  const classes = useStyles();
  return /* @__PURE__ */ jsxs(Grid, { container: true, item: true, spacing: 0, className: classes.pullRequestRow, xs: 12, children: [
    /* @__PURE__ */ jsx(Grid, { item: true, style: { paddingLeft: "1rem", flexShrink: 0 }, xs: "auto", children: getStatusIconType(pr) }),
    /* @__PURE__ */ jsxs(Grid, { item: true, xs: 10, className: classes.middleColumn, children: [
      /* @__PURE__ */ jsx(Typography, { variant: "body1", noWrap: true, className: classes.title, children: /* @__PURE__ */ jsxs(Fragment, { children: [
        /* @__PURE__ */ jsx(
          Link,
          {
            className: `${classes.secondaryText} ${classes.link}`,
            target: "_blank",
            rel: "noopener noreferrer",
            underline: "none",
            href: pr.htmlUrl,
            children: new URL(pr.repositoryUrl).pathname.split("/").filter(Boolean).slice(1, 3).join("/")
          }
        ),
        /* @__PURE__ */ jsx(
          Link,
          {
            className: classes.link,
            target: "_blank",
            rel: "noopener noreferrer",
            underline: "none",
            href: pr.pullRequest.htmlUrl,
            children: pr.title
          }
        )
      ] }) }),
      /* @__PURE__ */ jsxs(Typography, { variant: "caption", className: classes.secondaryText, children: [
        "#",
        pr.number,
        " opened by",
        " ",
        /* @__PURE__ */ jsx(
          Link,
          {
            className: `${classes.secondaryText} ${classes.link}`,
            target: "_blank",
            rel: "noopener noreferrer",
            underline: "none",
            href: pr.user.htmlUrl,
            children: pr.user.login
          }
        )
      ] }),
      /* @__PURE__ */ jsx(Typography, { variant: "caption", className: classes.secondaryText, children: pr.pullRequest.created_at })
    ] }),
    /* @__PURE__ */ jsx(
      Grid,
      {
        item: true,
        justifyContent: "flex-end",
        container: true,
        spacing: 1,
        xs: 1,
        style: { flexShrink: 0, marginLeft: "auto" },
        className: classes.secondaryText,
        children: /* @__PURE__ */ jsx(Grid, { item: true, children: /* @__PURE__ */ jsxs(
          Link,
          {
            className: `${classes.secondaryText} ${classes.link}`,
            target: "_blank",
            rel: "noopener noreferrer",
            underline: "none",
            href: pr.htmlUrl,
            children: [
              /* @__PURE__ */ jsx(CommentIcon, {}),
              " ",
              pr.comments
            ]
          }
        ) })
      }
    )
  ] });
};
const PullRequestsListView = (props) => {
  const { data, emptyStateText } = props;
  const classes = useStyles();
  if (!data || data.length < 1) {
    return /* @__PURE__ */ jsx(
      Grid,
      {
        container: true,
        className: classes.container,
        spacing: 0,
        direction: "column",
        alignItems: "center",
        justifyContent: "center",
        children: /* @__PURE__ */ jsx(Grid, { item: true, md: 12, children: /* @__PURE__ */ jsx(Typography, { className: classes.secondaryText, variant: "h6", children: emptyStateText }) })
      }
    );
  }
  return /* @__PURE__ */ jsx(Grid, { container: true, className: classes.container, spacing: 1, children: data.map((pr) => /* @__PURE__ */ jsx(PullRequestItem, { pr }, pr.id)) });
};
const SkeletonPullRequestsListView = () => {
  const classes = useStyles();
  return /* @__PURE__ */ jsxs(Grid, { container: true, className: classes.container, spacing: 1, children: [
    /* @__PURE__ */ jsx(SkeletonPullRequestItem, {}),
    /* @__PURE__ */ jsx(SkeletonPullRequestItem, {}),
    /* @__PURE__ */ jsx(SkeletonPullRequestItem, {})
  ] });
};

export { PullRequestsListView, SkeletonPullRequestItem, SkeletonPullRequestsListView };
