// @backstage-community/plugin-tech-insights-backend/dist/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var plugin = require('./plugin/plugin.cjs.js');
var createFactRetriever = require('./service/fact/createFactRetriever.cjs.js');
var entityMetadataFactRetriever = require('./service/fact/factRetrievers/entityMetadataFactRetriever.cjs.js');
var entityOwnershipFactRetriever = require('./service/fact/factRetrievers/entityOwnershipFactRetriever.cjs.js');
var techdocsFactRetriever = require('./service/fact/factRetrievers/techdocsFactRetriever.cjs.js');
var persistenceContext = require('./service/persistence/persistenceContext.cjs.js');
var router = require('./service/router.cjs.js');
var techInsightsContextBuilder = require('./service/techInsightsContextBuilder.cjs.js');



exports.default = plugin.techInsightsPlugin;
exports.createFactRetrieverRegistration = createFactRetriever.createFactRetrieverRegistration;
exports.entityMetadataFactRetriever = entityMetadataFactRetriever.entityMetadataFactRetriever;
exports.entityOwnershipFactRetriever = entityOwnershipFactRetriever.entityOwnershipFactRetriever;
exports.techdocsFactRetriever = techdocsFactRetriever.techdocsFactRetriever;
exports.initializePersistenceContext = persistenceContext.initializePersistenceContext;
exports.createRouter = router.createRouter;
exports.buildTechInsightsContext = techInsightsContextBuilder.buildTechInsightsContext;
