// @backstage/plugin-kubernetes/dist/alpha/apis.esm.js
import { ApiBlueprint, fetchApiRef, discoveryApiRef } from '@backstage/frontend-plugin-api';
import { KubernetesBackendClient, kubernetesAuthProvidersApiRef, kubernetesApiRef, KubernetesProxyClient, kubernetesProxyApiRef, KubernetesAuthProviders, kubernetesClusterLinkFormatterApiRef, getDefaultFormatters, KubernetesClusterLinkFormatter, DEFAULT_FORMATTER_NAME } from '@backstage/plugin-kubernetes-react';
import { oneloginAuthApiRef, oktaAuthApiRef, microsoftAuthApiRef, googleAuthApiRef, gitlabAuthApiRef } from '@backstage/core-plugin-api';

const kubernetesApiExtension = ApiBlueprint.make({
  params: (defineParams) => defineParams({
    api: kubernetesApiRef,
    deps: {
      discoveryApi: discoveryApiRef,
      fetchApi: fetchApiRef,
      kubernetesAuthProvidersApi: kubernetesAuthProvidersApiRef
    },
    factory: ({ discoveryApi, fetchApi, kubernetesAuthProvidersApi: kubernetesAuthProvidersApi2 }) => new KubernetesBackendClient({
      discoveryApi,
      fetchApi,
      kubernetesAuthProvidersApi: kubernetesAuthProvidersApi2
    })
  })
});
const kubernetesProxyApi = ApiBlueprint.make({
  name: "proxy",
  params: (defineParams) => defineParams({
    api: kubernetesProxyApiRef,
    deps: {
      kubernetesApi: kubernetesApiRef
    },
    factory: ({ kubernetesApi }) => new KubernetesProxyClient({
      kubernetesApi
    })
  })
});
const kubernetesAuthProvidersApi = ApiBlueprint.make({
  name: "auth-providers",
  params: (defineParams) => defineParams({
    api: kubernetesAuthProvidersApiRef,
    deps: {
      gitlabAuthApi: gitlabAuthApiRef,
      googleAuthApi: googleAuthApiRef,
      microsoftAuthApi: microsoftAuthApiRef,
      oktaAuthApi: oktaAuthApiRef,
      oneloginAuthApi: oneloginAuthApiRef
    },
    factory: ({
      gitlabAuthApi,
      googleAuthApi,
      microsoftAuthApi,
      oktaAuthApi,
      oneloginAuthApi
    }) => {
      const oidcProviders = {
        gitlab: gitlabAuthApi,
        google: googleAuthApi,
        microsoft: microsoftAuthApi,
        okta: oktaAuthApi,
        onelogin: oneloginAuthApi
      };
      return new KubernetesAuthProviders({
        microsoftAuthApi,
        googleAuthApi,
        oidcProviders
      });
    }
  })
});
const kubernetesClusterLinkFormatterApi = ApiBlueprint.make({
  name: "cluster-link-formatter",
  params: (defineParams) => defineParams({
    api: kubernetesClusterLinkFormatterApiRef,
    deps: { googleAuthApi: googleAuthApiRef },
    factory: (deps) => {
      const formatters = getDefaultFormatters(deps);
      return new KubernetesClusterLinkFormatter({
        formatters,
        defaultFormatterName: DEFAULT_FORMATTER_NAME
      });
    }
  })
});

export { kubernetesApiExtension, kubernetesAuthProvidersApi, kubernetesClusterLinkFormatterApi, kubernetesProxyApi };
