// @backstage-community/plugin-sonarqube-backend/dist/plugin.cjs.js
'use strict';

var backendPluginApi = require('@backstage/backend-plugin-api');
var sonarqubeInfoProvider = require('./service/sonarqubeInfoProvider.cjs.js');
var router = require('./service/router.cjs.js');

const sonarqubePlugin = backendPluginApi.createBackendPlugin({
  pluginId: "sonarqube",
  register(env) {
    env.registerInit({
      deps: {
        logger: backendPluginApi.coreServices.logger,
        config: backendPluginApi.coreServices.rootConfig,
        httpRouter: backendPluginApi.coreServices.httpRouter
      },
      async init({ logger, config, httpRouter }) {
        httpRouter.use(
          await router.createRouter({
            /**
             * Logger for logging purposes
             */
            logger,
            /**
             * Info provider to be able to get all necessary information for the APIs
             */
            sonarqubeInfoProvider: sonarqubeInfoProvider.DefaultSonarqubeInfoProvider.fromConfig(config)
          })
        );
      }
    });
  }
});

exports.sonarqubePlugin = sonarqubePlugin;
