// @backstage-community/plugin-tech-insights/dist/components/ScorecardsInfo/ScorecardInfo.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Alert from '@material-ui/lab/Alert';
import { ScorecardsList } from '../ScorecardsList/ScorecardsList.esm.js';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Accordion from '@material-ui/core/Accordion';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import { useApi } from '@backstage/core-plugin-api';
import { MarkdownContent } from '@backstage/core-components';
import { techInsightsApiRef } from '@backstage-community/plugin-tech-insights-react';

const useStyles = makeStyles((theme) => ({
  subheader: {
    paddingLeft: theme.spacing(0.5)
  },
  accordionHeader: {
    borderBottom: `1px solid ${theme.palette.border}`
  },
  accordionHeaderContent: {
    margin: `${theme.spacing(2)}px 0 !important`
  },
  accordionDetails: {
    margin: 0,
    paddingBottom: 0
  }
}));
const infoAccordion = (title, children, classes, expanded, summary) => /* @__PURE__ */ jsxs(Accordion, { defaultExpanded: expanded, children: [
  /* @__PURE__ */ jsx(
    AccordionSummary,
    {
      expandIcon: /* @__PURE__ */ jsx(ExpandMoreIcon, {}),
      className: classes.accordionHeader,
      classes: {
        content: classes.accordionHeaderContent
      },
      children: /* @__PURE__ */ jsxs(Grid, { container: true, justifyContent: "space-between", alignItems: "center", children: [
        /* @__PURE__ */ jsx(Grid, { item: true, children: /* @__PURE__ */ jsx(Typography, { variant: "h5", children: title }) }),
        /* @__PURE__ */ jsx(Grid, { item: true, children: /* @__PURE__ */ jsx(Typography, { children: summary }) })
      ] })
    }
  ),
  /* @__PURE__ */ jsx(AccordionDetails, { classes: { root: classes.accordionDetails }, children })
] });
const infoDetails = (description, classes, element) => {
  return /* @__PURE__ */ jsxs(Grid, { container: true, children: [
    description && /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, children: /* @__PURE__ */ jsx(
      Typography,
      {
        className: classes.subheader,
        variant: "body1",
        gutterBottom: true,
        children: /* @__PURE__ */ jsx(MarkdownContent, { content: description })
      }
    ) }),
    /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, children: element })
  ] });
};
const infoCard = (title, description, classes, element, expanded, summary) => /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, children: infoAccordion(
  title,
  infoDetails(description, classes, element),
  classes,
  expanded,
  summary
) });
const ScorecardInfo = (props) => {
  const {
    checkResults,
    title,
    entity,
    description,
    noWarning,
    expanded = true,
    dense,
    hideDescription = dense
  } = props;
  const classes = useStyles();
  const api = useApi(techInsightsApiRef);
  if (!checkResults.length) {
    if (noWarning) {
      return infoCard(
        title,
        description,
        classes,
        /* @__PURE__ */ jsx(Alert, { severity: "info", children: "All checks passed, or no checks have been performed yet" }),
        expanded
      );
    }
    return infoCard(
      title,
      description,
      classes,
      /* @__PURE__ */ jsx(Alert, { severity: "warning", children: "No checks have any data yet." }),
      expanded
    );
  }
  return infoCard(
    title,
    description,
    classes,
    /* @__PURE__ */ jsx(
      ScorecardsList,
      {
        checkResults,
        entity,
        dense,
        hideDescription
      }
    ),
    expanded,
    `${checkResults.filter((checkResult) => !api.isCheckResultFailed(checkResult)).length}/${checkResults.length}`
  );
};

export { ScorecardInfo };
