// @backstage-community/plugin-tech-insights-backend/dist/service/fact/factRetrievers/entityOwnershipFactRetriever.cjs.js
'use strict';

var catalogClient = require('@backstage/catalog-client');
var luxon = require('luxon');

const entityOwnershipFactRetriever = {
  id: "entityOwnershipFactRetriever",
  version: "0.0.1",
  title: "Entity Ownership",
  description: "Generates facts which indicate the quality of data in the spec.owner field",
  entityFilter: [
    { kind: ["component", "domain", "system", "api", "resource", "template"] }
  ],
  schema: {
    hasOwner: {
      type: "boolean",
      description: "The spec.owner field is set"
    },
    hasGroupOwner: {
      type: "boolean",
      description: "The spec.owner field is set and refers to a group"
    }
  },
  handler: async ({ discovery, entityFilter, auth }) => {
    const { token } = await auth.getPluginRequestToken({
      onBehalfOf: await auth.getOwnServiceCredentials(),
      targetPluginId: "catalog"
    });
    const catalogClient$1 = new catalogClient.CatalogClient({
      discoveryApi: discovery
    });
    const entities = await catalogClient$1.getEntities(
      { filter: entityFilter },
      { token }
    );
    return entities.items.map((entity) => {
      return {
        entity: {
          namespace: entity.metadata.namespace,
          kind: entity.kind,
          name: entity.metadata.name
        },
        facts: {
          hasOwner: Boolean(entity.spec?.owner),
          hasGroupOwner: Boolean(
            entity.spec?.owner && !(entity.spec?.owner).startsWith("user:")
          )
        },
        timestamp: luxon.DateTime.now()
      };
    });
  }
};

exports.entityOwnershipFactRetriever = entityOwnershipFactRetriever;
