// @backstage-community/plugin-tech-insights-backend/dist/service/fact/factRetrievers/techdocsFactRetriever.cjs.js
'use strict';

var catalogClient = require('@backstage/catalog-client');
var utils = require('./utils.cjs.js');
var luxon = require('luxon');

const techdocsAnnotation = "backstage.io/techdocs-ref";
const techdocsEntityAnnotation = "backstage.io/techdocs-entity";
const techdocsAnnotationFactName = utils.generateAnnotationFactName(techdocsAnnotation);
const techdocsEntityAnnotationFactName = utils.generateAnnotationFactName(
  techdocsEntityAnnotation
);
const techdocsFactRetriever = {
  id: "techdocsFactRetriever",
  version: "0.1.0",
  title: "Tech Docs",
  description: "Generates facts related to the completeness of techdocs configuration for entities",
  schema: {
    [techdocsAnnotationFactName]: {
      type: "boolean",
      description: "The entity has a TechDocs reference annotation"
    },
    [techdocsEntityAnnotationFactName]: {
      type: "boolean",
      description: "The entity has a TechDocs entity annotation"
    }
  },
  handler: async ({ discovery, entityFilter, auth }) => {
    const { token } = await auth.getPluginRequestToken({
      onBehalfOf: await auth.getOwnServiceCredentials(),
      targetPluginId: "catalog"
    });
    const catalogClient$1 = new catalogClient.CatalogClient({
      discoveryApi: discovery
    });
    const entities = await catalogClient$1.getEntities(
      { filter: entityFilter },
      { token }
    );
    return entities.items.map((entity) => {
      return {
        entity: {
          namespace: entity.metadata.namespace,
          kind: entity.kind,
          name: entity.metadata.name
        },
        facts: {
          [techdocsAnnotationFactName]: utils.entityHasAnnotation(
            entity,
            techdocsAnnotation
          ),
          [techdocsEntityAnnotationFactName]: utils.entityHasAnnotation(
            entity,
            techdocsEntityAnnotation
          )
        },
        timestamp: luxon.DateTime.now()
      };
    });
  }
};

exports.techdocsFactRetriever = techdocsFactRetriever;
