// @backstage/core-components/dist/components/DependencyGraph/types.esm.js
var DependencyGraphTypes;
((DependencyGraphTypes2) => {
  DependencyGraphTypes2.Direction = {
    /**
     * Top to Bottom
     */
    TOP_BOTTOM: "TB",
    /**
     * Bottom to Top
     */
    BOTTOM_TOP: "BT",
    /**
     * Left to Right
     */
    LEFT_RIGHT: "LR",
    /**
     * Right to Left
     */
    RIGHT_LEFT: "RL"
  };
  DependencyGraphTypes2.Alignment = {
    /**
     * Up Left
     */
    UP_LEFT: "UL",
    /**
     * Up Right
     */
    UP_RIGHT: "UR",
    /**
     * Down Left
     */
    DOWN_LEFT: "DL",
    /**
     * Down Right
     */
    DOWN_RIGHT: "DR"
  };
  DependencyGraphTypes2.Ranker = {
    /**
     * {@link https://en.wikipedia.org/wiki/Network_simplex_algorithm | Network Simplex} algorithm
     */
    NETWORK_SIMPLEX: "network-simplex",
    /**
     * Tight Tree algorithm
     */
    TIGHT_TREE: "tight-tree",
    /**
     * Longest path algorithm
     *
     * @remarks
     *
     * Simplest and fastest
     */
    LONGEST_PATH: "longest-path"
  };
  DependencyGraphTypes2.LabelPosition = {
    LEFT: "l",
    RIGHT: "r",
    CENTER: "c"
  };
})(DependencyGraphTypes || (DependencyGraphTypes = {}));

export { DependencyGraphTypes };
