// @backstage/plugin-home/dist/homePageComponents/FeaturedDocsCard/Content.esm.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import useAsync from 'react-use/esm/useAsync';
import { Progress, ErrorPanel, Link, EmptyState, LinkButton } from '@backstage/core-components';
import { catalogApiRef, EntityDisplayName } from '@backstage/plugin-catalog-react';
import { useApi } from '@backstage/core-plugin-api';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import { stringifyEntityRef } from '@backstage/catalog-model';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const useStyles = makeStyles(
  (theme) => ({
    docDescription: {
      fontSize: theme.typography.body1.fontSize,
      fontWeight: theme.typography.body1.fontWeight,
      marginBottom: theme.spacing(2),
      marginTop: theme.spacing(2)
    },
    docSubLink: {
      fontSize: theme.typography.subtitle1.fontSize,
      fontWeight: theme.typography.subtitle1.fontWeight,
      lineHeight: theme.typography.subtitle1.lineHeight
    },
    docsTitleLink: {
      fontSize: theme.typography.h6.fontSize,
      fontWeight: theme.typography.h6.fontWeight,
      lineHeight: theme.typography.h6.lineHeight
    }
  }),
  { name: "HomeFeaturedDocsCard" }
);
const Content = (props) => {
  const { emptyState, filter, linkDestination, responseLimit, subLinkText } = props;
  const { t } = useTranslationRef(homeTranslationRef);
  const linkText = subLinkText || t("featuredDocsCard.learnMoreTitle");
  const styles = useStyles();
  const catalogApi = useApi(catalogApiRef);
  const {
    value: entities,
    loading,
    error
  } = useAsync(async () => {
    const response = await catalogApi.getEntities({
      filter,
      limit: responseLimit || 10
    });
    return response.items;
  });
  if (loading) {
    return /* @__PURE__ */ jsx(Progress, {});
  }
  if (error) {
    return /* @__PURE__ */ jsx(ErrorPanel, { error });
  }
  return entities?.length ? /* @__PURE__ */ jsx(Fragment, { children: entities.map((d) => /* @__PURE__ */ jsxs(
    "div",
    {
      "data-testid": "docs-card-content",
      children: [
        /* @__PURE__ */ jsx(
          Link,
          {
            className: styles.docsTitleLink,
            "data-testid": "docs-card-title",
            to: linkDestination || `/docs/${d.metadata.namespace || "default"}/${d.kind}/${d.metadata.name}/`,
            children: /* @__PURE__ */ jsx(EntityDisplayName, { entityRef: stringifyEntityRef(d) })
          }
        ),
        d.metadata.description && /* @__PURE__ */ jsx(Typography, { className: styles.docDescription, children: d.metadata.description }),
        /* @__PURE__ */ jsx(
          Link,
          {
            className: styles.docSubLink,
            "data-testid": "docs-card-sub-link",
            to: linkDestination || `/docs/${d.metadata.namespace || "default"}/${d.kind}/${d.metadata.name}/`,
            children: linkText
          }
        )
      ]
    },
    `${d.metadata.name}-${d.kind}-${d.metadata.namespace}`
  )) }) : emptyState || /* @__PURE__ */ jsx(
    EmptyState,
    {
      missing: "data",
      title: t("featuredDocsCard.empty.title"),
      description: t("featuredDocsCard.empty.description"),
      action: /* @__PURE__ */ jsx(
        LinkButton,
        {
          to: "https://backstage.io/docs/features/techdocs/getting-started",
          variant: "contained",
          children: t("featuredDocsCard.empty.learnMoreLinkTitle")
        }
      )
    }
  );
};

export { Content };
