// @roadiehq/backstage-plugin-github-insights/dist/components/Widgets/ContributorsCard/components/ContributorTooltipContent/ContributorTooltipContent.esm.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { makeStyles, Grid, Avatar, Typography, Link, Box, Divider } from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';
import { Progress } from '@backstage/core-components';
import LocationOnIcon from '@material-ui/icons/LocationOn';
import { useEntityGithubScmIntegration } from '../../../../../hooks/useEntityGithubScmIntegration.esm.js';
import { useContributor } from '../../../../../hooks/useContributor.esm.js';
import { useEntity } from '@backstage/plugin-catalog-react';

const useStyles = makeStyles((theme) => ({
  contributorsTooltipContainer: {
    padding: theme.spacing(1, 0)
  }
}));
const ContributorTooltipContent = ({ contributorLogin }) => {
  const classes = useStyles();
  const { entity } = useEntity();
  const { hostname } = useEntityGithubScmIntegration(entity);
  const { contributor, loading } = useContributor(contributorLogin);
  if (loading) {
    return /* @__PURE__ */ jsx(Progress, {});
  } else if (!contributor?.login) {
    return /* @__PURE__ */ jsx(Alert, { severity: "error", children: "Fetching failed!" });
  }
  return /* @__PURE__ */ jsxs(Grid, { container: true, className: classes.contributorsTooltipContainer, children: [
    /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, sm: 2, children: /* @__PURE__ */ jsx(
      Avatar,
      {
        alt: contributor.login,
        src: contributor.avatar_url
      },
      contributor.login
    ) }),
    /* @__PURE__ */ jsxs(Grid, { item: true, xs: 12, sm: 10, children: [
      /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, children: /* @__PURE__ */ jsxs(Typography, { variant: "h6", children: [
        /* @__PURE__ */ jsx(
          Link,
          {
            href: `https://${hostname}/${contributor.login}`,
            color: "inherit",
            target: "_blank",
            rel: "noopener noreferrer",
            children: contributor.name
          }
        ),
        /* @__PURE__ */ jsx(Box, { component: "span", ml: 2, children: /* @__PURE__ */ jsx(Typography, { variant: "caption", children: contributor.login }) })
      ] }) }),
      contributor.bio && /* @__PURE__ */ jsx(Grid, { item: true, xs: 12, children: /* @__PURE__ */ jsx(Typography, { variant: "subtitle2", children: contributor.bio }) }),
      contributor.location && /* @__PURE__ */ jsxs(Grid, { item: true, xs: 12, children: [
        /* @__PURE__ */ jsx(Box, { my: 2, children: /* @__PURE__ */ jsx(Divider, {}) }),
        /* @__PURE__ */ jsxs(Typography, { variant: "caption", children: [
          /* @__PURE__ */ jsx(LocationOnIcon, { fontSize: "inherit" }),
          " ",
          contributor.location
        ] })
      ] })
    ] })
  ] });
};

export { ContributorTooltipContent as default };
