// @harnessio/backstage-plugin-ci-cd/dist/index.esm.js
import { createRouteRef, createSubRouteRef, createPlugin, createRoutableExtension, useApi, identityApiRef, discoveryApiRef, configApiRef } from '@backstage/core-plugin-api';
import React, { useState, useCallback, useEffect } from 'react';
import { Routes, Route } from 'react-router';
import { Box, Typography, Link, IconButton, CircularProgress, makeStyles, FormControl, MenuItem } from '@material-ui/core';
import FormHelperText from '@mui/material/FormHelperText';
import InputLabel from '@mui/material/InputLabel';
import ListSubheader from '@mui/material/ListSubheader';
import { Grid, Tooltip, Dialog, DialogTitle, DialogContent, DialogContentText, DialogActions, Button } from '@mui/material';
import { OverflowTooltip, Table, StatusWarning, StatusAborted, StatusOK, StatusError, StatusRunning, EmptyState, MissingAnnotationEmptyState } from '@backstage/core-components';
import Select from '@mui/material/Select';
import Swal from 'sweetalert2';
import { useEntity } from '@backstage/plugin-catalog-react';
import { match } from 'path-to-regexp';
import dayjs from 'dayjs';
import durationPlugin from 'dayjs/plugin/duration';
import relativeTimePlugin from 'dayjs/plugin/relativeTime';
import ReplayIcon from '@material-ui/icons/Replay';
import useAsyncRetry from 'react-use/lib/useAsyncRetry';

const harnessCiCdRouteRef = createRouteRef({
  id: "harness-ci-cd"
});
const harnessCIBuildRouteRef = createSubRouteRef({
  id: "harness-ci-cd/build",
  parent: harnessCiCdRouteRef,
  path: "/:buildId"
});

const rootRouteRef = createRouteRef({
  id: "harness-ci-cd"
});

const harnessCiCdPlugin = createPlugin({
  id: "harness-ci-cd",
  routes: {
    root: rootRouteRef
  }
});
const HarnessCiCdPage = harnessCiCdPlugin.provide(
  createRoutableExtension({
    name: "HarnessCiCdPage",
    component: () => Promise.resolve().then(function () { return Router$1; }).then((m) => m.Router),
    mountPoint: harnessCiCdRouteRef
  })
);
const EntityHarnessCiCdContent = harnessCiCdPlugin.provide(
  createRoutableExtension({
    name: "HarnnessCiCdContent",
    component: () => Promise.resolve().then(function () { return Router$1; }).then((m) => m.Router),
    mountPoint: harnessCiCdRouteRef
  })
);

const useProjectSlugFromEntity = (env, isNewAnnotationPresent, selectedPipelineUrl) => {
  var _a, _b, _c, _d;
  let accountId;
  let orgId;
  let projectId;
  let pipelineid;
  let serviceid;
  let url;
  let projectids;
  const { entity } = useEntity();
  if (isNewAnnotationPresent) {
    const containsModule = selectedPipelineUrl.includes("/module/");
    const containsAll = selectedPipelineUrl.includes("/all/");
    let pipelineUrlMatch;
    let serviceUrlMatch;
    if (containsModule) {
      pipelineUrlMatch = match(
        "(.*)/account/:accountId/module/:module/orgs/:orgId/projects/:projectId/pipelines/:pipelineId/(.*)",
        {
          decode: decodeURIComponent
        }
      );
      serviceUrlMatch = match(
        "(.*)/account/:accountId/module/:module/orgs/:orgId/projects/:projectId/services/:serviceId",
        {
          decode: decodeURIComponent
        }
      );
    } else if (containsAll) {
      pipelineUrlMatch = match(
        "(.*)/account/:accountId/all/orgs/:orgId/projects/:projectId/pipelines/:pipelineId/(.*)",
        {
          decode: decodeURIComponent
        }
      );
      serviceUrlMatch = match(
        "(.*)/account/:accountId/all/orgs/:orgId/projects/:projectId/services/:serviceId",
        {
          decode: decodeURIComponent
        }
      );
    } else {
      pipelineUrlMatch = match(
        "(.*)/account/:accountId/:module/orgs/:orgId/projects/:projectId/pipelines/:pipelineId/(.*)",
        {
          decode: decodeURIComponent
        }
      );
      serviceUrlMatch = match(
        "(.*)/account/:accountId/:module/orgs/:orgId/projects/:projectId/services/:serviceId",
        {
          decode: decodeURIComponent
        }
      );
    }
    const hostname2 = new URL(selectedPipelineUrl).hostname;
    const baseUrl12 = new URL(selectedPipelineUrl).origin;
    const envAB = hostname2.split(".harness.io")[0];
    const envFromUrl = envAB.includes("qa") ? "qa" : "prod";
    const urlParams2 = pipelineUrlMatch(selectedPipelineUrl);
    if (urlParams2) {
      return {
        orgId: urlParams2.params.orgId,
        accountId: urlParams2.params.accountId,
        pipelineId: urlParams2.params.pipelineId,
        urlParams: urlParams2,
        baseUrl1: baseUrl12,
        projectIds: urlParams2.params.projectId,
        envFromUrl
      };
    }
    const serviceUrlParams = serviceUrlMatch(
      selectedPipelineUrl.split("?")[0]
    );
    if (serviceUrlParams) {
      return {
        orgId: serviceUrlParams.params.orgId,
        accountId: serviceUrlParams.params.accountId,
        serviceId: serviceUrlParams.params.serviceId,
        urlParams: serviceUrlParams,
        baseUrl1: baseUrl12,
        projectIds: serviceUrlParams.params.projectId,
        envFromUrl
      };
    }
    return {
      error: "Invalid URL format"
    };
  }
  if (env && env !== "prod") {
    pipelineid = `harness.io/ci-pipelineIds-${env}`;
    serviceid = `harness.io/cd-serviceId-${env}`;
    url = `harness.io/project-url-${env}`;
    projectids = `harness.io/projects-${env}`;
  } else {
    pipelineid = "harness.io/ci-pipelineIds";
    serviceid = "harness.io/cd-serviceId";
    url = "harness.io/project-url";
    projectids = "harness.io/projects";
  }
  let projectIds = (_a = entity.metadata.annotations) == null ? void 0 : _a[projectids];
  const pipelineId = (_b = entity.metadata.annotations) == null ? void 0 : _b[pipelineid];
  const serviceId = (_c = entity.metadata.annotations) == null ? void 0 : _c[serviceid];
  const uRL = (_d = entity.metadata.annotations) == null ? void 0 : _d[url];
  const hostname = new URL(uRL != null ? uRL : "").hostname;
  const baseUrl1 = new URL(uRL != null ? uRL : "").origin;
  const urlMatch = match(
    "(.*)/account/:accountId/:module/orgs/:orgId/projects/:projectId/(.*)",
    {
      decode: decodeURIComponent
    }
  );
  const urlParams = urlMatch(uRL != null ? uRL : "");
  if (urlParams) {
    accountId = urlParams.params.accountId;
    orgId = urlParams.params.orgId;
    projectId = urlParams.params.projectId;
    if (!projectIds) {
      projectIds = projectId;
    }
  }
  return {
    orgId,
    accountId,
    pipelineId,
    serviceId,
    urlParams,
    hostname,
    baseUrl1,
    projectIds
  };
};

dayjs.extend(durationPlugin);
dayjs.extend(relativeTimePlugin);
function relativeTimeTo(time, withoutSuffix = false) {
  return dayjs().to(dayjs(time), withoutSuffix);
}
function durationHumanized(startTime, endTime) {
  return dayjs.duration(dayjs(startTime).diff(dayjs(endTime))).humanize();
}

function PrintCard(props) {
  const row = props.props;
  if (row.cdser === "") {
    return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Box, null, /* @__PURE__ */ React.createElement(
      Grid,
      {
        container: true,
        justifyContent: "center",
        alignItems: "flex-start",
        columns: 20
      },
      row.branch === "undefined" && row.sourcebranch === "undefined" && row.tag === "undefined" ? null : /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(
        Grid,
        {
          item: true,
          md: 2,
          alignItems: "center",
          sx: { paddingTop: "2px" }
        },
        /* @__PURE__ */ React.createElement(Tooltip, { title: "Continuous Integration", arrow: true }, /* @__PURE__ */ React.createElement(
          "svg",
          {
            viewBox: "0 0 32 32",
            xmlns: "http://www.w3.org/2000/svg",
            width: "25",
            height: "25"
          },
          /* @__PURE__ */ React.createElement(
            "path",
            {
              fillRule: "evenodd",
              clipRule: "evenodd",
              d: "M29 16a13.02 13.02 0 01-.854 4.6.387.387 0 01-.637.13l-3.99-3.94a.534.534 0 01-.085-.628c.432-.827.649-1.773.649-2.762.056-3.222-2.6-5.878-5.879-5.878-1.04 0-1.994.26-2.823.706a.53.53 0 01-.623-.08l-3.551-3.504a.48.48 0 01.154-.794A12.79 12.79 0 0116 3c7.178 0 13 5.822 13 13zM8.476 5.42a.582.582 0 01.746.072l4.035 4.035c.179.178.194.46.054.67a5.829 5.829 0 00-.985 3.26 5.934 5.934 0 005.935 5.934 5.829 5.829 0 003.26-.985.531.531 0 01.67.054l4.367 4.368a.582.582 0 01.065.756C24.225 26.86 20.4 29 16 29 8.822 29 3 23.178 3 16c0-4.403 2.144-8.231 5.476-10.58zm9.785 11.71a3.932 3.932 0 003.956-3.956 3.932 3.932 0 00-3.956-3.957 3.932 3.932 0 00-3.957 3.957 3.932 3.932 0 003.957 3.956z",
              fill: "url(#ci-main_svg__a)"
            }
          ),
          /* @__PURE__ */ React.createElement("defs", null, /* @__PURE__ */ React.createElement(
            "linearGradient",
            {
              id: "ci-main_svg__a",
              x1: "-5.398",
              y1: "11.398",
              x2: "11.398",
              y2: "37.398",
              gradientUnits: "userSpaceOnUse"
            },
            /* @__PURE__ */ React.createElement("stop", { stopColor: "#73DFE7" }),
            /* @__PURE__ */ React.createElement("stop", { offset: "1", stopColor: "#0095F7" })
          ))
        ))
      ), /* @__PURE__ */ React.createElement(Grid, { sx: { paddingLeft: "5px" }, item: true, md: 18 }, /* @__PURE__ */ React.createElement(PrintCI, { props: row })))
    )));
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Box, null, /* @__PURE__ */ React.createElement(
    Grid,
    {
      container: true,
      justifyContent: "center",
      alignItems: "flex-start",
      columns: 20
    },
    row.branch === "undefined" && row.sourcebranch === "undefined" && row.tag === "undefined" ? null : /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Grid, { item: true, md: 2, alignItems: "center", sx: { paddingTop: "2px" } }, /* @__PURE__ */ React.createElement(Tooltip, { title: "Continuous Integration", arrow: true }, /* @__PURE__ */ React.createElement(
      "svg",
      {
        viewBox: "0 0 32 32",
        xmlns: "http://www.w3.org/2000/svg",
        width: "25",
        height: "25"
      },
      /* @__PURE__ */ React.createElement(
        "path",
        {
          fillRule: "evenodd",
          clipRule: "evenodd",
          d: "M29 16a13.02 13.02 0 01-.854 4.6.387.387 0 01-.637.13l-3.99-3.94a.534.534 0 01-.085-.628c.432-.827.649-1.773.649-2.762.056-3.222-2.6-5.878-5.879-5.878-1.04 0-1.994.26-2.823.706a.53.53 0 01-.623-.08l-3.551-3.504a.48.48 0 01.154-.794A12.79 12.79 0 0116 3c7.178 0 13 5.822 13 13zM8.476 5.42a.582.582 0 01.746.072l4.035 4.035c.179.178.194.46.054.67a5.829 5.829 0 00-.985 3.26 5.934 5.934 0 005.935 5.934 5.829 5.829 0 003.26-.985.531.531 0 01.67.054l4.367 4.368a.582.582 0 01.065.756C24.225 26.86 20.4 29 16 29 8.822 29 3 23.178 3 16c0-4.403 2.144-8.231 5.476-10.58zm9.785 11.71a3.932 3.932 0 003.956-3.956 3.932 3.932 0 00-3.956-3.957 3.932 3.932 0 00-3.957 3.957 3.932 3.932 0 003.957 3.956z",
          fill: "url(#ci-main_svg__a)"
        }
      ),
      /* @__PURE__ */ React.createElement("defs", null, /* @__PURE__ */ React.createElement(
        "linearGradient",
        {
          id: "ci-main_svg__a",
          x1: "-5.398",
          y1: "11.398",
          x2: "11.398",
          y2: "37.398",
          gradientUnits: "userSpaceOnUse"
        },
        /* @__PURE__ */ React.createElement("stop", { stopColor: "#73DFE7" }),
        /* @__PURE__ */ React.createElement("stop", { offset: "1", stopColor: "#0095F7" })
      ))
    ))), /* @__PURE__ */ React.createElement(
      Grid,
      {
        sx: { paddingLeft: "5px", paddingBottom: "10px" },
        item: true,
        md: 18
      },
      /* @__PURE__ */ React.createElement(PrintCI, { props: row })
    )),
    /* @__PURE__ */ React.createElement(Grid, { item: true, md: 2, alignItems: "top", sx: { paddingTop: "10px" } }, /* @__PURE__ */ React.createElement(Tooltip, { title: "Continuous Delivery", arrow: true }, /* @__PURE__ */ React.createElement(
      "svg",
      {
        viewBox: "0 0 32 32",
        xmlns: "http://www.w3.org/2000/svg",
        width: "25",
        height: "25"
      },
      /* @__PURE__ */ React.createElement(
        "path",
        {
          fillRule: "evenodd",
          clipRule: "evenodd",
          d: "M26.427 7.06A5.073 5.073 0 0129 11.52l-.077 8.934a5.073 5.073 0 01-2.537 4.35l-8.002 4.618a5.073 5.073 0 01-5.035.021l-7.776-4.402A5.073 5.073 0 013 20.583l.077-8.935a5.073 5.073 0 012.537-4.35l8.002-4.619a5.073 5.073 0 015.035-.02l7.776 4.402zm-2.83 6.938c-.42 1.586-2.029 2.538-3.601 2.141l-7.489-2.026a4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.925 4.835 4.835 0 00.479 3.664 4.757 4.757 0 002.905 2.251 4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.924l.26-1.047a.25.25 0 00-.177-.302l-1.258-.34a.25.25 0 00-.307.18l-.256 1.025c-.426 1.602-2.064 2.556-3.651 2.127-1.582-.428-2.523-2.073-2.1-3.67.421-1.586 2.03-2.538 3.602-2.142l7.489 2.027a4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.925 4.835 4.835 0 00-.479-3.664 4.757 4.757 0 00-2.905-2.251 4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.924l-.247.828a.25.25 0 00.174.312l1.259.34a.25.25 0 00.304-.169l.248-.827c.426-1.602 2.064-2.557 3.651-2.127 1.582.428 2.523 2.073 2.1 3.67z",
          fill: "url(#cd-main_svg__a)"
        }
      ),
      /* @__PURE__ */ React.createElement("defs", null, /* @__PURE__ */ React.createElement(
        "linearGradient",
        {
          id: "cd-main_svg__a",
          x1: "6.774",
          y1: "2",
          x2: "21.5",
          y2: "30",
          gradientUnits: "userSpaceOnUse"
        },
        /* @__PURE__ */ React.createElement("stop", { stopColor: "#62F91F" }),
        /* @__PURE__ */ React.createElement("stop", { offset: "1", stopColor: "#45BD35" })
      ))
    ))),
    /* @__PURE__ */ React.createElement(Grid, { sx: { paddingLeft: "5px" }, item: true, md: 18 }, /* @__PURE__ */ React.createElement(PrintCD, { props: row }))
  )));
}
function PrintCI(props) {
  const row = props.props;
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(PrintBranch, { props: row }), /* @__PURE__ */ React.createElement(PrintCommit, { props: row }), /* @__PURE__ */ React.createElement(PrintPR, { props: row }));
}
function PrintBranch(props) {
  const row = props.props;
  if (row.tag !== "undefined") {
    return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("span", { style: { padding: "3px" } }, /* @__PURE__ */ React.createElement(
      "svg",
      {
        viewBox: "0 0 13 14",
        xmlns: "http://www.w3.org/2000/svg",
        width: "14",
        height: "14"
      },
      /* @__PURE__ */ React.createElement(
        "path",
        {
          d: "M12.438 2.452l-.006-.017a.367.367 0 00-.067-.114l-.004-.007L10.409.125a.374.374 0 00-.28-.125H2.327a.376.376 0 00-.28.125L.095 2.314l-.005.008a.374.374 0 00-.066.113l-.006.017a.37.37 0 00-.018.11v11.064c0 .206.168.374.374.374h11.707a.375.375 0 00.375-.374V2.563a.369.369 0 00-.018-.111zm-7.91 7.86h3.4v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357zM.75 9.563V6.625h3.03v.357c0 .588.479 1.066 1.067 1.066H7.61c.588 0 1.066-.478 1.066-1.066v-.357h3.03v2.938H.75zm3.03-6.626v.357c0 .588.479 1.067 1.067 1.067H7.61c.589 0 1.067-.479 1.067-1.067v-.357h3.03v2.939H.75V2.937h3.03zm4.15 0v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357h3.4zm-3.4 3.688h3.4v.357a.318.318 0 01-.319.317H4.846a.317.317 0 01-.318-.317v-.357zM2.493.749h7.468l1.284 1.44H1.21L2.494.748zm9.213 12.502H.75v-2.939h3.03v.357c0 .589.479 1.067 1.067 1.067H7.61c.588 0 1.066-.478 1.066-1.067v-.357h3.03v2.94z",
          fill: "currentColor"
        }
      )
    ), /* @__PURE__ */ React.createElement(
      Typography,
      {
        style: {
          display: "inline",
          padding: "2px 8px 2px 2px",
          fontSize: "0.9rem"
        }
      },
      row.reponame
    ), /* @__PURE__ */ React.createElement("svg", { "data-icon": "tag", width: "15", height: "15", viewBox: "0 0 16 16" }, /* @__PURE__ */ React.createElement("desc", null, "tag"), /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M1 3a2 2 0 012-2h4.584a2 2 0 011.414.586l5.413 5.412a2 2 0 010 2.829L9.827 14.41a2 2 0 01-2.829 0L1.586 8.998A2 2 0 011 7.584V3zm3.487-.007a1.494 1.494 0 100 2.988 1.494 1.494 0 000-2.988z",
        fillRule: "evenodd",
        fill: "#25b5ba"
      }
    )), /* @__PURE__ */ React.createElement(
      Typography,
      {
        style: { display: "inline", padding: "3px", fontSize: "0.9rem" }
      },
      row.tag
    ))));
  }
  if (row.branch === "undefined" && row.sourcebranch === "undefined") {
    return null;
  } else if (row.targetbranch === "undefined") {
    return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("span", { style: { padding: "3px" } }, /* @__PURE__ */ React.createElement(
      "svg",
      {
        viewBox: "0 0 13 14",
        xmlns: "http://www.w3.org/2000/svg",
        width: "14",
        height: "14"
      },
      /* @__PURE__ */ React.createElement(
        "path",
        {
          d: "M12.438 2.452l-.006-.017a.367.367 0 00-.067-.114l-.004-.007L10.409.125a.374.374 0 00-.28-.125H2.327a.376.376 0 00-.28.125L.095 2.314l-.005.008a.374.374 0 00-.066.113l-.006.017a.37.37 0 00-.018.11v11.064c0 .206.168.374.374.374h11.707a.375.375 0 00.375-.374V2.563a.369.369 0 00-.018-.111zm-7.91 7.86h3.4v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357zM.75 9.563V6.625h3.03v.357c0 .588.479 1.066 1.067 1.066H7.61c.588 0 1.066-.478 1.066-1.066v-.357h3.03v2.938H.75zm3.03-6.626v.357c0 .588.479 1.067 1.067 1.067H7.61c.589 0 1.067-.479 1.067-1.067v-.357h3.03v2.939H.75V2.937h3.03zm4.15 0v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357h3.4zm-3.4 3.688h3.4v.357a.318.318 0 01-.319.317H4.846a.317.317 0 01-.318-.317v-.357zM2.493.749h7.468l1.284 1.44H1.21L2.494.748zm9.213 12.502H.75v-2.939h3.03v.357c0 .589.479 1.067 1.067 1.067H7.61c.588 0 1.066-.478 1.066-1.067v-.357h3.03v2.94z",
          fill: "currentColor"
        }
      )
    ), /* @__PURE__ */ React.createElement(
      Typography,
      {
        style: {
          display: "inline",
          padding: "2px 8px 2px 2px",
          fontSize: "0.9rem"
        }
      },
      row.reponame
    ), /* @__PURE__ */ React.createElement(
      "svg",
      {
        viewBox: "0 0 448 512",
        id: "IconChangeColor",
        height: "15",
        width: "15"
      },
      /* @__PURE__ */ React.createElement(
        "path",
        {
          d: "M160 80C160 112.8 140.3 140.1 112 153.3V241.1C130.8 230.2 152.7 224 176 224H272C307.3 224 336 195.3 336 160V153.3C307.7 140.1 288 112.8 288 80C288 35.82 323.8 0 368 0C412.2 0 448 35.82 448 80C448 112.8 428.3 140.1 400 153.3V160C400 230.7 342.7 288 272 288H176C140.7 288 112 316.7 112 352V358.7C140.3 371 160 399.2 160 432C160 476.2 124.2 512 80 512C35.82 512 0 476.2 0 432C0 399.2 19.75 371 48 358.7V153.3C19.75 140.1 0 112.8 0 80C0 35.82 35.82 0 80 0C124.2 0 160 35.82 160 80V80zM80 104C93.25 104 104 93.25 104 80C104 66.75 93.25 56 80 56C66.75 56 56 66.75 56 80C56 93.25 66.75 104 80 104zM368 56C354.7 56 344 66.75 344 80C344 93.25 354.7 104 368 104C381.3 104 392 93.25 392 80C392 66.75 381.3 56 368 56zM80 456C93.25 456 104 445.3 104 432C104 418.7 93.25 408 80 408C66.75 408 56 418.7 56 432C56 445.3 66.75 456 80 456z",
          id: "mainIconPathAttribute",
          fill: "#03989e"
        }
      )
    ), /* @__PURE__ */ React.createElement(
      Typography,
      {
        style: { display: "inline", padding: "2px", fontSize: "0.9rem" }
      },
      row.branch
    ))));
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("span", { style: { padding: "3px" } }, /* @__PURE__ */ React.createElement(
    "svg",
    {
      viewBox: "0 0 13 14",
      xmlns: "http://www.w3.org/2000/svg",
      width: "14",
      height: "14"
    },
    /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M12.438 2.452l-.006-.017a.367.367 0 00-.067-.114l-.004-.007L10.409.125a.374.374 0 00-.28-.125H2.327a.376.376 0 00-.28.125L.095 2.314l-.005.008a.374.374 0 00-.066.113l-.006.017a.37.37 0 00-.018.11v11.064c0 .206.168.374.374.374h11.707a.375.375 0 00.375-.374V2.563a.369.369 0 00-.018-.111zm-7.91 7.86h3.4v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357zM.75 9.563V6.625h3.03v.357c0 .588.479 1.066 1.067 1.066H7.61c.588 0 1.066-.478 1.066-1.066v-.357h3.03v2.938H.75zm3.03-6.626v.357c0 .588.479 1.067 1.067 1.067H7.61c.589 0 1.067-.479 1.067-1.067v-.357h3.03v2.939H.75V2.937h3.03zm4.15 0v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357h3.4zm-3.4 3.688h3.4v.357a.318.318 0 01-.319.317H4.846a.317.317 0 01-.318-.317v-.357zM2.493.749h7.468l1.284 1.44H1.21L2.494.748zm9.213 12.502H.75v-2.939h3.03v.357c0 .589.479 1.067 1.067 1.067H7.61c.588 0 1.066-.478 1.066-1.067v-.357h3.03v2.94z",
        fill: "currentColor"
      }
    )
  ), /* @__PURE__ */ React.createElement(
    Typography,
    {
      style: {
        display: "inline",
        padding: "2px 8px 2px 2px",
        fontSize: "0.9rem"
      }
    },
    row.reponame
  ), /* @__PURE__ */ React.createElement(
    "svg",
    {
      viewBox: "0 0 448 512",
      id: "IconChangeColor",
      height: "15",
      width: "15"
    },
    /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M160 80C160 112.8 140.3 140.1 112 153.3V241.1C130.8 230.2 152.7 224 176 224H272C307.3 224 336 195.3 336 160V153.3C307.7 140.1 288 112.8 288 80C288 35.82 323.8 0 368 0C412.2 0 448 35.82 448 80C448 112.8 428.3 140.1 400 153.3V160C400 230.7 342.7 288 272 288H176C140.7 288 112 316.7 112 352V358.7C140.3 371 160 399.2 160 432C160 476.2 124.2 512 80 512C35.82 512 0 476.2 0 432C0 399.2 19.75 371 48 358.7V153.3C19.75 140.1 0 112.8 0 80C0 35.82 35.82 0 80 0C124.2 0 160 35.82 160 80V80zM80 104C93.25 104 104 93.25 104 80C104 66.75 93.25 56 80 56C66.75 56 56 66.75 56 80C56 93.25 66.75 104 80 104zM368 56C354.7 56 344 66.75 344 80C344 93.25 354.7 104 368 104C381.3 104 392 93.25 392 80C392 66.75 381.3 56 368 56zM80 456C93.25 456 104 445.3 104 432C104 418.7 93.25 408 80 408C66.75 408 56 418.7 56 432C56 445.3 66.75 456 80 456z",
        id: "mainIconPathAttribute",
        fill: "#03989e"
      }
    )
  ), /* @__PURE__ */ React.createElement(
    Typography,
    {
      style: { display: "inline", padding: "2px", fontSize: "0.9rem" }
    },
    row.sourcebranch
  ), /* @__PURE__ */ React.createElement(
    "svg",
    {
      width: "18",
      height: "18",
      fill: "currentColor",
      className: "bi bi-arrow-right",
      viewBox: "0 -3 16 16",
      id: "IconChangeColor"
    },
    " ",
    /* @__PURE__ */ React.createElement(
      "path",
      {
        fillRule: "evenodd",
        d: "M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z",
        id: "mainIconPathAttribute"
      }
    ),
    " "
  ), " ", /* @__PURE__ */ React.createElement(
    "svg",
    {
      viewBox: "0 0 448 512",
      id: "IconChangeColor",
      height: "15",
      width: "15"
    },
    /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M160 80C160 112.8 140.3 140.1 112 153.3V241.1C130.8 230.2 152.7 224 176 224H272C307.3 224 336 195.3 336 160V153.3C307.7 140.1 288 112.8 288 80C288 35.82 323.8 0 368 0C412.2 0 448 35.82 448 80C448 112.8 428.3 140.1 400 153.3V160C400 230.7 342.7 288 272 288H176C140.7 288 112 316.7 112 352V358.7C140.3 371 160 399.2 160 432C160 476.2 124.2 512 80 512C35.82 512 0 476.2 0 432C0 399.2 19.75 371 48 358.7V153.3C19.75 140.1 0 112.8 0 80C0 35.82 35.82 0 80 0C124.2 0 160 35.82 160 80V80zM80 104C93.25 104 104 93.25 104 80C104 66.75 93.25 56 80 56C66.75 56 56 66.75 56 80C56 93.25 66.75 104 80 104zM368 56C354.7 56 344 66.75 344 80C344 93.25 354.7 104 368 104C381.3 104 392 93.25 392 80C392 66.75 381.3 56 368 56zM80 456C93.25 456 104 445.3 104 432C104 418.7 93.25 408 80 408C66.75 408 56 418.7 56 432C56 445.3 66.75 456 80 456z",
        id: "mainIconPathAttribute",
        fill: "#03989e"
      }
    )
  ), /* @__PURE__ */ React.createElement(
    Typography,
    {
      style: { display: "inline", padding: "2px", fontSize: "0.9rem" }
    },
    row.targetbranch
  ))));
}
function PrintCommit(props) {
  var _a;
  const row = props.props;
  if (row.commitId === "undefined") {
    return null;
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", { style: { display: "block" } }, /* @__PURE__ */ React.createElement(
    "svg",
    {
      style: { display: "inline" },
      viewBox: "0 -150 640 512",
      id: "IconChangeColor",
      height: "18",
      width: "18"
    },
    /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M476.8 288C461.1 361 397.4 416 320 416C242.6 416 178 361 163.2 288H32C14.33 288 0 273.7 0 256C0 238.3 14.33 224 32 224H163.2C178 150.1 242.6 96 320 96C397.4 96 461.1 150.1 476.8 224H608C625.7 224 640 238.3 640 256C640 273.7 625.7 288 608 288H476.8zM320 336C364.2 336 400 300.2 400 256C400 211.8 364.2 176 320 176C275.8 176 240 211.8 240 256C240 300.2 275.8 336 320 336z",
        id: "mainIconPathAttribute",
        stroke: "#0b3724",
        strokeWidth: "0",
        fill: "#000000"
      }
    )
  ), /* @__PURE__ */ React.createElement(
    Box,
    {
      style: { display: "inline-block", padding: "5px" },
      maxWidth: "200px"
    },
    /* @__PURE__ */ React.createElement(OverflowTooltip, { text: row.message })
  ), /* @__PURE__ */ React.createElement(
    Link,
    {
      style: { display: "inline" },
      href: row.commitlink,
      target: "_blank"
    },
    (_a = row.commitId) == null ? void 0 : _a.substring(0, 6)
  )));
}
function PrintPR(props) {
  var _a;
  const row = props.props;
  if (row.prId === "undefined") {
    return null;
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("div", { style: { display: "block" } }, /* @__PURE__ */ React.createElement(
    "svg",
    {
      viewBox: "-2 -3 24 24",
      fill: "none",
      id: "IconChangeColor",
      height: "21",
      width: "21"
    },
    /* @__PURE__ */ React.createElement(
      "path",
      {
        stroke: "#248ea8",
        strokeLinecap: "round",
        strokeLinejoin: "round",
        strokeWidth: "2",
        d: "M6 10a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm0 0v10m12-6a3 3 0 1 0 0 6 3 3 0 0 0 0-6zm0 0V9a2 2 0 0 0-2-2h-1m-2 0 2-2v2m-2 0h2m-2 0 2 2V7",
        id: "mainIconPathAttribute"
      }
    )
  ), /* @__PURE__ */ React.createElement(
    Box,
    {
      maxWidth: "200px",
      style: { display: "inline-block", padding: "5px" }
    },
    /* @__PURE__ */ React.createElement(OverflowTooltip, { text: row.prmessage })
  ), /* @__PURE__ */ React.createElement(Link, { style: { display: "inline" }, href: row.prlink, target: "_blank" }, "#", (_a = row.prId) == null ? void 0 : _a.substring(0, 6)))));
}
function PrintCD(props) {
  const row = props.props;
  return /* @__PURE__ */ React.createElement("div", { style: { display: "block" } }, /* @__PURE__ */ React.createElement(
    Typography,
    {
      component: "span",
      style: { paddingTop: "5px", fontSize: "0.9rem" }
    },
    "Service Deployed:",
    /* @__PURE__ */ React.createElement(OverflowTooltip, { text: row == null ? void 0 : row.cdser })
  ), /* @__PURE__ */ React.createElement(Typography, { component: "span", style: { fontSize: "0.9rem" } }, "Environments:", /* @__PURE__ */ React.createElement(OverflowTooltip, { text: row == null ? void 0 : row.cdenv })));
}

function AlertDialog(props) {
  const [open, setOpen] = useState(false);
  const { runPipeline } = props;
  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  return /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement(Tooltip, { title: "Re-run Pipeline", arrow: true }, /* @__PURE__ */ React.createElement(IconButton, { "aria-label": "replay", onClick: handleClickOpen }, /* @__PURE__ */ React.createElement(ReplayIcon, null))), /* @__PURE__ */ React.createElement(Dialog, { open, onClose: handleClose }, /* @__PURE__ */ React.createElement(DialogTitle, null, "Run Pipeline"), /* @__PURE__ */ React.createElement(DialogContent, null, /* @__PURE__ */ React.createElement(DialogContentText, null, "Do you want to re-run this pipeline?")), /* @__PURE__ */ React.createElement(DialogActions, null, /* @__PURE__ */ React.createElement(Button, { onClick: handleClose, variant: "outlined", color: "error" }, "cancel"), /* @__PURE__ */ React.createElement(
    Button,
    {
      onClick: () => {
        handleClose();
        runPipeline(
          Object(props.row),
          props.query1,
          props.setRefresh,
          props.refresh
        );
      },
      variant: "contained",
      color: "success"
    },
    "Run Pipeline"
  ))));
}

const getStatusComponent = (status = "") => {
  switch (status.toLocaleLowerCase("en-US")) {
    case "running":
      return /* @__PURE__ */ React.createElement(StatusRunning, null);
    case "failed":
      return /* @__PURE__ */ React.createElement(StatusError, null);
    case "success":
      return /* @__PURE__ */ React.createElement(StatusOK, null);
    case "aborted":
      return /* @__PURE__ */ React.createElement(StatusAborted, null);
    case "skipped":
      return /* @__PURE__ */ React.createElement(StatusAborted, null);
    default:
      return /* @__PURE__ */ React.createElement(StatusWarning, null);
  }
};
const stringsMap = {
  Aborted: "Aborted",
  Discontinuing: "Aborted",
  Running: "Running",
  AsyncWaiting: "Running",
  TaskWaiting: "Running",
  TimedWaiting: "Running",
  Failed: "Failed",
  Errored: "Failed",
  NotStarted: "NotStarted",
  Expired: "Expired",
  Queued: "Queued",
  Paused: "Paused",
  ResourceWaiting: "Waiting",
  Skipped: "Skipped",
  Success: "Success",
  IgnoreFailed: "Success",
  Suspended: "Suspended",
  Pausing: "Pausing",
  ApprovalRejected: "ApprovalRejected",
  InterventionWaiting: "Waiting",
  ApprovalWaiting: "ApprovalWaiting",
  InputWaiting: "Waiting",
  WaitStepRunning: "Waiting"
};
const ExecutionTable = ({
  baseUrl1,
  accountId,
  licenses,
  orgId,
  currProject,
  boolDisableRunPipeline,
  backendBaseUrl,
  setRefresh,
  refresh,
  pageSize,
  currTableData,
  // setState,
  page,
  handleChangePage,
  totalElements,
  handleChangeRowsPerPage,
  classes,
  runPipeline
}) => {
  const columns = [
    {
      title: "NO",
      field: "id",
      type: "numeric",
      width: "5%",
      align: "left",
      cellStyle: {
        paddingLeft: "30px"
      },
      render: (row) => {
        const link = `${baseUrl1}/ng/#/account/${accountId}/${licenses}/orgs/${orgId}/projects/${currProject}/pipelines/${row.pipelineId}/deployments/${row.planExecutionId}/pipeline`;
        const id = parseInt(row.id ? row.id : "0", 10);
        return /* @__PURE__ */ React.createElement(Link, { href: link, target: "_blank" }, /* @__PURE__ */ React.createElement("b", null, id));
      }
    },
    {
      title: "Pipeline Name",
      field: "col1",
      width: "22%",
      render: (row) => {
        const link = `${baseUrl1}/ng/#/account/${accountId}/${licenses}/orgs/${orgId}/projects/${currProject}/pipelines/${row.pipelineId}/deployments/${row.planExecutionId}/pipeline`;
        return /* @__PURE__ */ React.createElement(Typography, { style: { fontSize: "small", color: "grey" } }, /* @__PURE__ */ React.createElement(Link, { href: link, target: "_blank", style: { fontSize: "0.9rem" } }, /* @__PURE__ */ React.createElement("b", null, row.name, " ")), /* @__PURE__ */ React.createElement("br", null), "Run ID: ", row == null ? void 0 : row.runSequence);
      },
      customFilterAndSearch: (term, row) => {
        var _a;
        const temp = (_a = `${row == null ? void 0 : row.name} ${row == null ? void 0 : row.runSequence}`) != null ? _a : "";
        return temp.toLowerCase().indexOf(term.toLowerCase()) > -1;
      },
      customSort: (row1, row2) => {
        var _a, _b;
        const a = (_a = row1.name) != null ? _a : "";
        const b = (_b = row2.name) != null ? _b : "";
        return a > b ? 1 : -1;
      }
    },
    {
      title: "Pipeline Status",
      field: "col2",
      width: "18%",
      render: useCallback(
        (row) => {
          var _a, _b;
          return /* @__PURE__ */ React.createElement(Box, { display: "flex", alignItems: "center" }, getStatusComponent(stringsMap[(_a = row == null ? void 0 : row.status) != null ? _a : "Failed"]), /* @__PURE__ */ React.createElement(Box, { mr: 1 }), /* @__PURE__ */ React.createElement(Typography, { variant: "button" }, stringsMap[(_b = row == null ? void 0 : row.status) != null ? _b : "Failed"]));
        },
        []
      ),
      customFilterAndSearch: (term, row) => {
        var _a;
        const temp = (_a = row == null ? void 0 : row.status) != null ? _a : "";
        return temp.toLowerCase().indexOf(term.toLowerCase()) > -1;
      },
      customSort: (row1, row2) => {
        var _a, _b;
        const a = (_a = row1.status) != null ? _a : "";
        const b = (_b = row2.status) != null ? _b : "";
        return a > b ? 1 : -1;
      }
    },
    {
      title: "Details",
      field: "col3",
      width: "30%",
      sorting: false,
      render: useCallback(
        (row) => /* @__PURE__ */ React.createElement(PrintCard, { props: row }),
        []
      )
    },
    {
      title: "Pipeline time",
      field: "col4",
      type: "date",
      width: "20%",
      render: useCallback((row) => {
        if (durationHumanized(
          new Date(Number(row.startTime)),
          new Date(Number(row.endTime))
        ) === "NaN years") {
          return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Typography, { style: { fontSize: "0.9rem" } }, new Date(Number(row.startTime)).toUTCString()), /* @__PURE__ */ React.createElement(Typography, { variant: "body2" }, "run ", relativeTimeTo(new Date(Number(row.startTime)))));
        }
        return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Typography, { style: { fontSize: "0.9rem" } }, new Date(Number(row.startTime)).toUTCString()), /* @__PURE__ */ React.createElement(Typography, { variant: "body2" }, "run ", relativeTimeTo(new Date(Number(row.startTime)))), /* @__PURE__ */ React.createElement(Typography, { variant: "body2" }, "took", " ", durationHumanized(
          new Date(Number(row.startTime)),
          new Date(Number(row.endTime))
        )));
      }, []),
      customSort: (row1, row2) => {
        var _a, _b;
        const a = (_a = row1.startTime) != null ? _a : "";
        const b = (_b = row2.startTime) != null ? _b : "";
        return a > b ? 1 : -1;
      },
      customFilterAndSearch: (term, row) => {
        const temp = new Date(Number(row.startTime)).toUTCString();
        return temp.toLowerCase().indexOf(term.toLowerCase()) > -1;
      }
    }
  ];
  if (!boolDisableRunPipeline) {
    columns.push({
      title: "Run Pipeline",
      field: "col5",
      sorting: false,
      width: "5%",
      render: (row) => {
        const query1 = new URLSearchParams({
          accountIdentifier: `${accountId}`,
          routingId: `${accountId}`,
          orgIdentifier: `${orgId}`,
          projectIdentifier: `${currProject}`
        }).toString();
        return /* @__PURE__ */ React.createElement(
          AlertDialog,
          {
            row,
            backendBaseUrl,
            query1,
            setRefresh,
            refresh,
            runPipeline
          }
        );
      }
    });
  }
  return /* @__PURE__ */ React.createElement(
    Table,
    {
      options: {
        paging: true,
        filtering: false,
        emptyRowsWhenPaging: false,
        pageSize,
        pageSizeOptions: [5, 10, 25]
      },
      data: currTableData != null ? currTableData : [],
      columns,
      actions: [
        {
          icon: () => /* @__PURE__ */ React.createElement(ReplayIcon, null),
          tooltip: "Refresh Data",
          isFreeAction: true,
          onClick: () => {
            setRefresh(!refresh);
          }
        }
      ],
      emptyContent: /* @__PURE__ */ React.createElement("div", { className: classes.empty }, /* @__PURE__ */ React.createElement(CircularProgress, null)),
      title: "Execution History",
      page,
      onPageChange: handleChangePage,
      totalCount: totalElements,
      onRowsPerPageChange: handleChangeRowsPerPage
    }
  );
};

var AsyncStatus = /* @__PURE__ */ ((AsyncStatus2) => {
  AsyncStatus2[AsyncStatus2["Init"] = 0] = "Init";
  AsyncStatus2[AsyncStatus2["Loading"] = 1] = "Loading";
  AsyncStatus2[AsyncStatus2["Success"] = 2] = "Success";
  AsyncStatus2[AsyncStatus2["Error"] = 3] = "Error";
  AsyncStatus2[AsyncStatus2["Unauthorized"] = 4] = "Unauthorized";
  return AsyncStatus2;
})(AsyncStatus || {});

function getSecureHarnessKey(key) {
  try {
    const token = JSON.parse(decodeURI(atob(localStorage.getItem(key) || "")));
    return token ? `Bearer ${token}` : "";
  } catch (err) {
    console.log("Failed to read Harness tokens", err);
    return void 0;
  }
}

const useGetExecutionsList = ({
  accountId,
  orgId,
  currProject,
  pageSize,
  page,
  pipelineId,
  serviceId,
  env,
  backendBaseUrl,
  refresh
}) => {
  const [status, setStatus] = useState(AsyncStatus.Init);
  const [currTableData, setCurrTableData] = useState([]);
  const [totalElements, setTotalElements] = useState(50);
  const [flag, setFlag] = useState(false);
  const identityApi = useApi(identityApiRef);
  useAsyncRetry(async () => {
    const query = new URLSearchParams({
      accountIdentifier: `${accountId}`,
      routingId: `${accountId}`,
      orgIdentifier: `${orgId}`,
      projectIdentifier: `${currProject}`,
      size: `${pageSize}`,
      page: `${page}`
    });
    const { token: apiToken } = await identityApi.getCredentials();
    const token = getSecureHarnessKey("token") || apiToken;
    const value = token && token === apiToken ? `Bearer ${token}` : token;
    const headers = new Headers();
    if (value) {
      headers.append("Authorization", value);
      headers.append("Content-Type", "application/json");
    }
    let body;
    const identifiers = [];
    if (serviceId) {
      body = JSON.stringify({
        filterType: "PipelineExecution",
        moduleProperties: {
          cd: {
            serviceIdentifiers: [
              serviceId == null ? void 0 : serviceId.split(",").map((item) => item.trim())[0]
            ]
          }
        }
      });
    } else if (pipelineId && pipelineId.trim()) {
      pipelineId.split(",").forEach((item) => {
        const trimmedString = item.trim();
        if (trimmedString) {
          identifiers.push(trimmedString);
        }
      });
      body = JSON.stringify({
        filterType: "PipelineExecution",
        pipelineIdentifiers: identifiers
      });
    }
    setStatus(AsyncStatus.Loading);
    const response = await fetch(
      `${await backendBaseUrl}/harness/${env}/gateway/pipeline/api/pipelines/execution/summary?${query}`,
      {
        headers,
        method: "POST",
        body
      }
    );
    const getBuilds = (tableData, currentPageSize) => {
      return tableData.map((dataItem, index) => {
        var _a, _b, _c, _d, _e, _f, _g, _h, _i, _j, _k, _l, _m, _n, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z, _A, _B, _C, _D, _E, _F, _G, _H, _I, _J, _K, _L, _M, _N, _O, _P, _Q, _R, _S, _T, _U, _V;
        let serviceString = "";
        let envString = "";
        const request = typeof ((_c = (_b = (_a = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _a.ci) == null ? void 0 : _b.ciExecutionInfoDTO) == null ? void 0 : _c.pullRequest) === "undefined" ? "branch" : "pullRequest";
        if ((_d = dataItem == null ? void 0 : dataItem.modules) == null ? void 0 : _d.includes("cd")) {
          const serviceNames = /* @__PURE__ */ new Set();
          const envNames = /* @__PURE__ */ new Set();
          const mapdata = dataItem == null ? void 0 : dataItem.layoutNodeMap;
          Object.keys(mapdata).forEach((key) => {
            var _a2, _b2, _c2, _d2, _e2, _f2, _g2, _h2, _i2, _j2, _k2, _l2, _m2, _n2, _o2, _p2;
            if (mapdata[key].nodeType === "Deployment") {
              if ((_d2 = (_c2 = (_b2 = (_a2 = mapdata[key]) == null ? void 0 : _a2.moduleInfo) == null ? void 0 : _b2.cd) == null ? void 0 : _c2.infraExecutionSummary) == null ? void 0 : _d2.name)
                envNames.add(
                  (_h2 = (_g2 = (_f2 = (_e2 = mapdata[key]) == null ? void 0 : _e2.moduleInfo) == null ? void 0 : _f2.cd) == null ? void 0 : _g2.infraExecutionSummary) == null ? void 0 : _h2.name
                );
              if ((_l2 = (_k2 = (_j2 = (_i2 = mapdata[key]) == null ? void 0 : _i2.moduleInfo) == null ? void 0 : _j2.cd) == null ? void 0 : _k2.serviceInfo) == null ? void 0 : _l2.displayName)
                serviceNames.add(
                  (_p2 = (_o2 = (_n2 = (_m2 = mapdata[key]) == null ? void 0 : _m2.moduleInfo) == null ? void 0 : _n2.cd) == null ? void 0 : _o2.serviceInfo) == null ? void 0 : _p2.displayName
                );
            }
          });
          envString = Array.from(envNames).join(",");
          serviceString = Array.from(serviceNames).join(",");
        }
        return {
          id: `${page * currentPageSize + index + 1}`,
          name: `${dataItem == null ? void 0 : dataItem.name}`,
          status: `${dataItem == null ? void 0 : dataItem.status}`,
          startTime: `${dataItem == null ? void 0 : dataItem.startTs}`,
          endTime: `${dataItem == null ? void 0 : dataItem.endTs}`,
          pipelineId: `${dataItem == null ? void 0 : dataItem.pipelineIdentifier}`,
          planExecutionId: `${dataItem == null ? void 0 : dataItem.planExecutionId}`,
          runSequence: `${dataItem == null ? void 0 : dataItem.runSequence}`,
          commitId: `${(_j = (_i = (_h = (_g = (_f = (_e = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _e.ci) == null ? void 0 : _f.ciExecutionInfoDTO) == null ? void 0 : _g[request]) == null ? void 0 : _h.commits) == null ? void 0 : _i["0"]) == null ? void 0 : _j.id}`,
          commitlink: `${(_p = (_o = (_n = (_m = (_l = (_k = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _k.ci) == null ? void 0 : _l.ciExecutionInfoDTO) == null ? void 0 : _m[request]) == null ? void 0 : _n.commits) == null ? void 0 : _o["0"]) == null ? void 0 : _p.link}`,
          branch: `${(_r = (_q = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _q.ci) == null ? void 0 : _r.branch}`,
          message: `${(_x = (_w = (_v = (_u = (_t = (_s = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _s.ci) == null ? void 0 : _t.ciExecutionInfoDTO) == null ? void 0 : _u[request]) == null ? void 0 : _v.commits) == null ? void 0 : _w["0"]) == null ? void 0 : _x.message}`,
          prmessage: `${(_B = (_A = (_z = (_y = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _y.ci) == null ? void 0 : _z.ciExecutionInfoDTO) == null ? void 0 : _A.pullRequest) == null ? void 0 : _B.title}`,
          prlink: `${(_F = (_E = (_D = (_C = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _C.ci) == null ? void 0 : _D.ciExecutionInfoDTO) == null ? void 0 : _E.pullRequest) == null ? void 0 : _F.link}`,
          sourcebranch: `${(_J = (_I = (_H = (_G = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _G.ci) == null ? void 0 : _H.ciExecutionInfoDTO) == null ? void 0 : _I.pullRequest) == null ? void 0 : _J.sourceBranch}`,
          targetbranch: `${(_N = (_M = (_L = (_K = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _K.ci) == null ? void 0 : _L.ciExecutionInfoDTO) == null ? void 0 : _M.pullRequest) == null ? void 0 : _N.targetBranch}`,
          prId: `${(_R = (_Q = (_P = (_O = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _O.ci) == null ? void 0 : _P.ciExecutionInfoDTO) == null ? void 0 : _Q.pullRequest) == null ? void 0 : _R.id}`,
          cdenv: `${envString}`,
          cdser: `${serviceString}`,
          reponame: `${(_T = (_S = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _S.ci) == null ? void 0 : _T.repoName}`,
          tag: `${(_V = (_U = dataItem == null ? void 0 : dataItem.moduleInfo) == null ? void 0 : _U.ci) == null ? void 0 : _V.tag}`
        };
      });
    };
    if (response.status === 200) {
      const data = await response.json();
      setStatus(AsyncStatus.Success);
      const responseData = data.data.content;
      if (data.data.totalElements < 50) {
        setTotalElements(data.data.totalElements);
      }
      setCurrTableData(getBuilds(responseData, pageSize));
    } else if (response.status === 401)
      setStatus(AsyncStatus.Unauthorized);
    else
      setStatus(AsyncStatus.Error);
    setFlag(true);
  }, [
    refresh,
    page,
    pageSize,
    accountId,
    orgId,
    currProject,
    pipelineId,
    serviceId,
    env
  ]);
  return { status, currTableData, totalElements, flag };
};

const useGetLicenseWithAuth = ({
  backendBaseUrl,
  env,
  accountId
}) => {
  const [licenses, setLicenses] = useState("cd");
  const identityApi = useApi(identityApiRef);
  useAsyncRetry(async () => {
    var _a, _b, _c;
    const { token: apiToken } = await identityApi.getCredentials();
    const token = getSecureHarnessKey("token") || apiToken;
    const value = token && token === apiToken ? `Bearer ${token}` : token;
    const headers = new Headers();
    if (value) {
      headers.append("Authorization", value);
    }
    const response = await fetch(
      `${await backendBaseUrl}/harness/${env}/gateway/ng/api/licenses/account?routingId=${accountId}&accountIdentifier=${accountId}`,
      {
        headers
      }
    );
    if (response.status === 200) {
      const data = await response.json();
      if (((_c = (_b = (_a = data == null ? void 0 : data.data) == null ? void 0 : _a.allModuleLicenses) == null ? void 0 : _b.CD) == null ? void 0 : _c.length) === 0) {
        setLicenses("ci");
      }
    }
  }, [env, accountId]);
  return { licenses };
};

const useMutateRunPipeline = ({
  backendBaseUrl,
  env
}) => {
  const identityApi = useApi(identityApiRef);
  const runPipeline = async (row, query1) => {
    const { token: apiToken } = await identityApi.getCredentials();
    const token = getSecureHarnessKey("token") || apiToken;
    const value = token && token === apiToken ? `Bearer ${token}` : token;
    const headers = new Headers();
    if (value) {
      headers.append("Authorization", value);
    }
    const response = await fetch(
      `${await backendBaseUrl}/harness/${env}/gateway/pipeline/api/pipelines/execution/${row.planExecutionId}/inputset?${query1}`,
      {
        headers
      }
    );
    const data = await response.text();
    const postHeaders = {
      "content-type": "application/yaml"
    };
    if (value) {
      postHeaders.Authorization = value;
    }
    return await fetch(
      `${await backendBaseUrl}/harness/${env}/gateway/pipeline/api/pipeline/execute/rerun/${row.planExecutionId}/${row.pipelineId}?${query1}&moduleType=ci`,
      {
        headers: postHeaders,
        body: `${data}`,
        method: "POST"
      }
    );
  };
  return { runPipeline };
};

function convertStringToObject(inputString) {
  if (!inputString)
    return {};
  const object = {};
  const lines = inputString.split("\n");
  for (const line of lines) {
    if (line === "")
      continue;
    const [label, ...rest] = line.split(":");
    const trimmedLabel = label.trim();
    object[trimmedLabel] = rest.join(":").trim();
  }
  return object;
}
const usePipelineSlugEntity = () => {
  var _a, _b, _c, _d;
  const { entity } = useEntity();
  const isNewAnnotationPresent = Boolean((_a = entity.metadata.annotations) == null ? void 0 : _a["harness.io/pipelines"]) || Boolean((_b = entity.metadata.annotations) == null ? void 0 : _b["harness.io/services"]);
  const harnessPipelineObject = convertStringToObject(
    (_c = entity.metadata.annotations) == null ? void 0 : _c["harness.io/pipelines"]
  );
  const harnessServicesObject = convertStringToObject(
    (_d = entity.metadata.annotations) == null ? void 0 : _d["harness.io/services"]
  );
  return {
    isNewAnnotationPresent,
    harnessPipelineObject,
    harnessServicesObject
  };
};

const useStyles = makeStyles((theme) => ({
  container: {
    width: "100%"
  },
  label: {
    marginBottom: "2px",
    fontSize: "14px !important"
  },
  empty: {
    padding: theme.spacing(2),
    display: "flex",
    justifyContent: "center"
  }
}));
function ExecutionList() {
  var _a;
  const [refresh, setRefresh] = useState(false);
  const [page, setPage] = useState(0);
  const [pageSize, setPageSize] = useState(5);
  const classes = useStyles();
  const discoveryApi = useApi(discoveryApiRef);
  const config = useApi(configApiRef);
  const backendBaseUrl = discoveryApi.getBaseUrl("proxy");
  const {
    isNewAnnotationPresent,
    harnessPipelineObject,
    harnessServicesObject
  } = usePipelineSlugEntity();
  const [selectedPipelineUrl, setSelectedPipelineUrl] = useState(() => {
    if (Object.keys(harnessServicesObject).length) {
      return harnessServicesObject[Object.keys(harnessServicesObject)[0]];
    }
    return harnessPipelineObject[Object.keys(harnessPipelineObject)[0]];
  });
  const boolDisableRunPipeline = (_a = config.getOptionalBoolean("harness.disableRunPipeline")) != null ? _a : false;
  const { entity } = useEntity();
  const allEnvsAnnotations = [
    ["prod", "harness.io/project-url"],
    ["stage", "harness.io/project-url-stage"],
    ["qa", "harness.io/project-url-qa"],
    ["stress", "harness.io/project-url-stress"]
  ];
  const envIds = [];
  allEnvsAnnotations.forEach((envAnnotation) => {
    var _a2;
    if (Boolean((_a2 = entity.metadata.annotations) == null ? void 0 : _a2[envAnnotation[1]])) {
      envIds.push(envAnnotation[0]);
    }
  });
  const [env, setEnv] = useState(envIds[0]);
  const {
    orgId,
    accountId,
    pipelineId,
    serviceId,
    urlParams,
    baseUrl1,
    projectIds,
    envFromUrl
  } = useProjectSlugFromEntity(
    env,
    isNewAnnotationPresent,
    selectedPipelineUrl
  );
  const allProjects = projectIds == null ? void 0 : projectIds.split(",").map((item) => item.trim());
  const [currProject, setCurrProject] = useState(allProjects == null ? void 0 : allProjects[0]);
  const envToUse = isNewAnnotationPresent && envFromUrl ? envFromUrl : env;
  const projectToUse = isNewAnnotationPresent ? projectIds : currProject;
  useEffect(() => {
    setRefresh(!refresh);
    setCurrProject(allProjects == null ? void 0 : allProjects[0]);
  }, [env]);
  const [cicd, setCicd] = useState("CD");
  const [pipelineIdValue, setPipelineIdValue] = useState(pipelineId);
  const [serviceIdValue, setServiceIdValue] = useState(serviceId);
  const {
    status: state,
    currTableData,
    totalElements,
    flag
  } = useGetExecutionsList({
    accountId,
    orgId,
    currProject: projectToUse,
    pageSize,
    page,
    pipelineId: !isNewAnnotationPresent ? pipelineIdValue : pipelineId,
    serviceId: !isNewAnnotationPresent ? serviceIdValue : serviceId,
    env: envToUse,
    backendBaseUrl,
    refresh
  });
  const { licenses } = useGetLicenseWithAuth({
    backendBaseUrl,
    env: envToUse,
    accountId
  });
  const { runPipeline: executePipeline } = useMutateRunPipeline({
    backendBaseUrl,
    env: envToUse
  });
  async function runPipeline(row, query1, setRefresh2, refresh2) {
    const resp2 = await executePipeline(row, query1);
    if (resp2.status === 200) {
      const Toast = Swal.mixin({
        toast: true,
        position: "top",
        showConfirmButton: false,
        timer: 3e3,
        timerProgressBar: true
      });
      Toast.fire({
        icon: "success",
        title: "Pipeline ran successfully"
      });
    } else if (resp2.status === 403) {
      const Toast = Swal.mixin({
        toast: true,
        position: "top",
        showCloseButton: true,
        showConfirmButton: false,
        width: "500px"
      });
      Toast.fire({
        icon: "warning",
        title: "You don't have access to trigger this pipeline",
        text: "Please check your API key configuration"
      });
    } else {
      const Toast = Swal.mixin({
        toast: true,
        position: "top",
        showCloseButton: true,
        showConfirmButton: false
      });
      Toast.fire({
        icon: "error",
        title: "Pipeline Trigger Failed"
      });
    }
    setRefresh2(!refresh2);
  }
  const handleChangePage = (currentPage, currentPageSize) => {
    setPage(currentPage);
    setPageSize(currentPageSize);
  };
  const handleChangeRowsPerPage = (currentPageSize) => {
    setPage(0);
    setPageSize(currentPageSize);
  };
  const handleChange = async (event) => {
    setEnv(event.target.value);
  };
  const handleChangeProject = async (event) => {
    setCurrProject(event.target.value);
    setRefresh(!refresh);
  };
  const EnvDropDown = envIds.length > 1 ? /* @__PURE__ */ React.createElement(FormControl, { fullWidth: true }, /* @__PURE__ */ React.createElement(
    InputLabel,
    {
      htmlFor: "Environment",
      classes: {
        root: classes.label
      }
    },
    "Environment"
  ), /* @__PURE__ */ React.createElement(
    Select,
    {
      labelId: "Environment",
      id: "Environment",
      value: env,
      onChange: handleChange
    },
    envIds.map((envId) => /* @__PURE__ */ React.createElement(MenuItem, { value: envId }, envId))
  ), /* @__PURE__ */ React.createElement(FormHelperText, null)) : null;
  const ProjectDropDown = allProjects && (allProjects == null ? void 0 : allProjects.length) > 1 ? /* @__PURE__ */ React.createElement(FormControl, { fullWidth: true }, /* @__PURE__ */ React.createElement(
    InputLabel,
    {
      htmlFor: "Project",
      classes: {
        root: classes.label
      }
    },
    "Project"
  ), /* @__PURE__ */ React.createElement(
    Select,
    {
      labelId: "Project",
      id: "Project",
      value: currProject,
      onChange: handleChangeProject
    },
    allProjects.map((proj) => /* @__PURE__ */ React.createElement(MenuItem, { value: proj }, proj))
  ), /* @__PURE__ */ React.createElement(FormHelperText, null)) : null;
  const handleUrlChange = async (event) => {
    setSelectedPipelineUrl(event.target.value);
    setRefresh(!refresh);
  };
  const NewUrlDropDown = /* @__PURE__ */ React.createElement(FormControl, { fullWidth: true }, /* @__PURE__ */ React.createElement(
    InputLabel,
    {
      htmlFor: "Service/Pipeline",
      classes: {
        root: classes.label
      }
    },
    "Service / Pipeline"
  ), /* @__PURE__ */ React.createElement(
    Select,
    {
      displayEmpty: true,
      labelId: "Service/Pipeline",
      id: "Service/Pipeline",
      value: selectedPipelineUrl,
      onChange: handleUrlChange
    },
    Object.keys(harnessServicesObject).length ? /* @__PURE__ */ React.createElement(ListSubheader, null, "Services") : null,
    Object.keys(harnessServicesObject).map((envId) => /* @__PURE__ */ React.createElement(MenuItem, { value: harnessServicesObject[envId] }, envId)),
    Object.keys(harnessPipelineObject).length > 0 ? /* @__PURE__ */ React.createElement(ListSubheader, null, "Pipelines") : null,
    Object.keys(harnessPipelineObject).map((envId) => /* @__PURE__ */ React.createElement(MenuItem, { value: harnessPipelineObject[envId] }, envId))
  ), /* @__PURE__ */ React.createElement(FormHelperText, null));
  const changeFunction = (event) => {
    const newValue = event.target.value;
    setCicd(newValue);
    if (newValue === "CI") {
      setServiceIdValue("");
      setPipelineIdValue(pipelineId);
    } else if (newValue === "CD") {
      setPipelineIdValue("");
      setServiceIdValue(serviceId);
    }
  };
  const OldUrlDropDown = /* @__PURE__ */ React.createElement(FormControl, { fullWidth: true }, /* @__PURE__ */ React.createElement(
    InputLabel,
    {
      htmlFor: "CI/CD",
      classes: {
        root: classes.label
      }
    },
    "CI / CD"
  ), /* @__PURE__ */ React.createElement(
    Select,
    {
      displayEmpty: true,
      labelId: "CI/CD",
      id: "CI/CD",
      value: cicd,
      onChange: changeFunction
    },
    /* @__PURE__ */ React.createElement(MenuItem, { value: "CI" }, "CI"),
    /* @__PURE__ */ React.createElement(MenuItem, { value: "CD" }, "CD")
  ), /* @__PURE__ */ React.createElement(FormHelperText, null));
  const DropDownComponent = /* @__PURE__ */ React.createElement(Grid, { container: true, spacing: 3, marginBottom: 4, marginTop: 0, marginLeft: 0 }, envIds.length > 1 && !isNewAnnotationPresent ? /* @__PURE__ */ React.createElement(Grid, { item: true, md: 3 }, EnvDropDown) : null, allProjects && (allProjects == null ? void 0 : allProjects.length) > 1 && !isNewAnnotationPresent ? /* @__PURE__ */ React.createElement(Grid, { item: true, md: 3 }, ProjectDropDown) : null, isNewAnnotationPresent ? /* @__PURE__ */ React.createElement(Grid, { item: true, md: 3 }, NewUrlDropDown) : null, !isNewAnnotationPresent ? /* @__PURE__ */ React.createElement(Grid, { item: true, md: 3 }, OldUrlDropDown) : null);
  if (state === AsyncStatus.Init || state === AsyncStatus.Loading || state === AsyncStatus.Success && !flag) {
    return /* @__PURE__ */ React.createElement("div", { className: classes.empty }, /* @__PURE__ */ React.createElement(CircularProgress, null));
  }
  if (!urlParams || state === AsyncStatus.Error || state === AsyncStatus.Unauthorized || state === AsyncStatus.Success && currTableData.length === 0 && flag) {
    let description = "";
    if (state === AsyncStatus.Unauthorized)
      description = "Could not find the pipeline executions, the x-api-key is either missing or incorrect in app-config.yaml under proxy settings.";
    else if (!urlParams)
      description = "Could not find the pipeline executions, please check your project-url configuration in catalog-info.yaml.";
    else if (state === AsyncStatus.Success && currTableData.length === 0)
      description = "No executions found";
    else
      description = "Could not find the pipeline executions, please check your configurations in catalog-info.yaml or check your permissions.";
    return /* @__PURE__ */ React.createElement(React.Fragment, null, DropDownComponent, /* @__PURE__ */ React.createElement(
      EmptyState,
      {
        title: "Harness CI-CD pipelines",
        description,
        missing: "data"
      }
    ));
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("div", { className: classes.container }, DropDownComponent, /* @__PURE__ */ React.createElement(
    ExecutionTable,
    {
      baseUrl1,
      accountId,
      licenses,
      orgId,
      currProject: projectToUse,
      boolDisableRunPipeline,
      backendBaseUrl,
      setRefresh,
      refresh,
      pageSize,
      currTableData,
      page,
      handleChangePage,
      totalElements,
      handleChangeRowsPerPage,
      classes,
      runPipeline
    }
  )));
}

const isHarnessCiCdAvailable = (entity) => {
  var _a, _b, _c, _d, _e, _f;
  return Boolean((_a = entity.metadata.annotations) == null ? void 0 : _a["harness.io/project-url"]) || Boolean((_b = entity.metadata.annotations) == null ? void 0 : _b["harness.io/project-url-qa"]) || Boolean((_c = entity.metadata.annotations) == null ? void 0 : _c["harness.io/project-url-stress"]) || Boolean((_d = entity.metadata.annotations) == null ? void 0 : _d["harness.io/project-url-stage"]) || Boolean((_e = entity.metadata.annotations) == null ? void 0 : _e["harness.io/pipelines"]) || Boolean((_f = entity.metadata.annotations) == null ? void 0 : _f["harness.io/services"]);
};
const Router = () => {
  const { entity } = useEntity();
  if (!isHarnessCiCdAvailable(entity)) {
    return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(MissingAnnotationEmptyState, { annotation: "harness.io/project-url" }));
  }
  return /* @__PURE__ */ React.createElement(Routes, null, /* @__PURE__ */ React.createElement(Route, { path: "/", element: /* @__PURE__ */ React.createElement(ExecutionList, null) }));
};

var Router$1 = /*#__PURE__*/Object.freeze({
  __proto__: null,
  isHarnessCiCdAvailable: isHarnessCiCdAvailable,
  Router: Router
});

export { EntityHarnessCiCdContent, HarnessCiCdPage, Router, harnessCIBuildRouteRef, harnessCiCdPlugin, harnessCiCdRouteRef, isHarnessCiCdAvailable };
