// @backstage/backend-defaults/dist/entrypoints/auth/authServiceFactory.cjs.js
'use strict';

var backendPluginApi = require('@backstage/backend-plugin-api');
var DefaultAuthService = require('./DefaultAuthService.cjs.js');
var ExternalAuthTokenHandler = require('./external/ExternalAuthTokenHandler.cjs.js');
var PluginTokenHandler = require('./plugin/PluginTokenHandler.cjs.js');
var createPluginKeySource = require('./plugin/keys/createPluginKeySource.cjs.js');
var UserTokenHandler = require('./user/UserTokenHandler.cjs.js');

const pluginTokenHandlerDecoratorServiceRef = backendPluginApi.createServiceRef({
  id: "core.auth.pluginTokenHandlerDecorator",
  defaultFactory: async (service) => backendPluginApi.createServiceFactory({
    service,
    deps: {},
    factory: async () => {
      return (impl) => impl;
    }
  })
});
const authServiceFactory = backendPluginApi.createServiceFactory({
  service: backendPluginApi.coreServices.auth,
  deps: {
    config: backendPluginApi.coreServices.rootConfig,
    logger: backendPluginApi.coreServices.logger,
    discovery: backendPluginApi.coreServices.discovery,
    plugin: backendPluginApi.coreServices.pluginMetadata,
    database: backendPluginApi.coreServices.database,
    pluginTokenHandlerDecorator: pluginTokenHandlerDecoratorServiceRef,
    externalTokenHandlers: ExternalAuthTokenHandler.externalTokenHandlersServiceRef
  },
  async factory({
    config,
    discovery,
    plugin,
    logger,
    database,
    pluginTokenHandlerDecorator,
    externalTokenHandlers
  }) {
    const disableDefaultAuthPolicy = config.getOptionalBoolean(
      "backend.auth.dangerouslyDisableDefaultAuthPolicy"
    ) ?? false;
    const keyDuration = { hours: 1 };
    const keySource = await createPluginKeySource.createPluginKeySource({
      config,
      database,
      logger,
      keyDuration
    });
    const userTokens = UserTokenHandler.UserTokenHandler.create({
      discovery,
      logger
    });
    const pluginTokens = pluginTokenHandlerDecorator(
      PluginTokenHandler.DefaultPluginTokenHandler.create({
        ownPluginId: plugin.getId(),
        logger,
        keySource,
        keyDuration,
        discovery
      })
    );
    const externalTokens = ExternalAuthTokenHandler.ExternalAuthTokenHandler.create({
      ownPluginId: plugin.getId(),
      config,
      logger,
      externalTokenHandlers
    });
    return new DefaultAuthService.DefaultAuthService(
      userTokens,
      pluginTokens,
      externalTokens,
      plugin.getId(),
      disableDefaultAuthPolicy,
      keySource
    );
  }
});

exports.authServiceFactory = authServiceFactory;
exports.pluginTokenHandlerDecoratorServiceRef = pluginTokenHandlerDecoratorServiceRef;
