// @backstage-community/plugin-sonarqube/dist/components/SonarQubeCard/RatingCard.esm.js
import { Link } from '@backstage/core-components';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';

const useStyles = makeStyles((theme) => {
  return {
    root: {
      margin: theme.spacing(1, 0),
      minWidth: "140px"
    },
    upper: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center"
    },
    cardTitle: {
      textAlign: "center"
    },
    wrapIcon: {
      display: "inline-flex",
      verticalAlign: "baseline"
    },
    left: {
      display: "flex"
    },
    right: {
      display: "flex",
      marginLeft: theme.spacing(0.5)
    }
  };
});
const RatingCard = ({
  leftSlot,
  rightSlot,
  title,
  titleIcon,
  link
}) => {
  const classes = useStyles();
  return /* @__PURE__ */ React.createElement(Link, { to: link, color: "inherit", underline: "none" }, /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.root }, /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.upper }, /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.left }, leftSlot), /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.right }, rightSlot)), /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.cardTitle }, /* @__PURE__ */ React.createElement(Typography, { variant: "body1", className: classes.wrapIcon }, titleIcon, " ", title))));
};

export { RatingCard };
