// @backstage-community/plugin-sonarqube/dist/components/SonarQubeCard/SonarQubeCard.esm.js
import { useEntity, MissingAnnotationEmptyState } from '@backstage/plugin-catalog-react';
import { sonarQubeApiRef, useProjectInfo, SONARQUBE_PROJECT_KEY_ANNOTATION } from '@backstage-community/plugin-sonarqube-react';
import Chip from '@material-ui/core/Chip';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import BugReport from '@material-ui/icons/BugReport';
import Lock from '@material-ui/icons/Lock';
import LockOpen from '@material-ui/icons/LockOpen';
import Security from '@material-ui/icons/Security';
import SentimentVeryDissatisfied from '@material-ui/icons/SentimentVeryDissatisfied';
import SentimentVerySatisfied from '@material-ui/icons/SentimentVerySatisfied';
import React, { useMemo } from 'react';
import useAsync from 'react-use/esm/useAsync';
import { Percentage } from './Percentage.esm.js';
import { Rating } from './Rating.esm.js';
import { RatingCard } from './RatingCard.esm.js';
import { Value } from './Value.esm.js';
import { InfoCard, Progress, EmptyState } from '@backstage/core-components';
import { useApi } from '@backstage/core-plugin-api';
import { DateTime } from 'luxon';

const useStyles = makeStyles((theme) => ({
  badgeLabel: {
    color: theme.palette.common.white
  },
  badgeError: {
    margin: 0,
    backgroundColor: theme.palette.error.main
  },
  badgeSuccess: {
    margin: 0,
    backgroundColor: theme.palette.success.main
  },
  badgeUnknown: {
    backgroundColor: theme.palette.grey[500]
  },
  header: {
    padding: theme.spacing(2, 2, 2, 2.5)
  },
  action: {
    margin: 0
  },
  lastAnalyzed: {
    color: theme.palette.text.secondary
  }
}));
const defaultDuplicationRatings = [
  { greaterThan: 0, rating: "1.0" },
  { greaterThan: 3, rating: "2.0" },
  { greaterThan: 5, rating: "3.0" },
  { greaterThan: 10, rating: "4.0" },
  { greaterThan: 20, rating: "5.0" }
];
const SonarQubeCard = (props) => {
  const {
    variant = "gridItem",
    duplicationRatings = defaultDuplicationRatings,
    missingAnnotationReadMoreUrl
  } = props;
  const { entity } = useEntity();
  const sonarQubeApi = useApi(sonarQubeApiRef);
  const { projectKey: projectTitle, projectInstance } = useProjectInfo(entity);
  const { value, loading } = useAsync(
    async () => sonarQubeApi.getFindingSummary({
      componentKey: projectTitle,
      projectInstance
    }),
    [sonarQubeApi, projectTitle]
  );
  const deepLink = !loading && value ? {
    title: "View more",
    link: value.projectUrl
  } : void 0;
  const classes = useStyles();
  let gateLabel = "Not computed";
  let gateColor = classes.badgeUnknown;
  if (value?.metrics.alert_status) {
    const gatePassed = value.metrics.alert_status === "OK";
    gateLabel = gatePassed ? "Gate passed" : "Gate failed";
    gateColor = gatePassed ? classes.badgeSuccess : classes.badgeError;
  }
  const qualityBadge = !loading && value && /* @__PURE__ */ React.createElement(
    Chip,
    {
      label: gateLabel,
      classes: { root: gateColor, label: classes.badgeLabel }
    }
  );
  const duplicationRating = useMemo(() => {
    if (loading || !value || !value.metrics.duplicated_lines_density) {
      return "";
    }
    let rating = "";
    for (const r of duplicationRatings) {
      if (+value.metrics.duplicated_lines_density >= r.greaterThan) {
        rating = r.rating;
      }
    }
    return rating;
  }, [loading, value, duplicationRatings]);
  return /* @__PURE__ */ React.createElement(
    InfoCard,
    {
      title: "Code Quality",
      deepLink,
      variant,
      headerProps: {
        action: qualityBadge,
        classes: {
          root: classes.header,
          action: classes.action
        }
      }
    },
    loading && /* @__PURE__ */ React.createElement(Progress, null),
    !loading && !projectTitle && /* @__PURE__ */ React.createElement(
      MissingAnnotationEmptyState,
      {
        annotation: SONARQUBE_PROJECT_KEY_ANNOTATION,
        readMoreUrl: missingAnnotationReadMoreUrl
      }
    ),
    !loading && projectTitle && !value && /* @__PURE__ */ React.createElement(
      EmptyState,
      {
        missing: "info",
        title: "No information to display",
        description: `There is no SonarQube project with key '${projectTitle}'.`
      }
    ),
    !loading && value && /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(
      Grid,
      {
        item: true,
        container: true,
        direction: "column",
        justifyContent: "space-between",
        alignItems: "center",
        style: { height: "100%" },
        spacing: 0
      },
      /* @__PURE__ */ React.createElement(Grid, { item: true, container: true, justifyContent: "space-around" }, /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          titleIcon: /* @__PURE__ */ React.createElement(BugReport, null),
          title: "Bugs",
          link: value.getIssuesUrl("BUG"),
          leftSlot: /* @__PURE__ */ React.createElement(Value, { value: value.metrics.bugs }),
          rightSlot: /* @__PURE__ */ React.createElement(Rating, { rating: value.metrics.reliability_rating })
        }
      ), /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          titleIcon: value.metrics.vulnerabilities === "0" ? /* @__PURE__ */ React.createElement(Lock, null) : /* @__PURE__ */ React.createElement(LockOpen, null),
          title: "Vulnerabilities",
          link: value.getIssuesUrl("VULNERABILITY"),
          leftSlot: /* @__PURE__ */ React.createElement(Value, { value: value.metrics.vulnerabilities }),
          rightSlot: /* @__PURE__ */ React.createElement(Rating, { rating: value.metrics.security_rating })
        }
      ), /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          titleIcon: value.metrics.code_smells === "0" ? /* @__PURE__ */ React.createElement(SentimentVerySatisfied, null) : /* @__PURE__ */ React.createElement(SentimentVeryDissatisfied, null),
          title: "Code Smells",
          link: value.getIssuesUrl("CODE_SMELL"),
          leftSlot: /* @__PURE__ */ React.createElement(Value, { value: value.metrics.code_smells }),
          rightSlot: /* @__PURE__ */ React.createElement(Rating, { rating: value.metrics.sqale_rating })
        }
      ), value.metrics.security_review_rating && /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          titleIcon: /* @__PURE__ */ React.createElement(Security, null),
          title: "Hotspots Reviewed",
          link: value.getSecurityHotspotsUrl(),
          leftSlot: /* @__PURE__ */ React.createElement(
            Value,
            {
              value: value.metrics.security_hotspots_reviewed ? `${value.metrics.security_hotspots_reviewed}%` : "\u2014"
            }
          ),
          rightSlot: /* @__PURE__ */ React.createElement(Rating, { rating: value.metrics.security_review_rating })
        }
      ), /* @__PURE__ */ React.createElement("div", { style: { width: "100%" } }), /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          link: value.getComponentMeasuresUrl("COVERAGE"),
          title: "Coverage",
          leftSlot: /* @__PURE__ */ React.createElement(Percentage, { value: value.metrics.coverage }),
          rightSlot: /* @__PURE__ */ React.createElement(
            Value,
            {
              value: value.metrics.coverage !== void 0 ? `${value.metrics.coverage}%` : "\u2014"
            }
          )
        }
      ), /* @__PURE__ */ React.createElement(
        RatingCard,
        {
          title: "Duplications",
          link: value.getComponentMeasuresUrl("DUPLICATED_LINES_DENSITY"),
          leftSlot: /* @__PURE__ */ React.createElement(Rating, { rating: duplicationRating, hideValue: true }),
          rightSlot: /* @__PURE__ */ React.createElement(Value, { value: `${value.metrics.duplicated_lines_density}%` })
        }
      )),
      /* @__PURE__ */ React.createElement(Grid, { item: true, className: classes.lastAnalyzed }, "Last analyzed on", " ", DateTime.fromISO(value.lastAnalysis).toLocaleString(
        DateTime.DATETIME_MED
      ))
    ))
  );
};

export { SonarQubeCard };
