// @backstage-community/plugin-sonarqube/dist/plugin.esm.js
import { SonarQubeClient } from './api/SonarQubeClient.esm.js';
import { createPlugin, createApiFactory, discoveryApiRef, identityApiRef, createComponentExtension } from '@backstage/core-plugin-api';
import { sonarQubeApiRef } from '@backstage-community/plugin-sonarqube-react';

const sonarQubePlugin = createPlugin({
  id: "sonarqube",
  apis: [
    createApiFactory({
      api: sonarQubeApiRef,
      deps: {
        discoveryApi: discoveryApiRef,
        identityApi: identityApiRef
      },
      factory: ({ discoveryApi, identityApi }) => new SonarQubeClient({
        discoveryApi,
        identityApi
      })
    })
  ]
});
const EntitySonarQubeCard = sonarQubePlugin.provide(
  createComponentExtension({
    name: "EntitySonarQubeCard",
    component: {
      lazy: () => import('./components/SonarQubeCard/index.esm.js').then((m) => m.SonarQubeCard)
    }
  })
);
const EntitySonarQubeContentPage = sonarQubePlugin.provide(
  createComponentExtension({
    name: "EntitySonarQubeContentPage",
    component: {
      lazy: () => import('./components/SonarQubeContentPage/index.esm.js').then(
        (m) => m.SonarQubeContentPage
      )
    }
  })
);

export { EntitySonarQubeCard, EntitySonarQubeContentPage, sonarQubePlugin };
