// @backstage/backend-defaults/dist/entrypoints/auditor/WinstonRootAuditorService.cjs.js
'use strict';

var winston = require('winston');
require('../rootLogger/rootLoggerServiceFactory.cjs.js');
var WinstonLogger = require('../rootLogger/WinstonLogger.cjs.js');
var DefaultAuditorService = require('./DefaultAuditorService.cjs.js');

function _interopNamespaceCompat(e) {
  if (e && typeof e === 'object' && 'default' in e) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var winston__namespace = /*#__PURE__*/_interopNamespaceCompat(winston);

const defaultFormatter = winston__namespace.format.combine(
  winston__namespace.format.timestamp({
    format: "YYYY-MM-DD HH:mm:ss"
  }),
  winston__namespace.format.errors({ stack: true }),
  winston__namespace.format.splat(),
  winston__namespace.format.json()
);
const auditorFieldFormat = winston__namespace.format((info) => {
  return { ...info, isAuditEvent: true };
})();
class WinstonRootAuditorService {
  constructor(winstonLogger) {
    this.winstonLogger = winstonLogger;
  }
  /**
   * Creates a {@link WinstonRootAuditorService} instance.
   */
  static create(options) {
    let winstonLogger = WinstonLogger.WinstonLogger.create({
      meta: {
        service: "backstage"
      },
      level: "info",
      format: winston__namespace.format.combine(
        auditorFieldFormat,
        options?.format ?? defaultFormatter
      ),
      transports: options?.transports
    });
    if (options?.meta) {
      winstonLogger = winstonLogger.child(options.meta);
    }
    return new WinstonRootAuditorService(winstonLogger);
  }
  forPlugin(deps) {
    return DefaultAuditorService.DefaultAuditorService.create(
      (e) => this.winstonLogger.info(`${e.plugin}.${e.eventId}`, e),
      deps
    );
  }
}

exports.WinstonRootAuditorService = WinstonRootAuditorService;
exports.auditorFieldFormat = auditorFieldFormat;
exports.defaultFormatter = defaultFormatter;
