// @backstage/plugin-home/dist/alpha.esm.js
import { jsx } from 'react/jsx-runtime';
import { createRouteRef, createExtensionDataRef, PageBlueprint, createExtensionInput, coreExtensionData, AppRootElementBlueprint, ApiBlueprint, identityApiRef, storageApiRef, createFrontendPlugin } from '@backstage/frontend-plugin-api';
import { compatWrapper } from '@backstage/core-compat-api';
import 'react-router-dom';
import './components/CustomHomepage/CustomHomepageGrid.esm.js';
import { VisitListener } from './components/VisitListener.esm.js';
import './components/VisitList/Context.esm.js';
import { VisitsStorageApi } from './api/VisitsStorageApi.esm.js';
import '@backstage/core-app-api';
import { visitsApiRef } from './api/VisitsApi.esm.js';
export { homeTranslationRef } from './translation.esm.js';

const rootRouteRef = createRouteRef();
const titleExtensionDataRef = createExtensionDataRef().with({
  id: "title"
});
const homePage = PageBlueprint.makeWithOverrides({
  inputs: {
    props: createExtensionInput(
      [
        coreExtensionData.reactElement.optional(),
        titleExtensionDataRef.optional()
      ],
      {
        singleton: true,
        optional: true
      }
    )
  },
  factory: (originalFactory, { inputs }) => {
    return originalFactory({
      path: "/home",
      routeRef: rootRouteRef,
      loader: () => import('./components/index.esm.js').then(
        (m) => compatWrapper(
          /* @__PURE__ */ jsx(
            m.HomepageCompositionRoot,
            {
              children: inputs.props?.get(coreExtensionData.reactElement),
              title: inputs.props?.get(titleExtensionDataRef)
            }
          )
        )
      )
    });
  }
});
const visitListenerAppRootElement = AppRootElementBlueprint.make({
  name: "visit-listener",
  params: {
    element: /* @__PURE__ */ jsx(VisitListener, {})
  }
});
const visitsApi = ApiBlueprint.make({
  name: "visits",
  params: (defineParams) => defineParams({
    api: visitsApiRef,
    deps: {
      storageApi: storageApiRef,
      identityApi: identityApiRef
    },
    factory: ({ storageApi, identityApi }) => VisitsStorageApi.create({ storageApi, identityApi })
  })
});
var alpha = createFrontendPlugin({
  pluginId: "home",
  info: { packageJson: () => import('./package.json.esm.js') },
  extensions: [homePage, visitsApi, visitListenerAppRootElement],
  routes: {
    root: rootRouteRef
  }
});

export { alpha as default, titleExtensionDataRef };
