// @backstage/plugin-home/dist/index.esm.js
export { ComponentAccordion, ComponentTab, ComponentTabs, FeaturedDocsCard, HeaderWorldClock, HomePageCompanyLogo, HomePageRandomJoke, HomePageRecentlyVisited, HomePageStarredEntities, HomePageToolkit, HomePageTopVisited, HomepageCompositionRoot, QuickStartCard, WelcomeTitle, homePlugin } from './plugin.esm.js';
import 'react/jsx-runtime';
import 'react-router-dom';
export { CustomHomepageGrid } from './components/CustomHomepage/CustomHomepageGrid.esm.js';
export { VisitListener } from './components/VisitListener.esm.js';
export { VisitDisplayProvider, useVisitDisplay } from './components/VisitList/Context.esm.js';
export { TemplateBackstageLogo } from './assets/TemplateBackstageLogo.esm.js';
export { TemplateBackstageLogoIcon } from './assets/TemplateBackstageLogoIcon.esm.js';
export { SettingsModal, createCardExtension } from './deprecated.esm.js';
export { VisitsStorageApi } from './api/VisitsStorageApi.esm.js';
export { VisitsWebStorageApi } from './api/VisitsWebStorageApi.esm.js';
export { isOperator, visitsApiRef } from './api/VisitsApi.esm.js';
