// @backstage-community/plugin-tech-insights-backend-module-jsonfc/dist/module/techInsightsModuleJsonRulesEngineFactCheckerFactory.cjs.js
'use strict';

var backendPluginApi = require('@backstage/backend-plugin-api');
var pluginTechInsightsNode = require('@backstage-community/plugin-tech-insights-node');
var JsonRulesEngineFactChecker = require('../service/JsonRulesEngineFactChecker.cjs.js');

const techInsightsModuleJsonRulesEngineFactCheckerFactory = backendPluginApi.createBackendModule({
  pluginId: "tech-insights",
  moduleId: "json-rules-engine-fact-checker-factory",
  register(env) {
    env.registerInit({
      deps: {
        config: backendPluginApi.coreServices.rootConfig,
        logger: backendPluginApi.coreServices.logger,
        techInsights: pluginTechInsightsNode.techInsightsFactCheckerFactoryExtensionPoint
      },
      async init({ config, logger, techInsights }) {
        const factory = JsonRulesEngineFactChecker.JsonRulesEngineFactCheckerFactory.fromConfig(config, {
          logger
        });
        techInsights.setFactCheckerFactory(factory);
      }
    });
  }
});

exports.techInsightsModuleJsonRulesEngineFactCheckerFactory = techInsightsModuleJsonRulesEngineFactCheckerFactory;
