// @backstage/plugin-auth-backend/dist/service/router.cjs.js
'use strict';

var express = require('express');
var Router = require('express-promise-router');
var cookieParser = require('cookie-parser');
var errors = require('@backstage/errors');
var KeyStores = require('../identity/KeyStores.cjs.js');
var TokenFactory = require('../identity/TokenFactory.cjs.js');
var UserInfoDatabase = require('../database/UserInfoDatabase.cjs.js');
var session = require('express-session');
var connectSessionKnex = require('connect-session-knex');
var passport = require('passport');
var AuthDatabase = require('../database/AuthDatabase.cjs.js');
var readTokenExpiration = require('./readTokenExpiration.cjs.js');
var StaticTokenIssuer = require('../identity/StaticTokenIssuer.cjs.js');
var StaticKeyStore = require('../identity/StaticKeyStore.cjs.js');
var router = require('../providers/router.cjs.js');
var OidcRouter = require('./OidcRouter.cjs.js');
var OidcDatabase = require('../database/OidcDatabase.cjs.js');

function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e : { default: e }; }

var express__default = /*#__PURE__*/_interopDefaultCompat(express);
var Router__default = /*#__PURE__*/_interopDefaultCompat(Router);
var cookieParser__default = /*#__PURE__*/_interopDefaultCompat(cookieParser);
var session__default = /*#__PURE__*/_interopDefaultCompat(session);
var connectSessionKnex__default = /*#__PURE__*/_interopDefaultCompat(connectSessionKnex);
var passport__default = /*#__PURE__*/_interopDefaultCompat(passport);

async function createRouter(options) {
  const {
    logger,
    config,
    discovery,
    database: db,
    tokenFactoryAlgorithm,
    providerFactories = {},
    httpAuth
  } = options;
  const router$1 = Router__default.default();
  const appUrl = config.getString("app.baseUrl");
  const authUrl = await discovery.getExternalBaseUrl("auth");
  const backstageTokenExpiration = readTokenExpiration.readBackstageTokenExpiration(config);
  const database = AuthDatabase.AuthDatabase.create(db);
  const keyStore = await KeyStores.KeyStores.fromConfig(config, {
    logger,
    database
  });
  const userInfo = await UserInfoDatabase.UserInfoDatabase.create({
    database
  });
  const omitClaimsFromToken = config.getOptionalBoolean(
    "auth.omitIdentityTokenOwnershipClaim"
  ) ? ["ent"] : [];
  const createTokenIssuer = (opts) => {
    if (keyStore instanceof StaticKeyStore.StaticKeyStore) {
      return new StaticTokenIssuer.StaticTokenIssuer(
        {
          logger: opts.logger,
          issuer: authUrl,
          sessionExpirationSeconds: opts.expirationSeconds,
          omitClaimsFromToken
        },
        keyStore
      );
    }
    return new TokenFactory.TokenFactory({
      issuer: authUrl,
      keyStore,
      keyDurationSeconds: opts.expirationSeconds,
      logger: opts.logger,
      algorithm: tokenFactoryAlgorithm ?? config.getOptionalString("auth.identityTokenAlgorithm"),
      omitClaimsFromToken
    });
  };
  const tokenIssuer = createTokenIssuer({
    logger: logger.child({ component: "token-factory" }),
    expirationSeconds: backstageTokenExpiration
  });
  const secret = config.getOptionalString("auth.session.secret");
  if (secret) {
    router$1.use(cookieParser__default.default(secret));
    const enforceCookieSSL = authUrl.startsWith("https");
    const KnexSessionStore = connectSessionKnex__default.default(session__default.default);
    router$1.use(
      session__default.default({
        secret,
        saveUninitialized: false,
        resave: false,
        cookie: { secure: enforceCookieSSL ? "auto" : false },
        store: new KnexSessionStore({
          createtable: false,
          knex: await database.get()
        })
      })
    );
    router$1.use(passport__default.default.initialize());
    router$1.use(passport__default.default.session());
  } else {
    router$1.use(cookieParser__default.default());
  }
  router$1.use(express__default.default.urlencoded({ extended: false }));
  router$1.use(express__default.default.json());
  router.bindProviderRouters(router$1, {
    providers: providerFactories,
    appUrl,
    baseUrl: authUrl,
    tokenIssuer,
    ...options,
    auth: options.auth,
    userInfo
  });
  const dcrTokenExpiration = readTokenExpiration.readDcrTokenExpiration(config);
  const oidcTokenIssuer = createTokenIssuer({
    logger: logger.child({ component: "oidc-token-factory" }),
    expirationSeconds: dcrTokenExpiration
  });
  const oidc = await OidcDatabase.OidcDatabase.create({ database });
  const oidcRouter = OidcRouter.OidcRouter.create({
    auth: options.auth,
    tokenIssuer: oidcTokenIssuer,
    baseUrl: authUrl,
    appUrl,
    userInfo,
    oidc,
    logger,
    httpAuth,
    config
  });
  router$1.use(oidcRouter.getRouter());
  router$1.use("/:provider/", (req) => {
    const { provider } = req.params;
    throw new errors.NotFoundError(`Unknown auth provider '${provider}'`);
  });
  return router$1;
}

exports.createRouter = createRouter;
