// @backstage/ui/dist/components/SearchField/SearchField.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { forwardRef, useState, useEffect } from 'react';
import { SearchField as SearchField$1, Input, Button } from 'react-aria-components';
import clsx from 'clsx';
import { FieldLabel } from '../FieldLabel/FieldLabel.esm.js';
import { FieldError } from '../FieldError/FieldError.esm.js';
import { RiSearch2Line, RiCloseCircleLine } from '@remixicon/react';
import { useStyles } from '../../hooks/useStyles.esm.js';
import { SearchFieldDefinition } from './definition.esm.js';
import styles from './SearchField.module.css.esm.js';

const SearchField = forwardRef(
  (props, ref) => {
    const {
      label,
      "aria-label": ariaLabel,
      "aria-labelledby": ariaLabelledBy
    } = props;
    const [isCollapsed, setIsCollapsed] = useState(false);
    const [shouldCollapse, setShouldCollapse] = useState(true);
    useEffect(() => {
      if (!label && !ariaLabel && !ariaLabelledBy) {
        console.warn(
          "SearchField requires either a visible label, aria-label, or aria-labelledby for accessibility"
        );
      }
    }, [label, ariaLabel, ariaLabelledBy]);
    const { classNames, dataAttributes, style, cleanedProps } = useStyles(
      SearchFieldDefinition,
      {
        size: "small",
        placeholder: "Search",
        startCollapsed: false,
        ...props
      }
    );
    const {
      className,
      description,
      icon,
      isRequired,
      secondaryLabel,
      placeholder,
      startCollapsed,
      ...rest
    } = cleanedProps;
    const secondaryLabelText = secondaryLabel || (isRequired ? "Required" : null);
    const handleClick = (isFocused) => {
      props.onFocusChange?.(isFocused);
      if (shouldCollapse) {
        if (isFocused) {
          setIsCollapsed(true);
        } else {
          setIsCollapsed(false);
        }
      }
    };
    const handleChange = (value) => {
      props.onChange?.(value);
      if (value.length > 0) {
        setShouldCollapse(false);
      } else {
        setShouldCollapse(true);
      }
    };
    return /* @__PURE__ */ jsxs(
      SearchField$1,
      {
        className: clsx(classNames.root, styles[classNames.root], className),
        ...dataAttributes,
        "aria-label": ariaLabel,
        "aria-labelledby": ariaLabelledBy,
        "data-collapsed": isCollapsed,
        onFocusChange: handleClick,
        onChange: handleChange,
        style,
        ...rest,
        ref,
        children: [
          /* @__PURE__ */ jsx(
            FieldLabel,
            {
              label,
              secondaryLabel: secondaryLabelText,
              description
            }
          ),
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: clsx(
                classNames.inputWrapper,
                styles[classNames.inputWrapper]
              ),
              "data-size": dataAttributes["data-size"],
              children: [
                icon !== false && /* @__PURE__ */ jsx(
                  "div",
                  {
                    className: clsx(
                      classNames.inputIcon,
                      styles[classNames.inputIcon]
                    ),
                    "data-size": dataAttributes["data-size"],
                    "aria-hidden": "true",
                    children: icon || /* @__PURE__ */ jsx(RiSearch2Line, {})
                  }
                ),
                /* @__PURE__ */ jsx(
                  Input,
                  {
                    className: clsx(classNames.input, styles[classNames.input]),
                    ...icon !== false && { "data-icon": true },
                    placeholder
                  }
                ),
                /* @__PURE__ */ jsx(
                  Button,
                  {
                    className: clsx(classNames.clear, styles[classNames.clear]),
                    "data-size": dataAttributes["data-size"],
                    children: /* @__PURE__ */ jsx(RiCloseCircleLine, {})
                  }
                )
              ]
            }
          ),
          /* @__PURE__ */ jsx(FieldError, {})
        ]
      }
    );
  }
);
SearchField.displayName = "searchField";

export { SearchField };
