// @backstage/plugin-scaffolder-backend/dist/scaffolder/actions/TemplateActionRegistry.cjs.js
'use strict';

var errors = require('@backstage/errors');
var lodash = require('lodash');

class DefaultTemplateActionRegistry {
  constructor(actionsRegistry, logger) {
    this.actionsRegistry = actionsRegistry;
    this.logger = logger;
  }
  actions = /* @__PURE__ */ new Map();
  register(action) {
    if (this.actions.has(action.id)) {
      throw new errors.ConflictError(
        `Template action with ID '${action.id}' has already been registered`
      );
    }
    this.actions.set(action.id, action);
  }
  async get(actionId, options) {
    const action = (await this.list(options)).get(actionId);
    if (!action) {
      throw new errors.NotFoundError(
        `Template action with ID '${actionId}' is not registered. See https://backstage.io/docs/features/software-templates/builtin-actions/ on how to add a new action module.`
      );
    }
    return action;
  }
  async list(options) {
    const ret = new Map(this.actions);
    const { actions } = await this.actionsRegistry.list({
      credentials: options.credentials
    });
    for (const action of actions) {
      if (ret.has(action.id)) {
        this.logger.warn(
          `Template action with ID '${action.id}' has already been registered, skipping action provided by actions service`
        );
        continue;
      }
      ret.set(action.id, {
        id: action.id,
        description: action.description,
        examples: [],
        supportsDryRun: action.attributes?.readOnly === true && action.attributes?.destructive === false,
        handler: async (ctx) => {
          const { output } = await this.actionsRegistry.invoke({
            id: action.id,
            input: ctx.input,
            credentials: await ctx.getInitiatorCredentials()
          });
          if (lodash.isPlainObject(output)) {
            for (const [key, value] of Object.entries(output)) {
              ctx.output(key, value);
            }
          }
        },
        schema: {
          input: action.schema.input,
          output: action.schema.output
        }
      });
    }
    return ret;
  }
}

exports.DefaultTemplateActionRegistry = DefaultTemplateActionRegistry;
