// @backstage/plugin-home/dist/components/CustomHomepage/AddWidgetDialog.esm.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import AddIcon from '@material-ui/icons/Add';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const getTitle = (widget) => {
  return widget.title || widget.name;
};
const AddWidgetDialog = (props) => {
  const { widgets, handleAdd } = props;
  const { t } = useTranslationRef(homeTranslationRef);
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(DialogTitle, { children: t("addWidgetDialog.title") }),
    /* @__PURE__ */ jsx(DialogContent, { children: /* @__PURE__ */ jsx(List, { dense: true, children: widgets.map((widget) => {
      return /* @__PURE__ */ jsxs(
        ListItem,
        {
          button: true,
          onClick: () => handleAdd(widget),
          children: [
            /* @__PURE__ */ jsx(ListItemAvatar, { children: /* @__PURE__ */ jsx(AddIcon, {}) }),
            /* @__PURE__ */ jsx(
              ListItemText,
              {
                secondary: widget.description && /* @__PURE__ */ jsx(
                  Typography,
                  {
                    component: "span",
                    variant: "caption",
                    color: "textPrimary",
                    children: widget.description
                  }
                ),
                primary: /* @__PURE__ */ jsx(Typography, { variant: "body1", color: "textPrimary", children: getTitle(widget) })
              }
            )
          ]
        },
        widget.name
      );
    }) }) })
  ] });
};

export { AddWidgetDialog };
