// @backstage/plugin-home/dist/components/CustomHomepage/CustomHomepageButtons.esm.js
import { jsx, Fragment, jsxs } from 'react/jsx-runtime';
import Button from '@material-ui/core/Button';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import SaveIcon from '@material-ui/icons/Save';
import DeleteIcon from '@material-ui/icons/Delete';
import AddIcon from '@material-ui/icons/Add';
import EditIcon from '@material-ui/icons/Edit';
import CancelIcon from '@material-ui/icons/Cancel';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const useStyles = makeStyles(
  (theme) => createStyles({
    contentHeaderBtn: {
      marginLeft: theme.spacing(2)
    },
    widgetWrapper: {
      "& > *:first-child": {
        width: "100%",
        height: "100%"
      }
    }
  })
);
const CustomHomepageButtons = (props) => {
  const {
    editMode,
    numWidgets,
    clearLayout,
    setAddWidgetDialogOpen,
    changeEditMode,
    defaultConfigAvailable,
    restoreDefault
  } = props;
  const styles = useStyles();
  const { t } = useTranslationRef(homeTranslationRef);
  return /* @__PURE__ */ jsx(Fragment, { children: !editMode && numWidgets > 0 ? /* @__PURE__ */ jsx(
    Button,
    {
      variant: "contained",
      color: "primary",
      onClick: () => changeEditMode(true),
      size: "small",
      startIcon: /* @__PURE__ */ jsx(EditIcon, {}),
      children: t("customHomepageButtons.edit")
    }
  ) : /* @__PURE__ */ jsxs(Fragment, { children: [
    defaultConfigAvailable && /* @__PURE__ */ jsx(
      Button,
      {
        variant: "contained",
        className: styles.contentHeaderBtn,
        onClick: restoreDefault,
        size: "small",
        startIcon: /* @__PURE__ */ jsx(CancelIcon, {}),
        children: t("customHomepageButtons.restoreDefaults")
      }
    ),
    numWidgets > 0 && /* @__PURE__ */ jsx(
      Button,
      {
        variant: "contained",
        color: "secondary",
        className: styles.contentHeaderBtn,
        onClick: clearLayout,
        size: "small",
        startIcon: /* @__PURE__ */ jsx(DeleteIcon, {}),
        children: t("customHomepageButtons.clearAll")
      }
    ),
    /* @__PURE__ */ jsx(
      Button,
      {
        variant: "contained",
        className: styles.contentHeaderBtn,
        onClick: () => setAddWidgetDialogOpen(true),
        size: "small",
        startIcon: /* @__PURE__ */ jsx(AddIcon, {}),
        children: t("customHomepageButtons.addWidget")
      }
    ),
    numWidgets > 0 && /* @__PURE__ */ jsx(
      Button,
      {
        className: styles.contentHeaderBtn,
        variant: "contained",
        color: "primary",
        onClick: () => changeEditMode(false),
        size: "small",
        startIcon: /* @__PURE__ */ jsx(SaveIcon, {}),
        children: t("customHomepageButtons.save")
      }
    )
  ] }) });
};

export { CustomHomepageButtons };
