// @backstage/plugin-home/dist/components/CustomHomepage/WidgetSettingsOverlay.esm.js
import { jsxs, jsx } from 'react/jsx-runtime';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import DeleteIcon from '@material-ui/icons/Delete';
import SettingsIcon from '@material-ui/icons/Settings';
import { useState } from 'react';
import { withTheme } from '@rjsf/core';
import { Theme } from '@rjsf/material-ui';
import validator from '@rjsf/validator-ajv8';
import { useTranslationRef } from '@backstage/frontend-plugin-api';
import { homeTranslationRef } from '../../translation.esm.js';

const Form = withTheme(Theme);
const useStyles = makeStyles(
  (theme) => createStyles({
    iconGrid: {
      height: "100%",
      "& *": {
        padding: 0
      }
    },
    settingsOverlay: {
      position: "absolute",
      backgroundColor: "rgba(40, 40, 40, 0.93)",
      width: "100%",
      height: "100%",
      top: 0,
      left: 0,
      padding: theme.spacing(2),
      color: "white"
    }
  })
);
const WidgetSettingsOverlay = (props) => {
  const { id, widget, settings, handleRemove, handleSettingsSave, deletable } = props;
  const [settingsDialogOpen, setSettingsDialogOpen] = useState(false);
  const styles = useStyles();
  const onClose = () => setSettingsDialogOpen(false);
  const { t } = useTranslationRef(homeTranslationRef);
  return /* @__PURE__ */ jsxs("div", { className: styles.settingsOverlay, children: [
    widget.settingsSchema && /* @__PURE__ */ jsx(
      Dialog,
      {
        open: settingsDialogOpen,
        className: "widgetSettingsDialog",
        onClose,
        children: /* @__PURE__ */ jsx(DialogContent, { children: /* @__PURE__ */ jsx(
          Form,
          {
            validator,
            showErrorList: false,
            schema: widget.settingsSchema,
            uiSchema: widget.uiSchema,
            noHtml5Validate: true,
            formData: settings,
            formContext: { settings },
            onSubmit: ({ formData, errors }) => {
              if (errors.length === 0) {
                handleSettingsSave(id, formData);
                setSettingsDialogOpen(false);
              }
            },
            experimental_defaultFormStateBehavior: {
              allOf: "populateDefaults"
            },
            children: /* @__PURE__ */ jsxs(DialogActions, { children: [
              /* @__PURE__ */ jsx(Button, { color: "primary", variant: "contained", type: "submit", children: t("widgetSettingsOverlay.submitButtonTitle") }),
              /* @__PURE__ */ jsx(Button, { color: "secondary", onClick: onClose, children: t("widgetSettingsOverlay.cancelButtonTitle") })
            ] })
          }
        ) })
      }
    ),
    /* @__PURE__ */ jsxs(
      Grid,
      {
        container: true,
        className: styles.iconGrid,
        alignItems: "center",
        justifyContent: "center",
        children: [
          widget.settingsSchema && /* @__PURE__ */ jsx(Grid, { item: true, className: "overlayGridItem", children: /* @__PURE__ */ jsx(Tooltip, { title: t("widgetSettingsOverlay.editSettingsTooptip"), children: /* @__PURE__ */ jsx(
            IconButton,
            {
              color: "primary",
              onClick: () => setSettingsDialogOpen(true),
              children: /* @__PURE__ */ jsx(SettingsIcon, { fontSize: "large" })
            }
          ) }) }),
          deletable !== false && /* @__PURE__ */ jsx(Grid, { item: true, className: "overlayGridItem", children: /* @__PURE__ */ jsx(Tooltip, { title: t("widgetSettingsOverlay.deleteWidgetTooltip"), children: /* @__PURE__ */ jsx(IconButton, { color: "secondary", onClick: () => handleRemove(id), children: /* @__PURE__ */ jsx(DeleteIcon, { fontSize: "large" }) }) }) })
        ]
      }
    )
  ] });
};

export { WidgetSettingsOverlay };
