// @backstage/core-plugin-api/dist/routing/RouteRef.esm.js
import { routeRefType } from './types.esm.js';

class RouteRefImpl {
  [routeRefType] = "absolute";
  id;
  params;
  constructor(id, params) {
    this.id = id;
    this.params = params;
  }
  get title() {
    return this.id;
  }
  toString() {
    if (this.#nfsId) {
      return `routeRef{id=${this.#nfsId},legacyId=${this.id}}`;
    }
    return `routeRef{type=absolute,id=${this.id}}`;
  }
  // NFS implementation below
  $$type = "@backstage/RouteRef";
  version = "v1";
  T = void 0;
  alias = void 0;
  #nfsId = void 0;
  getParams() {
    return this.params;
  }
  getDescription() {
    if (this.#nfsId) {
      return this.#nfsId;
    }
    return this.id;
  }
  setId(newId) {
    if (!newId) {
      throw new Error(`RouteRef id must be a non-empty string`);
    }
    if (this.#nfsId && this.#nfsId !== newId) {
      throw new Error(
        `RouteRef was referenced twice as both '${this.#nfsId}' and '${newId}'`
      );
    }
    this.#nfsId = newId;
  }
}
function createRouteRef(config) {
  return new RouteRefImpl(
    config.id,
    config.params ?? []
  );
}

export { RouteRefImpl, createRouteRef };
