// @backstage/plugin-home/dist/componentRenderers/ComponentAccordion.esm.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { useState } from 'react';
import { SettingsModal } from '@backstage/plugin-home-react';
import Accordion from '@material-ui/core/Accordion';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import { makeStyles } from '@material-ui/core/styles';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import SettingsIcon from '@material-ui/icons/Settings';

const useStyles = makeStyles((theme) => ({
  settingsIconButton: {
    padding: theme.spacing(0, 1, 0, 0)
  },
  contentContainer: {
    width: "100%"
  }
}));
const ComponentAccordion = (props) => {
  const {
    title,
    expanded = false,
    Content,
    Actions,
    Settings,
    ContextProvider,
    ...childProps
  } = props;
  const classes = useStyles();
  const [settingsIsExpanded, setSettingsIsExpanded] = useState(false);
  const [isExpanded, setIsExpanded] = useState(expanded);
  const handleOpenSettings = (e) => {
    e.stopPropagation();
    setSettingsIsExpanded((prevState) => !prevState);
  };
  const innerContent = /* @__PURE__ */ jsxs(Fragment, { children: [
    Settings && /* @__PURE__ */ jsx(
      SettingsModal,
      {
        open: settingsIsExpanded,
        close: () => setSettingsIsExpanded(false),
        componentName: title,
        children: /* @__PURE__ */ jsx(Settings, {})
      }
    ),
    /* @__PURE__ */ jsxs(
      Accordion,
      {
        expanded: isExpanded,
        onChange: (_e, expandedValue) => setIsExpanded(expandedValue),
        children: [
          /* @__PURE__ */ jsxs(AccordionSummary, { expandIcon: /* @__PURE__ */ jsx(ExpandMoreIcon, {}), children: [
            Settings && /* @__PURE__ */ jsx(
              IconButton,
              {
                onClick: handleOpenSettings,
                className: classes.settingsIconButton,
                children: /* @__PURE__ */ jsx(SettingsIcon, {})
              }
            ),
            /* @__PURE__ */ jsx(Typography, { children: title })
          ] }),
          /* @__PURE__ */ jsx(AccordionDetails, { children: /* @__PURE__ */ jsxs("div", { className: classes.contentContainer, children: [
            /* @__PURE__ */ jsx(Content, {}),
            Actions && /* @__PURE__ */ jsx(Actions, {})
          ] }) })
        ]
      }
    )
  ] });
  return ContextProvider ? /* @__PURE__ */ jsx(ContextProvider, { ...childProps, children: innerContent }) : innerContent;
};

export { ComponentAccordion };
